---
permalink: /cbio-who/
title: "Lab members"
excerpt: "The team"
modified: 2022-01-14
author_profile: false
sidebar:
  nav: "cbio"
---

{% include toc %}


## CBIO core

- [**Laurent Gatto**](http://lgatto.github.io/about/) (ORCID:
  [0000-0002-1520-2268](https://orcid.org/0000-0002-1520-2268),
  Github: [lgatto](https://github.com/lgatto/), fosstodon:
  [@lgatto](https://fosstodon.org/@lgatto)), head of the unit. He runs
  the CBIO group and teaches bioinformatics in the Facutly of Pharmacy
  and Biomedical Sciences (FASB).

- **Axelle Loriot** (ORCID:
  [0000-0002-5288-8561](https://orcid.org/0000-0002-5288-8561),
  Github: [aloriot](https://github.com/aloriot)) is a senior
  researcher at the de Duve Institute. Since 2001, she has been part
  of the epigenetics group led by Charles De Smet. She joined the CBIO
  group in 2018, focusing on transcriptomics and epigenetics. She
  developed algorithms to identify aberrant transcripts induced by DNA
  hypomethylation in tumor cells, and DNA hypermethylation events
  linked to overlapping expression. Axelle plays a major role in the
  bioinformatics core facility.

- **Philippe Hauchamps** (Github:
  [phauchamps](https://github.com/phauchamps)) is, since September
  2021, a PhD student working on flow cytometry data analyis. He
  joined the CBIO group in October 2019 as an intern and carried on
  with a Master's project in computational proteomics, developing
  Bayesian models to analyse quantitative proteomics data. He has a
  Master degree in Computer Science Engineering and one in
  Biostatistics.

- **Julie Devis** (Github:
  [JulieDevis](https://github.com/JulieDevis)) did her master's
  project in the CBIO lab and is now, since September 2022, a PhD
  student in biomedical sciences. She is working on deciphering
  complex mechanisms of epigenetic alterations in cancer genomes. She
  is pursuing her research in collaboration with Axelle Loriot and
  [Prof. Charles De
  Smet](https://www.deduveinstitute.be/cancer-epigenetics) from the
  epigenetics group.

- **Samuel Grégoire** (Github:
  [samgregoire](https://github.com/samgregoire)) did his master's
  project in the CBIO lab and is now, since September 2022, a PhD
  student in biomedical sciences. He is working on single-cell biology
  in AML, including single-cell proteomics, in collaboration with
  Didier Vertommen, who runs the institute's mass spectrometry core
  facility.

- **Andrea Vicini** (Github:
  [andreavicini](https://github.com/andreavicini)) joined the lab in
  October 2022 as a research assistant in bioinformatics. He focuses
  on integrative multi-omics approach to study infectious diseases as
  part of the large, inter-institute
  [HYGIEIA](https://www.saintluc.be/fr/hygieia) project.

- **Enes Sefa Ayar** (Github:
  [EnesSefaAyar](https://github.com/EnesSefaAyar)) joined the lab in
  November 2023 as a PhD student working on quantitative analysis of
  single-cell proteomics. He did a master's in computational sciences
  and engineering at the Koç University. He is funded by a FNRS
  'Projet de Recherche' (PDR) awarded to L. Gatto.

- **Jérôme Linden** (GitHub:
  [jerlinden](https://github.com/jerlinden)) joined the lab in
  November 2023 as a Project Manager for
  [e-OMIX](https://www.eomix.be), a CBIO-led project aiming at
  establishing a collaborative platform for the sharing and analyses
  of multiomics data. He has a PhD from ULiège and is funded by the
  European Regional Development Fund (ERDF) awarded to L. Gatto and
  S. Jodogne.

- **Loïc Guille** (ORCID:
  [0000-0002-8387-1092](https://orcid.org/0000-0002-8387-1092),
  GitHub: [loguille](https://github.com/loguille)) joined the lab in
  May 2024 as a bioinformatician for [e-OMIX](https://www.eomix.be) a
  CBIO-led project.  He has a master's degree in bioinformatic from
  Lyon 1 university.

- **Riya Dogra** (GitHub: [rdogra8](https://github.com/rdogra8)), is a
  PhD student at the de Duve Institute since June 2024, working on
  analysing PTMs using spatial and single cell proteomics
  approaches. She has a Master's in Systems Biology and Bioinformatics
  from Panjab University, and worked in the Genomics Research lab at
  CSIR-CCMB.

- **Guillaume Deflandre** (GitHub: [guideflandre](https://github.com/guideflandre)),
  did his Master's thesis in the CBIO lab and is now pursuing a PhD in
  biomedical sciences at the de Duve institute since October 2024. His
  project is centered around the improvement of peptide identifications
  in single-cell proteomics. He has a Master's degree in chemistry and
  bio-industries from the UCLouvain.

## Undergraduates

- **Léopold Guyot** (GitHub: [leopoldguyot](https://github.com/leopoldguyot)), is a final year Bioinformatics student at the "Université Libre de Bruxelles" (ULB).
    He is currently working on his Master's thesis on single-cell proteomics in the CBIO lab under the supervision of Laurent Gatto and [Matthieu Defrance](https://www.ulb.be/fr/matthieu-defrance-1). 


![CBIO lab photo](/images/CBIO-20240703.jpg)

From left to right: Samuel, Riya, Enes, Andrea, Guillaume, Philippe,
Axelle, Jérôme, Laurent, Loïc and Lieven (from the
[SMCS](https://uclouvain.be/en/technology-platforms/smcs.html)). Julie
is missing. Picture taken at the de Duve Institute on 3 July 2024.

<!-- ## Remote members -->

## [R for Mass Spectrometry](https://www.rformassspectrometry.org/) core team

- [**Sebastian Gibb**](http://sebastiangibb.de/), Department of
  Anesthesiology and Intensive Care, University Medicine Greifswald,
  Germany. Sebastian visited Laurent's group in Cambridge 2014 and
  since then contributes to various projects, in particular
  [`MSnbase`](http://lgatto.github.io/MSnbase),
  [`Pbase`](http://bioconductor.org/packages/Pbase) and
  [`synapter`](http://bioconductor.org/packages/synapter).

- [**Johannes
  Rainer**](http://www.eurac.edu/en/research/health/biomed/staff/Pages/staffdetails.aspx?persId=34084),
  Center for Biomedicine, EURAC, Bolzano, Italy is a contributor to
  the [`MSnbase`](http://lgatto.github.io/MSnbase) and
  [`Pbase`](http://bioconductor.org/packages/Pbase) packages.

## Key collaborators

<!-- - [**Paul -->
<!--   Kirk**](https://www.mrc-bsu.cam.ac.uk/people/in-alphabetical-order/h-to-m/paul-kirk/), -->
<!--   MRC Biostatistics Unit, University of Cambridge, UK. Paul is part of -->
<!--   the team developing principled Bayesian model for spatial -->
<!--   proteomics, such as the T-Augmented Gaussian Mixture (TAGM) model -->
<!--   ([Crook *et al.* -->
<!--   2018](https://doi.org/10.1371/journal.pcbi.1006516)). -->

<!-- - [**Kathyn S. Lilley**](http://www.bioc.cam.ac.uk/people/uto/lilley), -->
<!--   Cambridge Centre for Proteomics, Department of Biochemistry, -->
<!--   University of Cambridge, UK. Kathryn's lab developed the LOPIT -->
<!--   ([Dunkley *et al.* -->
<!--   2006](https://www.ncbi.nlm.nih.gov/pubmed/16618929)) and hyperLOPIT -->
<!--   ([Christoforou *et al.* -->
<!--   2016](https://www.ncbi.nlm.nih.gov/pubmed/26754106)) -->
<!--   technologies. Her lab is at the origin of some of the very best -->
<!--   spatial proteomics datasets in the world, which we help to analyse -->
<!--   and interpret by applying our dedicated software (see for example -->
<!--   [Breckels *et al.* 2016 in -->
<!--   F1000Research](https://f1000research.com/articles/5-2926/)). -->

<!-- - [**Sean Holden**](http://www.cl.cam.ac.uk/~sbh11/), Computer -->
<!--   Laboratory. We collaborate with Sean on the development and -->
<!--   application of machine learning algorithms to solve problems in -->
<!--   spatial proteomics. A recent example is transfer learning -->
<!--   ([Breckels *et al.* 2016](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004920)). -->

- **Manon Martin** (ORCID:
  [0000-0003-4800-0942](https://orcid.org/0000-0003-4800-0942),
  Github: [ManonMartin](https://github.com/ManonMartin)) is a research
  assistant in bioinformatics at the LIBST and ELI institues and the
  SMCS core facility on the Louvain-la-Neuve campus. She worked in the
  CBIO group from September 2020 to June 2022. She still collaborates
  with the lab as part of the bioinformatics core facility.

- [**Lieven Clement**](https://statomics.github.io/), statOmics group,
  UGent, Belgium. We collaborate on statistical analyses of
  quantitative proteomics data, in particular the
  [msqrob2](https://bioconductor.org/packages/release/bioc/html/msqrob2.html)
  R/Bioconductor package.

- [**Thomas
  Burger**](https://sites.google.com/site/thomasburgerswebpage/),
  BIG-BGE (Université Grenoble Alpes, CNRS, CEA, INSERM), France. We
  collaborate with Thomas and his group on various proteomics data
  analysis (for example missing data in [Lazar *et al*
  2016](https://www.ncbi.nlm.nih.gov/pubmed/26906401)) and software
  (see [Wieczorek *et al.*
  2016](https://www.ncbi.nlm.nih.gov/pubmed/27605098)). Thomas
  contributed the *perTurbo* algorithms to
  [*pRoloc*](https://bioconductor.org/packages/pRoloc).


- [**Christophe Vanderaa**](https://github.com/cvanderaa), statOmics group,
  UGent, Belgium. We pursue our collaboration on the development of
  computational solutions for the analysis of single-cell proteomics
  data.

## Alumni

- **Christophe Vanderaa** (ORCID:
  [0000-0001-7443-5427](https://orcid.org/0000-0001-7443-5427),
  Github: [cvanderaa](https://github.com/cvanderaa)) is a former PhD
  student working on single-cell assay bioinformatics. He did a
  masters in biomedical sciences at the UCLouvain and followed up with
  a masters in bioinformatics at the KULeuven before conducting his
  PhD training in the lab funded by an FNRS fellowship. Chris then
  enrolled as a postdoc at the UGent in the statOmics group (headed by
  Lieven Clement) to develop statistical frameworks for the analysis
  of single-cell proteomics data.

- **Badr-Ali Mouaden** finished his engineering degree in 2023 and did
  his master's project under the co-supervision of Laurent Gatto,
  Philippe Hauchamps and [Prof Michel
  Versleysen](https://mlg.info.ucl.ac.be/Members/MichelVerleysen) from
  the [Louvain Engineering School
  (EPL)](https://uclouvain.be/fr/facultes/epl). He worked on the
  development and application of deep convolution neural network on
  flow cytometry data.

- **Kristína Gömöryová**
  (Github:[KristinaGomoryova](https://github.com/KristinaGomoryova))
  joined the lab in September 2022 as an OpenMS intern. She is doing
  her PhD in Animal Physiology and Immunology in the lab of
  [prof. Bryja](https://www.sci.muni.cz/ofiz/en/bryja/) at Masaryk
  University. In CBIO lab she focuses on the analysis of DIA
  proteomics data within the HYGIEIA project.

- **Charlotte Léonard** (Github:
  [chachaleo](https://github.com/chachaleo)) did her Master's project
  in applied mathematics, working on single-cell multi-modal data
  integration. She is co-supervised by [Michel
  Verleysen](https://mlg.info.ucl.ac.be/Members/MichelVerleysen) from
  the Machine Learning Group at UCLouvain.

- **Manojkumar Selvaraju** (ORCID:
  [0000-0001-6439-5652](https://orcid.org/0000-0001-6439-5652),
  Github: [manogenome](https://github.com/manogenome)) worked from
  September 2020 to May 2022 on software infrastructure for the
  integration of genomics, transcriptomics and proteomics data.

- **Chong Tang** (ORCID:
  [0000-0002-0482-8361](https://orcid.org/0000-0002-0482-8361),
  Github: [plantton](https://github.com/plantton)) was part of the
  CBIO lab from the very beginning in 2019 to 2021. He worked on mass
  spectrometry-based proteomics software development with a special
  focus on using databases to store and retrieve massive MS datasets.

- **Aayush Grover** (ORCID:
  [0000-0002-3716-2850](https://orcid.org/0000-0002-3716-2850),
  Github: [AayushGrover](https://github.com/AayushGrover)) is a PhD
  student at the [Institute for Machine Learning of ETH Zurich](https://ml.inf.ethz.ch/)
  where he is working on computational epigenetics of cancer.
  He was a master thesis student at the CBIO lab after completing his summer
  internship (2020) at the CBIO lab. He worked on predicting protein subcellular
  location using deep learning techniques during his internship and focused on
  the task of survival prediction in cancer using multi-omics integration and
  few-shot learning.

- **Kilian Maes** (Github:
  [KilianMaes](https://github.com/KilianMaes)) did his Master's
  project in software engineering in the lab working on the clustering
  of mass spectrometry data. He was co-supervised by [Prof Pierre
  Dupont](https://www.info.ucl.ac.be/~pdupont/) from the Machine
  Learning Group. He now works part-time as a software engineer and
  continues his medical degree at the UCLouvain.

- [**Oliver Crook**](https://olivercrook.co.uk/), was a Wellcome Trust
  PhD student at the University of Cambridge, jointly supervised with
  [Paul
  Kirk](http://www.mrc-bsu.cam.ac.uk/people/in-alphabetical-order/h-to-m/paul-kirk/)
  from MRC Biostatistics Unit and [Kathryn
  Lilley](http://www.bioc.cam.ac.uk/people/uto/lilley). Olly worked on
  new Bayesian methods for spatial proteomics (see for example [Crook
  *et al.* 2018](https://doi.org/10.1371/journal.pcbi.1006516)),
  formalising the theoretical foundations and implementing the
  algorithms in the [`pRoloc`](https://lgatto.github.io/pRoloc/)
  package. He is now pursuing post-doctoral research at the [Protein
  Informatics Group](http://opig.stats.ox.ac.uk/) in the Department of
  Statistics at the University of Oxford.

- [**Lisa Breckels**](https://lmsimp.github.io/), Post-Doctoral
  Research Associa in the Cambridge Centre for Proteomics, at the
  University of Cambridge (UK). Her work focuses on the application
  and development of machine learning and pattern recognition methods
  to enable robust analysis of high-throughput quantitative proteomics
  data.  All her methods and algorithms form part of the
  [`pRoloc`](http://bioconductor.org/packages/pRoloc) framework for
  protein localisation prediction.

- [**Theo Killian**](https://github.com/tfkillian) was a
  bioinformatician and research programmer in the CBIO group at the de
  Duve Institute. Theo developes and maintains the
  [depmap](http://bioconductor.org/packages/release/data/experiment/html/depmap.html)
  package. He now works as a Bioinformatics scientist at the
  Laboratory of Translational Genetics at the KULeuven.

- [**Robin Kohze**](https://github.com/Kohze) comes from the Radboud
  University Nijmegen and visits the group in the frame of his
  master's internship. He developed a new infrastructure to facilitate
  the dissemination of spatial proteomics data. He currently pursues a
  PhD in the Department of Genetics at the University of Cambridge,
  UK.

- [**Thomas Naake**](https://tnaake.github.io/), visited the group in
  Cambridge in Spring 2014 as an ERASMUS student affiliated to the
  University of Freiburg. He developed the first version of
  [`pRolocGUI`](https://bioconductor.org/packages/release/bioc/html/pRolocGUI.html),
  an interactive visualisation tools for organelle proteomics
  data. Thomas has since earner a PhD from the Max Planck Institute of
  Molecular Plant Physiology.

- [**Victoria Carr**](https://orcid.org/0000-0002-3038-8929),
  was a Part III student in Systems Biology at the University of
  Cambridge, applying graph-based methods to study protein
  sub-cellular localisation. Vicky went on doing a PhD at King's
  College London and is now working as a Scientific Programmer in the
  Pathogen Informatics Unit at the Wellcome Sanger Institute.
