<!doctype html>
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin SEO -->









<title>            Becoming a better scientist with open and reproducible research      ~/      </title>




<meta name="description" content="This blog post summarises the notes for my talk at the Are you readyfor publishing reproducibleresearch?meeting at the TU Delft on the 16 May 2019. The slides are available here.">




<meta property="og:locale" content="en_GB">
<meta property="og:site_name" content="~/">
<meta property="og:title" content="Becoming a better scientist with open and reproducible research">


  <link rel="canonical" href="https://lgatto.github.io/rr-publ/">
  <meta property="og:url" content="https://lgatto.github.io/rr-publ/">



  <meta property="og:description" content="This blog post summarises the notes for my talk at the Are you readyfor publishing reproducibleresearch?meeting at the TU Delft on the 16 May 2019. The slides are available here.">





  

  





  <meta property="og:type" content="article">
  <meta property="article:published_time" content="2019-05-15T00:00:00+02:00">








  <script type="application/ld+json">
    {
      "@context" : "http://schema.org",
      "@type" : "Person",
      "name" : "Laurent Gatto",
      "url" : "https://lgatto.github.io",
      "sameAs" : null
    }
  </script>






<!-- end SEO -->


<link href="https://lgatto.github.io/feed.xml" type="application/atom+xml" rel="alternate" title="~/ Feed">

<!-- http://t.co/dKP3o1e -->
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="https://lgatto.github.io/assets/css/main.css">

<meta http-equiv="cleartype" content="on">
    <!-- start custom head snippets -->

<!-- insert favicons. use http://realfavicongenerator.net/ -->

<!-- end custom head snippets -->
  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->
    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <button><div class="navicon"></div></button>
        <ul class="visible-links">
          <li class="masthead__menu-item masthead__menu-item--lg"><a href="https://lgatto.github.io/">~/</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/about/">About</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/year-archive/">Posts</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/cbio-lab/">Lab</a></li>
          
        </ul>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>

    



<div id="main" role="main">
  
  <div class="sidebar sticky">
  

<div itemscope itemtype="http://schema.org/Person">

  
    <div class="author__avatar">
      
        <img src="https://lgatto.github.io/images/lgatto2.jpg" class="author__avatar" alt="Laurent Gatto" itemprop="image">
      
    </div>
  

  <div class="author__content">
    <h3 class="author__name" itemprop="name">Laurent Gatto</h3>
    
      <p class="author__bio" itemprop="description">
        Open science, reproducible research, data champion, computational biology, proteomics, more omics, emacs, a lot of R, quite a bit of running, and parenting.
      </p>
    
  </div>

  <div class="author__urls-wrapper">
    <button class="btn btn--inverse">Follow</button>
    <ul class="author__urls social-icons">
      

      

      

      
        <li>
          <a href="https://keybase.io/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-key" aria-hidden="true"></i> Keybase
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      
        <li>
          <a href="https://github.com/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      

      

      

      

      <!--
  <li>
    <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs">
      <i class="fa fa-fw" aria-hidden="true"></i> Custom Social Profile Link
    </a>
  </li>
-->


  <li>
    <a href="http://dissem.in/r/6231/laurent-gatto" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> dissem.in
    </a>
  </li>

  <li>
    <a href="https://orcid.org/0000-0002-1520-2268" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> ORCID
    </a>
  </li>

  <li>
    <a href="https://scholar.google.co.uk/citations?user=k5DrB74AAAAJ&hl=en" itemprop="sameAs">
      <i class="fa fa-fw fa-google" aria-hidden="true"></i> scholar
    </a>
  </li>

  <!-- <li> -->
  <!--   <a href="https://impactstory.org/u/0000-0002-1520-2268" itemprop="sameAs"> -->
  <!--     <i class="fa fa-fw fa-sun-o" aria-hidden="true"></i> Impactstory -->
  <!--   </a> -->
  <!-- </li> -->

    </ul>
  </div>
</div>

  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="Becoming a better scientist with open and reproducible research">
    <meta itemprop="description" content="This blog post summarises the notes for my talk at the Are you readyfor publishing reproducibleresearch?meeting at the TU Delft on the 16 May 2019. The slides are available here.">
    <meta itemprop="datePublished" content="May 15, 2019">
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 class="page__title" itemprop="headline">Becoming a better scientist with open and reproducible research
</h1>
          
            <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  17 minute read
</p>
          
        </header>
      

      <section class="page__content" itemprop="text">
        <p>This blog post summarises the notes for my talk at the <a href="https://www.eventbrite.com/e/are-you-ready-for-publishing-reproducible-research-tickets-55561449792">Are you ready
for publishing reproducible
research?</a>
meeting at the TU Delft on the 16 May 2019. The slides are available <a href="https://lgatto.github.io/2019_05_16_rr_Delft/">here</a>.</p>

<p>This material is available under CC-BY, unless otherwise stated.</p>

<p>The original title of the talk was <em>Being a better scientist with open
and reproducible research</em>, but I feel that this is some one
constantly aims for, hence the change to <em>becoming</em> one.</p>

<aside class="sidebar__right">
<nav class="toc">
    <header><h4 class="nav__title"><i class="fa fa-file-text"></i> On This Page</h4></header>
<ul class="toc__menu" id="markdown-toc">
  <li><a href="#open-and-reproducible-research" id="markdown-toc-open-and-reproducible-research">Open and reproducible research</a></li>
  <li><a href="#there-isnt-only-one-type-of-open-science" id="markdown-toc-there-isnt-only-one-type-of-open-science">There isn’t only one type of open science</a></li>
  <li><a href="#why-becoming-an-open-research-practitioners" id="markdown-toc-why-becoming-an-open-research-practitioners">Why becoming an open research practitioners</a></li>
  <li><a href="#why-reproducibility-is-important" id="markdown-toc-why-reproducibility-is-important">Why reproducibility is important</a></li>
  <li><a href="#but-what-do-we-mean-by-reproducibility" id="markdown-toc-but-what-do-we-mean-by-reproducibility">But what do we mean by reproducibility?</a></li>
  <li><a href="#why-becoming-a-reproducible-research-practitioners" id="markdown-toc-why-becoming-a-reproducible-research-practitioners">Why becoming a reproducible research practitioners</a></li>
  <li><a href="#what-can-you-do-to-improve-trust-in-your-research" id="markdown-toc-what-can-you-do-to-improve-trust-in-your-research">What can you do to improve trust in (your) research?</a>    <ul>
      <li><a href="#preprints-are-the-best" id="markdown-toc-preprints-are-the-best">Preprints are the best!</a></li>
      <li><a href="#promoting-open-research-through-peer-review" id="markdown-toc-promoting-open-research-through-peer-review">Promoting open research through peer review</a></li>
      <li><a href="#registered-reports" id="markdown-toc-registered-reports">Registered reports</a></li>
      <li><a href="#make-allies" id="markdown-toc-make-allies">Make allies</a></li>
      <li><a href="#just-do-it" id="markdown-toc-just-do-it">Just do it!</a></li>
    </ul>
  </li>
  <li><a href="#technical-solutions" id="markdown-toc-technical-solutions">Technical solutions</a></li>
  <li><a href="#some-examples-of-my-own-research" id="markdown-toc-some-examples-of-my-own-research">Some examples of my own research</a>    <ul>
      <li><a href="#software" id="markdown-toc-software">Software</a></li>
      <li><a href="#qsep-quantify-resolution-of-a-spatial-proteomics-experiment" id="markdown-toc-qsep-quantify-resolution-of-a-spatial-proteomics-experiment"><code class="language-plaintext highlighter-rouge">QSep</code>: quantify resolution of a spatial proteomics experiment</a></li>
      <li><a href="#spatialmap" id="markdown-toc-spatialmap">SpatialMap</a></li>
    </ul>
  </li>
  <li><a href="#inclusivity-open-research-and-open-research" id="markdown-toc-inclusivity-open-research-and-open-research">Inclusivity: open research and open research</a></li>
  <li><a href="#conclusions" id="markdown-toc-conclusions">Conclusions</a></li>
  <li><a href="#acknowledgements" id="markdown-toc-acknowledgements">Acknowledgements</a></li>
</ul>

  </nav>
</aside>

<p><strong>Disclaimer</strong>: I do not speak from authority<sup id="fnref:auth" role="doc-noteref"><a href="#fn:auth" class="footnote" rel="footnote">1</a></sup>. I speak of personal
experience. My experience is in computational biology, bioinformatics
and high throughput biology data. My experience doesn’t directly
translate to other fields or domains (for example when it comes to
data privacy) or even to other personalities in the same field.</p>

<p><strong>Note 1:</strong> A piece of open research doesn’t automatically make it good,
where good is defined as of high academic quality. A piece of closed
research doesn’t make it bad, where bad here is defined of low
academic quality. So openness doesn’t equate to academic quality. But
openness provides some desired quality (i.e. desirable property)
independent from academic excellent. Openness leads to trust.</p>

<p><strong>Note 2:</strong> A piece of reproducible research doesn’t automatically make
it good, where good is defined as of high academic quality. A piece of
non reproducible research doesn’t make it bad, where bad here is
defined of low academic quality. So reproducible doesn’t equate to
academic quality. But reproducibility provides some desired quality
(i.e. desirable property) independent from academic
excellent. Reproducibility leads (among other things) to trust.</p>

<h2 id="open-and-reproducible-research">Open and reproducible research</h2>

<blockquote>
  <p>Open != reproducible</p>
</blockquote>

<p>Open research and reproducible research aren’t the same thing, and one
doesn’t imply the other. They are historically also very different.</p>

<blockquote class="twitter-tweet" data-cards="hidden" data-lang="en"><p lang="en" dir="ltr">Many people seem to think that <a href="https://twitter.com/hashtag/OpenScience?src=hash&amp;ref_src=twsrc%5Etfw">#OpenScience</a> &amp; the reproducibility crisis in psychology are somehow causally related. They are not. Open science is decades old &amp; did not focus on reproducibility as a single issue — more here: <a href="https://t.co/KpJHIEqPj3">https://t.co/KpJHIEqPj3</a> &amp; here: <a href="https://t.co/KdMeK6PCUT">https://t.co/KdMeK6PCUT</a> <a href="https://t.co/qF5yPTqNqu">pic.twitter.com/qF5yPTqNqu</a></p>&mdash; Olivia Guest | Ολίβια Γκεστ (@o_guest) <a href="https://twitter.com/o_guest/status/1068791012481470464?ref_src=twsrc%5Etfw">December 1, 2018</a></blockquote>
<script async="" src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

<p>From a technical point of view:</p>

<ul>
  <li>When individual patronage funded scientists, discoveries were kept
private or publicised as codes in anagrams or cyphers (Source
<a href="https://en.wikipedia.org/wiki/Open_science#History">Wikipedia:Open
Science</a>).</li>
  <li>The concept of open access to scientific data was institutionally
established with the formation of the World Data Center system (now
the World Data System) in 1957. MEDLINE, later renamed PubMed, was
created in 1966. (Source <a href="https://en.wikipedia.org/wiki/Open_science_data#History">Wikipedia:Open Science
Data</a>).</li>
</ul>

<p>From a philosophical point of view:</p>

<p><strong><a href="https://en.wikipedia.org/wiki/Mertonian_norms">The Mertonian norms</a> (1942)</strong></p>

<ul>
  <li>
    <p><strong>Communism</strong>: all scientists should have common ownership of scientific
goods (intellectual property), to promote collective collaboration;
secrecy is the opposite of this norm.</p>
  </li>
  <li>
    <p><strong>Universalism</strong>: scientific validity is independent of the
sociopolitical status/personal attributes of its participants.</p>
  </li>
  <li>
    <p><strong>Disinterestedness</strong>: scientific institutions act for the benefit of a
common scientific enterprise, rather than for the personal gain of
individuals within them</p>
  </li>
  <li>
    <p><strong>Organised scepticism</strong>: scientific claims should be exposed to
critical scrutiny before being accepted: both in methodology and
institutional codes of conduct.</p>
  </li>
</ul>

<h2 id="there-isnt-only-one-type-of-open-science">There isn’t only one type of open science</h2>

<p>Open science has seen a continuous evolution since the 17th century,
with the advent of dissemination of research in scientific journals
and the societal demand to access scientific knowledge at
large. Technology and communication has further accelerated this
evolution, and put it in the spot light among researchers and
academics (for for examples funder mandates) and more widely in the
press with the cost of publications (see for example this Guardian
<em>long read</em> article <a href="https://www.theguardian.com/science/2017/jun/27/profitable-business-scientific-publishing-bad-for-science">Is the staggeringly profitable business of
scientific publishing bad for
science?</a>
or the <a href="https://paywallthemovie.com/">Paywall</a> movie).</p>

<p>Open science/research is the process of transparent dissemination and
access to knowledge, that can be applied to various scientific
practices (image below from Wikipedia):</p>

<p><img src="https://upload.wikimedia.org/wikipedia/commons/9/9c/Open_Science_-_Prinzipien.png" alt="The six principles of open science" /></p>

<p>As a result</p>

<blockquote>
  <p>Open science/research can mean different things to different people,
in particular when declined it along its many technical and
philosophical attributes.</p>
</blockquote>

<p>Take home message:</p>

<blockquote>
  <p>Open isn’t binary, it’s a gradient, it’s multidisciplinary, it’s multidimensional.</p>
</blockquote>

<p>How to be an open scientist:</p>

<blockquote>
  <p>Let’s be open and understanding of different situations and
constraints.</p>
</blockquote>

<h2 id="why-becoming-an-open-research-practitioners">Why becoming an open research practitioners</h2>

<p>It’s the right thing to do. See the <strong>The Mertonian norms</strong>… Or is it?</p>

<p><strong>Benefits for your academic career</strong>: some examples from the <a href="https://lgatto.github.io/EPFL-open-science/"><strong>Open
as a career boost</strong></a>
paragraph:</p>

<ul>
  <li>Open access articles get more citations.</li>
  <li>Data availability is associated with citation benefit.</li>
  <li>Openly available software more likely to be used. (I don’t have any
reference for this, and there are of course many couter examples).</li>
</ul>

<p><strong>Networking opportunities</strong> (I’m here thanks to my open research
activities with my former colleague Marta Teperek at the University of
Cambridge, UK).</p>

<p>See also <a href="http://whyopenresearch.org/index"><strong>Why Open Research</strong></a></p>

<ul>
  <li>
    <p><strong>Increase your visibility</strong>: Build a name for yourself. Share your
work and make it more visible.</p>
  </li>
  <li>
    <p><strong>Reduce publishing costs</strong>: Open publishing can cost the same or
less than traditional publishing.</p>
  </li>
  <li>
    <p><strong>Take back control</strong>: Know your rights. Keep your rights. Decide
how your work is used</p>
  </li>
  <li>
    <p><strong>Publish where you want</strong>: Publish in the journal of your choice
and archive an open copy. (See <a href="http://thecostofknowledge.com/">The cost of
knowledge</a> boycott of Elsevier).</p>
  </li>
  <li>
    <p><strong>Get more funding</strong>: Meet funder requirements, and qualify for
special funds such as the Wellcome Trust <a href="https://wellcome.ac.uk/funding/schemes/open-research-fund">Open Research
Fund</a>.</p>
  </li>
  <li>
    <p><strong>Get that promotion</strong>: Open research is increasingly recognised in
promotion and tenure. See also <a href="https://cos.io/blog/are-reproducibility-and-open-science-starting-matter-tenure-and-promotion-review/">Reproducibility and open science are
starting to matter in tenure and
promotion</a>
July 14th, 2017, Brian Nosek) and the EU’s <a href="https://cdn1.euraxess.org/sites/default/files/policy_library/os-rewards-wgreport-final_integrated_0.pdf">Evaluation of Research
Careers fully acknowledging Open Science
Practice</a>
defines an Open Science Career Assessment Matrix (OS-CAM):</p>
  </li>
</ul>

<p>But are there <strong>any risks</strong>?</p>

<blockquote>
  <p>Does it take more time to work openly?</p>
</blockquote>

<p>Isn’t it worth investing time is managing data in a way that others
(including future self) can find and understand it? That’s, IMHO,
particularly important from a group leader’s perspective, where I want
to build a corpus of data/software/research that other lab members can
find, mine and re-use.</p>

<blockquote>
  <p>Are senior academics always supportive?</p>
</blockquote>

<p>No.</p>

<blockquote>
  <p>Is there a risk of being scooped?</p>
</blockquote>

<p>There certainly is a benefit if releasing one’s research early!</p>

<p>But, importantly, working with open and reproducible research in mind
<strong>doesn’t mean releasing everything prematurely</strong>, it means</p>

<ul>
  <li>
    <p>managing research in a way one can find data and results at every
stage</p>
  </li>
  <li>
    <p>one can <strong>reproduce/repeat</strong> results, re-run/compare them with new
data or different methods/parameters, and</p>
  </li>
  <li>
    <p>one can <strong>release</strong> data (or parts thereof) when/if <strong>appropriate</strong>.</p>
  </li>
</ul>

<p>So, are there any risks?</p>

<blockquote>
  <p>The <a href="http://bulliedintobadscience.org/">Bullied Into Bad Science</a>
campaign is an initiative by early career researchers (ECRs) for
early career researchers who aim for a fairer, more open and ethical
research and publication environment.</p>
</blockquote>

<p><img src="https://raw.githubusercontent.com/BulliedIntoBadScience/stickers/master/logo_light.png" width="300" alt="Bullied Into Bad Science" /></p>

<h2 id="why-reproducibility-is-important">Why reproducibility is important</h2>

<ul>
  <li>
    <p>For <strong>scientific reasons</strong>: think reproducibility crisis.</p>
  </li>
  <li>
    <p>For <strong>political reasons</strong>: public trust in science, in data, in
experts; without (public) trust in science and research, there won’t
be any funding anymore.</p>
  </li>
</ul>

<h2 id="but-what-do-we-mean-by-reproducibility">But what do we mean by reproducibility?</h2>

<p>From a <a href="https://lgatto.github.io/rr-what-should-be-our-goals/"><em>But what to we mean by
reproducibility?</em></a>
blog post.</p>

<ul>
  <li>
    <p><strong>Repeat</strong> my experiment, i.e. obtain the same tables/graphs/results
using the same setup (data, software, …) in the same lab or on the
same computer. That’s basically re-running one of my analysis some
time after I original developed it.</p>
  </li>
  <li>
    <p><strong>Reproduce</strong> an experiment (not mine), i.e. obtain the same
tables/graphs/results in a different lab or on a different computer,
using the same setup (the data would be downloaded from a public
repository and the same software, but possibly different version,
different OS, is used). I suppose, we should differentiate
replication using a fresh install and a virtual machine or docker
image that replicates the original setup.</p>
  </li>
  <li>
    <p><strong>Replicate</strong> an experiment, i.e. obtain the same (similar enough)
tables/graphs/results in a different set up. The data could still be
downloaded from the public repository, or possibly
re-generate/re-simulate it, and the analysis would be re-implemented
based on the original description. This requires openness, and one
would clearly not be allowed the use a black box approach (VM,
docker image) or just re-running a script.</p>
  </li>
  <li>
    <p>Finally, <strong>re-use</strong> the information/knowledge from one experiment to
run a different experiment with the aim to confirm results from
scratch.</p>
  </li>
</ul>

<p>Another view (from a talk by <a href="https://figshare.com/articles/Publishing_a_reproducible_paper/4720996/1">Kirstie
Whitaker</a>):</p>

<table>
  <thead>
    <tr>
      <th> </th>
      <th>Same Data</th>
      <th>Different Data</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><strong>Same Code</strong></td>
      <td>reproduce</td>
      <td>replicate</td>
    </tr>
    <tr>
      <td><strong>Different Code</strong></td>
      <td>robust</td>
      <td>generalisable</td>
    </tr>
  </tbody>
</table>

<p>See also this opinion piece by Jeffrey T. Leek and Roger D. Peng,
<a href="https://www.pnas.org/content/112/6/1645"><em>Reproducible research can still be wrong: Adopting a prevention
approach</em></a>.</p>

<p>From</p>

<blockquote>
  <p>Gabriel Becker <a href="https://gmbecker.github.io/MayInstituteKeynote2019/outline.html"><em>An Imperfect Guide to Imperfect
Reproducibility</em></a>,
May Institute for Computational Proteomics, 2019.</p>
</blockquote>

<p><strong>(Computational) Reproducibility Is Not The Point</strong></p>

<p>Take home message:</p>

<blockquote>
  <p>The goal is <strong>trust</strong>, <strong>verification</strong> and <strong>guarantees</strong>:</p>
</blockquote>

<ul>
  <li><strong>Trust in Reporting</strong> - result is accurately reported</li>
  <li><strong>Trust in Implementation</strong> - analysis code successfully implements
chosen methods</li>
  <li><strong>Statistical Trust</strong> - data and methods are (still) appropriate</li>
  <li><strong>Scientific Trust</strong> - result convincingly supports claim(s) about
underlying systems or truths</li>
</ul>

<p>Reproducibility As A Trust Scale (copyright Genentech Inc)</p>

<p><img src="https://gmbecker.github.io/MayInstituteKeynote2019/trustscale3.png" alt="Reproducibility As A Trust Scale" /></p>

<p>Take home message:</p>

<blockquote>
  <p>Reproducibility isn’t binary, it’s a gradient, it’s multidisciplinary, it’s multidimensional.</p>
</blockquote>

<p>Another take home message:</p>

<blockquote>
  <p>Reproducibility isn’t easy.</p>
</blockquote>

<h2 id="why-becoming-a-reproducible-research-practitioners">Why becoming a reproducible research practitioners</h2>

<p>Florian Markowetz, <a href="https://doi.org/10.1186/s13059-015-0850-7"><strong>Five selfish reasons to work
reproducibly</strong></a>, Genome
Biology 2015, 16:274.</p>

<blockquote>
  <p>And so, my fellow scientists: ask not what you can do for
reproducibility; ask what reproducibility can do for you! Here, I
present five reasons why working reproducibly pays off in the long
run and is in the self-interest of every ambitious, career-oriented
scientist.</p>
</blockquote>

<ul>
  <li>Reason number 1: <strong>reproducibility helps to avoid disaster</strong></li>
  <li>Reason number 2: <strong>reproducibility makes it easier to write papers</strong></li>
  <li>Reason number 3: <strong>reproducibility helps reviewers see it your way</strong></li>
  <li>Reason number 4: <strong>reproducibility enables continuity of your work</strong></li>
  <li>Reason number 5: <strong>reproducibility helps to build your reputation</strong></li>
</ul>

<p>And career perspectives: <a href="https://www.nature.com/news/faculty-promotion-must-assess-reproducibility-1.22596">Faculty promotion must assess
reproducibility</a>.</p>

<h2 id="what-can-you-do-to-improve-trust-in-your-research">What can you do to improve trust in (your) research?</h2>

<ol>
  <li><strong>Be an open research practitioners</strong></li>
  <li><strong>Be an reproducible research practitioners</strong></li>
</ol>

<p>Includes (but not limited to)</p>

<h3 id="preprints-are-the-best">Preprints are the best!</h3>

<p>Read, post, review and cite <strong>preprints</strong> (see
<a href="https://asapbio.org/">ASAPbio</a> for lots of resources about
preprints).</p>

<h3 id="promoting-open-research-through-peer-review">Promoting open research through peer review</h3>

<p>This section is based on my <a href="https://rawgit.com/lgatto/2017-03-30-OSC-peerreview/master/slides.html">The role of peer-reviewers in <del>checking
supporting information</del> promoting open
science</a>
talk.</p>

<p>As an open researcher, I think it is important to apply and promote
the importance of data and good data management on a day-to-day basis
(see for example Marta Teperek’s 2017 <a href="http://doi.org/10.5281/zenodo.897785">Data Management: Why would I
bother?</a> slides), but also to
express this ethic in our academic capacity, such as peer review. My
responsibility as a reviewer is to</p>

<ul>
  <li>Accept sound/valid research and provide constructive comments</li>
</ul>

<p>and hence</p>

<ul>
  <li>Focus firstly on the validity of the research by inspecting the
data, software and method. If the methods and/or data fail, the rest
is meaningless.</li>
</ul>

<p>I don’t see novelty, relevance, news-worthiness as my business as a
reviewer. These factors are not the prime qualities of thorough
research, but rather characteristics of flashy news.</p>

<p>Here are some aspects that are easy enough to check, and go a long way
to verify the availability and validity and of the data</p>

<ol>
  <li>
    <p><strong>Availability</strong>: Are the data/software/methods accessible and
understandable in a way that would allow an informed researcher in
the same or close field to reproduce and/or verify the results
underlying the claims? Note that this doesn’t mean that as a
reviewer, I will necessarily try to repeat the whole analysis (that
would be too time consuming indeed). But, conversely, a submission
without data/software will be reviewed (and rejected, or more
appropriately send back for completion) in matters of minutes. Are
the data available in a public repository that guarantees that it
will remain accessible, such as a subject-specific or, if none is
available, a generic repository (such as
<a href="https://zenodo.org/">zenodo</a> or <a href="https://figshare.com/">figshare</a>,
…), an institutional repository, or, but less desirable,
supplementary information or a personal webpage<sup id="fnref:4" role="doc-noteref"><a href="#fn:4" class="footnote" rel="footnote">2</a></sup>.</p>
  </li>
  <li>
    <p><strong>Meta-data</strong>: It’s of course not enough to provide a wild dump of
the data/software/…, but these need to be appropriately
documented. Personally, I recommend an <code class="language-plaintext highlighter-rouge">README</code> file in every top
project directory to summarise the project, the data, …</p>
  </li>
  <li>
    <p><strong>Do numbers match?</strong>: The first thing when reproducing someone’s
analysis is to match the data files to the experimental
design. That is one of the first things I check when reviewing a
paper. For example if the experimental design says there are 2
groups, each with 3 replicates, I expect to find 6 (or a multiple
thereof) data files or data columns in the data matrix. Along these
lines, I also look at the file names (of column names in the data
matrix) for a consistent naming convention, that allows to match
the files (columns) to the experimental groups and replicates.</p>
  </li>
  <li>
    <p><strong>What data, what format</strong>: Is the data readable with standard and
open/free software? Are the raw and processed available, and have
the authors described how to get from one to the other?</p>
  </li>
  <li>
    <p><strong>License</strong>: Is the data shared in a way that allows users to
re-use it. Under what conditions? Is the research output shared
under a valid license?</p>
  </li>
</ol>

<p>Make sure that the data adhere to the
<strong><a href="http://www.nature.com/articles/sdata201618">FAIR</a> principles</strong>:</p>

<blockquote>
  <p>Findable and Accessible and Interoperable and Reusable</p>
</blockquote>

<p>Note that SI are not FAIR, not discoverable, not structured,
voluntary, used to bury stuff. A personal web page is likely to
disappear in the near future.</p>

<p>As a quick note, my <em>ideal</em> review system would be one where</p>

<ol>
  <li>
    <p>Submit your data to a repository, where it gets checked (by
specialists, data scientists, data curators) for quality,
annotation, meta-data.</p>
  </li>
  <li>
    <p>Submit your research with a link to the peer reviewed data. First
review the intro and methods, then only the results (to avoid
positive results bias).</p>
  </li>
</ol>

<p>When talking about open research and peer review, one logical
extension is <strong>open peer review</strong>.</p>

<p>While I personally value open peer review and practice it when
possible, it can be a difficult issue for ECRs, exposing them
unnecessarily when reviewing work from prominent peers. It also can
reinforce an already unwelcoming environment for underrepresented
minorities. See more about this in the <em>Inclusivity: open science and
open science</em> section below.</p>

<h3 id="registered-reports">Registered reports</h3>

<p>Define you data collection and analysis protocol in advance. Get it
reviewed and, if accepted, get right to publish once data have been
collected and analysed, irrespective of the (positive or negative)
result.</p>

<ul>
  <li>
    <p><strong>Three challenges</strong>: Restrictions on flexibility (no p-hacking of
HARKing), The time cost, Incentive structure isn’t in place yet.</p>
  </li>
  <li>
    <p><strong>Three benefits</strong>: Greater faith in research (no p-hacking of
HARKing), New helpful systems (see technical solutions below),
Investment in your future.</p>
  </li>
</ul>

<p>See
<a href="Open
science challenges, benefits and tips in early career and beyond">https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000246</a>
(2019).</p>

<h3 id="make-allies">Make allies</h3>

<blockquote>
  <p>This is <strong>very</strong> important!</p>
</blockquote>

<ul>
  <li>Other ECR</li>
  <li>Librarians</li>
  <li>Data stewards/champions</li>
  <li>Research Software engineers</li>
  <li>On/off-line networking</li>
</ul>

<p>Open research <strong>can</strong> lead to <strong>collaborative research</strong>. The
development of <a href="https://github.com/lgatto/MSnbase/"><code class="language-plaintext highlighter-rouge">MSnbase</code></a> is <a href="https://lgatto.github.io/msnbase-contribs/">an
example</a> I am very proud
of.</p>

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">The credit goes to the outstanding contributions and contributors!</p>&mdash; Laurent Gⓐtt⓪ (@lgatt0) <a href="https://twitter.com/lgatt0/status/1125497076241043456?ref_src=twsrc%5Etfw">May 6, 2019</a></blockquote>
<script async="" src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

<p>Collaborative work and cooperation is certainly one important concept
that gravitates around open science/research (see the Mertonian norm
of <em>communism</em>), but that isn’t necessary nor sufficient for open
science.</p>

<h3 id="just-do-it">Just do it!</h3>

<blockquote>
  <p>Build openness at the core your research</p>
</blockquote>

<p>(according to you possibilities)</p>

<p>Open and reproducible research doesn’t work if it’s an afterthought.</p>

<h2 id="technical-solutions">Technical solutions</h2>

<ul>
  <li>
    <p>Scripting, scripting, scripting (applies to code, data, analyses,
manuscripts, …).</p>
  </li>
  <li>
    <p>Avoid manual steps.</p>
  </li>
  <li>
    <p>Document everything, especially manual steps (which you should avoid anyway).</p>
  </li>
  <li>
    <p>Version control, such as git/<a href="https://lgatto.github.io/github-intro/">github</a>, bitbucket, …</p>
  </li>
  <li>
    <p>Literate analyses: reproducible documents with R markdown, Sweave (R
with LaTeX), Juyter notebooks, …</p>
  </li>
  <li>
    <p>Shareable compute environments (docker containers).</p>
  </li>
  <li>
    <p>Document and <strong>share</strong> all artefacts related to your research (when
possible): data, code, <a href="https://www.protocols.io/">protocols</a>, …</p>
  </li>
</ul>

<p>See also <a href="https://gmbecker.github.io/MayInstituteKeynote2019/outline.html"><em>An Imperfect Guide to Imperfect
Reproducibility</em></a>
for further details.</p>

<h2 id="some-examples-of-my-own-research">Some examples of my own research</h2>

<p><strong>Spatial proteomics software</strong>: systematic and high throughput
analysis of sub-cellular protein localisation.</p>

<h3 id="software">Software</h3>

<ul>
  <li>
    <p>Software: infrastructure with
  <a href="http://bioconductor.org/packages/MSnbase"><code class="language-plaintext highlighter-rouge">MSnbase</code></a> (Gatto and
  Lilley 2012); dedicated machine learning with
  <a href="http://bioconductor.org/packages/pRoloc"><code class="language-plaintext highlighter-rouge">pRoloc</code></a> (Gatto <em>et
  al.</em> 2014); interactive with
  <a href="http://bioconductor.org/packages/pRolocGUI"><code class="language-plaintext highlighter-rouge">pRolocGUI</code></a>; data
  with <a href="http://bioconductor.org/packages/pRolocdata"><code class="language-plaintext highlighter-rouge">pRolocdata</code></a>
  (Gatto <em>et al.</em> 2014).</p>
  </li>
  <li>
    <p>The <a href="http://bioconductor.org/">Bioconductor</a> project (Huber <em>et al.</em>
  2015) ecosystem for high throughput biology data analysis and
  comprehension: <strong>open source</strong>, and <strong>coordinated and
  collaborative</strong> (between and within domains/software) <strong>open
  development</strong>, enabling <strong>reproducible research</strong>, enables
  understanding of the data (not a black box) and <strong>drive scientific
  innovation</strong>.</p>
  </li>
</ul>

<p><img src="/images/rr-delft-pRoloc.png" alt="pRoloc screenshots" /></p>

<h3 id="qsep-quantify-resolution-of-a-spatial-proteomics-experiment"><code class="language-plaintext highlighter-rouge">QSep</code>: quantify resolution of a spatial proteomics experiment</h3>

<p><code class="language-plaintext highlighter-rouge">QSep</code> is a function within the <code class="language-plaintext highlighter-rouge">pRoloc</code> software.</p>

<p><img src="/images/rr-delft-qsep.png" alt="qsep screenshots" /></p>

<p>Evolution of Gatto <em>et al.</em> (2018)</p>

<ul>
  <li>left: <a href="https://github.com/lgatto/QSep-manuscript">reproducible document</a></li>
  <li>centre: <a href="https://doi.org/10.1101/377630">preprint</a></li>
  <li>right: <a href="https://doi.org/10.1016/j.cbpa.2018.11.015">peer reviewer paper</a></li>
</ul>

<h3 id="spatialmap">SpatialMap</h3>

<p><code class="language-plaintext highlighter-rouge">SpatialMap</code> is a project aiming at producing a visualisation and data
sharing platform for spatial proteomics. I decided to promote and
drive it as openly as possible in the frame of the <a href="http://osc.cam.ac.uk/open-research/open-research-pilot-project">Open Research
Pilot
Project</a>. The
ORPP is a joint project by the <a href="http://osc.cam.ac.uk/">Office of Scholarly
Communication</a> at the University of Cambridge
and the <a href="https://wellcome.ac.uk/what-we-do/our-work/open-research">Wellcome Trust Open research
team</a>. Here
are the reasons why the <em>SpatialMap</em> project is an open project:</p>

<ol>
  <li>
    <p>The <em>SpatialMap</em> project in itself is about opening up spatial
proteomics data by facilitating data sharing and providing tools to
further the comprehension of the data. One aim is to allow users to
use the <em>SpatialMap</em> web portal to upload, share, explore and
discuss their data privately with collaborators in a first instance
(few researchers share their data before publication), then make
the data available to reviewer, and finally, once reviewed, make it
public at the push of a button. The incentive for early utilisation
of the platform is to provide interactive data visualisation and
integration with other tools and sharing of the data with close
collaborators.</p>
  </li>
  <li>
    <p>The project is developed completely openly in a public <a href="https://github.com/SpatialMap/SpatialMapDev">GitHub
repository</a>. Absolutely
all code and contributions are publicly available. Anyone can
collaborate, or even <em>fork</em> the project and build their own.</p>
  </li>
  <li>
    <p>I publicly announced the <em>SpatialMap</em> project in a <a href="https://lgatto.github.io/spatialmap-proposal/">blog
post</a>. The blog post
was written as a legitimate grant application (albeit a little bit
shorter and sticking to the most important parts).</p>
  </li>
</ol>

<p>Note that I do not have any dedicated funding for this project. The
progress so far was the result of a masters student visiting my
group, and is currently not actively developed anymore.</p>

<h2 id="inclusivity-open-research-and-open-research">Inclusivity: <a href="https://lgatto.github.io/open-and-open/">open research and open research</a></h2>

<p>There is</p>

<blockquote>
  <p>Open Science as in widely disseminated and openly accessible</p>
</blockquote>

<p>and</p>

<blockquote>
  <p>Open Science as in inclusive and welcoming</p>
</blockquote>

<p>On being inclusive -
<a href="https://twitter.com/CameronNeylon/status/895546085468495873">Twitter thread</a> by
<a href="https://twitter.com/CameronNeylon">Cameron Neylon</a>:</p>

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">The primary value proposition of <a href="https://twitter.com/hashtag/openscience?src=hash">#openscience</a>
is that diverse contributions allow better critique, refinement, and
application 3/n</p>&mdash; CⓐmeronNeylon (@CameronNeylon) <a href="https://twitter.com/CameronNeylon/status/895546764861853696">August
10, 2017</a></blockquote>
<script async="" src="//platform.twitter.com/widgets.js" charset="utf-8"></script>

<blockquote class="twitter-tweet" data-conversation="none" data-lang="en"><p lang="en" dir="ltr">It was a damned hard community
to break into. Any step I took to be more open, I felt attacked for
not doing enough/doing it right.</p>&mdash; Christie Bahlai (@cbahlai)
<a href="https://twitter.com/cbahlai/status/871413258107981824">June
4, 2017</a></blockquote>
<script async="" src="//platform.twitter.com/widgets.js" charset="utf-8"></script>

<p>As far as I was concerned for a long time (until June 2017 to be
accurate - this section is based this <a href="https://lgatto.github.io/open-and-open/">Open science and open
science</a> post), the former
more technical definition was always what I was focusing on, and the
second community-level aspect of openness was, somehow, implicit from
the former, but that’s clearly not the case.</p>

<p>Even if there are efforts to promote diversity, under-represented
minorities (URM) don’t necessarily feel
<a href="http://science.sciencemag.org/content/357/6356/1101.full">included</a>.
When it comes to open science/research URMs can be further
discriminated against by greater exposure or, can’t always afford to
be vocal.</p>

<ul>
  <li>Not everybody has the privilege to be open.</li>
  <li>There are different levels in how open one wants, or how open one
could afford to be.</li>
  <li>Every voice and support is welcome.</li>
</ul>

<h2 id="conclusions">Conclusions</h2>

<blockquote>
  <p><strong>Standing on the shoulders of giants</strong> only really makes sense in
the context of open and reproducible research.</p>
</blockquote>

<ul>
  <li>
    <p>If you are here (or have read this), chances are you are on the path
towards open and reproducible research.</p>
  </li>
  <li>
    <p>You are the architect of the kind of research and researcher you
want to become. I hope these include openness and reproducibility.</p>
  </li>
  <li>
    <p>It’s a long path, that constantly evolves, depending on constraints,
aspirations, environment, …</p>
  </li>
  <li>
    <p>The sky is the limit, be creative: work out the (open and
reproducible) research that works for you now …</p>
  </li>
  <li>
    <p>… and that you want to work (for you and others) in the future.</p>
  </li>
</ul>

<h2 id="acknowledgements">Acknowledgements</h2>

<p>One of my advice was to <strong>make allies</strong>. I have been lucky to meet
wonderful allies and inspiring friends along the path towards open and
reproducible research that works for me. Among these, I would like to
highlight <a href="https://twitter.com/LoganCorina">Corina Logan</a>, <a href="https://twitter.com/StephenEglen">Stephen
Eglen</a>, <a href="https://twitter.com/martateperek">Marta
Teperek</a>, <a href="https://twitter.com/kirstie_j">Kirstie
Whitaker</a>, <a href="https://twitter.com/chartgerink">Chris
Hartgenink</a>, <a href="https://twitter.com/npscience">Naomie
Penfold</a>, <a href="https://twitter.com/yvonnenobis">Yvonne
Nobis</a>.</p>
<div class="footnotes" role="doc-endnotes">
  <ol>
    <li id="fn:auth" role="doc-endnote">
      <p>I actually think that authority (or seniority) isn’t doing
any favours when it comes to open research and
reproducibility. The more senior stakeholder all too often aren’t
those that drive research toward more openness and
reproducibility. There are, fortunately, notable exceptions. <a href="#fnref:auth" class="reversefootnote" role="doc-backlink">&#8617;</a></p>
    </li>
    <li id="fn:4" role="doc-endnote">
      <p>There is often no perfect solution, and a combination of the above might be desirable. <a href="#fnref:4" class="reversefootnote" role="doc-backlink">&#8617;</a></p>
    </li>
  </ol>
</div>

        
      </section>

      <footer class="page__meta">
        
        
  


  
  
  

  <p class="page__taxonomy">
    <strong><i class="fa fa-fw fa-tags" aria-hidden="true"></i> Tags: </strong>
    <span itemprop="keywords">
    
      
      
      <a href="https://lgatto.github.io/tags/#ecr" class="page__taxonomy-item" rel="tag">ECR</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#open-research" class="page__taxonomy-item" rel="tag">open research</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#reproducible-research" class="page__taxonomy-item" rel="tag">reproducible research</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#talks" class="page__taxonomy-item" rel="tag">talks</a>
    
    </span>
  </p>




        
          <p class="page__date"><strong><i class="fa fa-fw fa-calendar" aria-hidden="true"></i> Updated:</strong> <time datetime="2019-05-15T00:00:00+02:00">May 15, 2019</time></p>
        
      </footer>

      

      
  <nav class="pagination">
    
      <a href="https://lgatto.github.io/interview-questions/" class="pagination--pager" title="Notes on interview questions
">Previous</a>
    
    
      <a href="https://lgatto.github.io/remote-teaching/" class="pagination--pager" title="Notes on remote/online teaching
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
    <div class="page__related">
      
        <h4 class="page__related-title">You May Also Enjoy</h4>
      
      <div class="grid__wrapper">
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/ia-irss/" rel="permalink">On cats, farts and parrots.
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  11 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">I was invited to contribute to a
seminar/discussion
on AI, and using language models in research. These are the notes I
prepared my short presentation. I am ...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/oublions-les-notes/" rel="permalink">Podcasts du LLL: Et si on oubliait les notes?
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  less than 1 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">Deuxième billet en français référençant un podcast du Louvain
Learning Lab (le
premier est ici). Cette
fois, je fais un peu d’auto-promotion:

</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/ecr-recognition/" rel="permalink">HUPO ECR Online Panel Discussion - Getting recognised for your work
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  8 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">The HUPO Early Career Researcher (ECR) committee has organised a
discussion panel on Getting recognised for your work and have asked
me to participate - than...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/the-grid-in-r/" rel="permalink">The Grid poster, in R
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  3 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">The MuseeL is the
UCLouvain University museum in
Louvain-la-Neuve. Highly recommended. It’s located to the lively
place des sciences, in a nice brutalist sty...</p>
  </article>
</div>
        
      </div>
    </div>
  
</div>

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
      <li><strong>Follow:</strong></li>
    
    
    
    
      <li><a href="http://github.com/lgatto"><i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li>
    
    
    <li><a href="https://lgatto.github.io/feed.xml"><i class="fa fa-fw fa-rss-square" aria-hidden="true"></i> Feed</a></li>
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2024 Laurent Gatto CC-BY. Powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    <script src="https://lgatto.github.io/assets/js/main.min.js"></script>





  </body>
</html>
