<!doctype html>
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin SEO -->









<title>            Lab members      ~/      </title>




<meta name="description" content="The team">




<meta property="og:locale" content="en_GB">
<meta property="og:site_name" content="~/">
<meta property="og:title" content="Lab members">


  <link rel="canonical" href="https://lgatto.github.io/cbio-who/">
  <meta property="og:url" content="https://lgatto.github.io/cbio-who/">



  <meta property="og:description" content="The team">





  

  












  <script type="application/ld+json">
    {
      "@context" : "http://schema.org",
      "@type" : "Person",
      "name" : "Laurent Gatto",
      "url" : "https://lgatto.github.io",
      "sameAs" : null
    }
  </script>






<!-- end SEO -->


<link href="https://lgatto.github.io/feed.xml" type="application/atom+xml" rel="alternate" title="~/ Feed">

<!-- http://t.co/dKP3o1e -->
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="https://lgatto.github.io/assets/css/main.css">

<meta http-equiv="cleartype" content="on">
    <!-- start custom head snippets -->

<!-- insert favicons. use http://realfavicongenerator.net/ -->

<!-- end custom head snippets -->
  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->
    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <button><div class="navicon"></div></button>
        <ul class="visible-links">
          <li class="masthead__menu-item masthead__menu-item--lg"><a href="https://lgatto.github.io/">~/</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/about/">About</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/year-archive/">Posts</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/cbio-lab/">Lab</a></li>
          
        </ul>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>

    



<div id="main" role="main">
  
  <div class="sidebar sticky">
  
  
    
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          
          

          <a href="https://lgatto.github.io/cbio-lab/"><span class="nav__sub-title">The lab</span></a>
        

        
      </li>
    
      <li>
        
          
          

          <a href="https://lgatto.github.io/cbio-who/"><span class="nav__sub-title">Who</span></a>
        

        
      </li>
    
      <li>
        
          
          

          <a href="https://lgatto.github.io/cbio-coc/"><span class="nav__sub-title">Code of conduct</span></a>
        

        
      </li>
    
  </ul>
</nav>
    
  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="Lab members">
    <meta itemprop="description" content="The team">
    
    <meta itemprop="dateModified" content="January 14, 2022">

    <div class="page__inner-wrap">
      
        <header>
          <h1 class="page__title" itemprop="headline">Lab members
</h1>
          
        </header>
      

      <section class="page__content" itemprop="text">
        <aside class="sidebar__right">
<nav class="toc">
    <header><h4 class="nav__title"><i class="fa fa-file-text"></i> On This Page</h4></header>
<ul class="toc__menu" id="markdown-toc">
  <li><a href="#cbio-core" id="markdown-toc-cbio-core">CBIO core</a></li>
  <li><a href="#undergraduates" id="markdown-toc-undergraduates">Undergraduates</a></li>
  <li><a href="#r-for-mass-spectrometry-core-team" id="markdown-toc-r-for-mass-spectrometry-core-team">R for Mass Spectrometry core team</a></li>
  <li><a href="#key-collaborators" id="markdown-toc-key-collaborators">Key collaborators</a></li>
  <li><a href="#alumni" id="markdown-toc-alumni">Alumni</a></li>
</ul>

  </nav>
</aside>

<h2 id="cbio-core">CBIO core</h2>

<ul>
  <li>
    <p><a href="http://lgatto.github.io/about/"><strong>Laurent Gatto</strong></a> (ORCID:
<a href="https://orcid.org/0000-0002-1520-2268">0000-0002-1520-2268</a>,
Github: <a href="https://github.com/lgatto/">lgatto</a>, fosstodon:
<a href="https://fosstodon.org/@lgatto">@lgatto</a>), head of the unit. He runs
the CBIO group and teaches bioinformatics in the Facutly of Pharmacy
and Biomedical Sciences (FASB).</p>
  </li>
  <li>
    <p><strong>Axelle Loriot</strong> (ORCID:
<a href="https://orcid.org/0000-0002-5288-8561">0000-0002-5288-8561</a>,
Github: <a href="https://github.com/aloriot">aloriot</a>) is a senior
researcher at the de Duve Institute. Since 2001, she has been part
of the epigenetics group led by Charles De Smet. She joined the CBIO
group in 2018, focusing on transcriptomics and epigenetics. She
developed algorithms to identify aberrant transcripts induced by DNA
hypomethylation in tumor cells, and DNA hypermethylation events
linked to overlapping expression. Axelle plays a major role in the
bioinformatics core facility.</p>
  </li>
  <li>
    <p><strong>Philippe Hauchamps</strong> (Github:
<a href="https://github.com/phauchamps">phauchamps</a>) is, since September
2021, a PhD student working on flow cytometry data analyis. He
joined the CBIO group in October 2019 as an intern and carried on
with a Master’s project in computational proteomics, developing
Bayesian models to analyse quantitative proteomics data. He has a
Master degree in Computer Science Engineering and one in
Biostatistics.</p>
  </li>
  <li>
    <p><strong>Julie Devis</strong> (Github:
<a href="https://github.com/JulieDevis">JulieDevis</a>) did her master’s
project in the CBIO lab and is now, since September 2022, a PhD
student in biomedical sciences. She is working on deciphering
complex mechanisms of epigenetic alterations in cancer genomes. She
is pursuing her research in collaboration with Axelle Loriot and
<a href="https://www.deduveinstitute.be/cancer-epigenetics">Prof. Charles De
Smet</a> from the
epigenetics group.</p>
  </li>
  <li>
    <p><strong>Samuel Grégoire</strong> (Github:
<a href="https://github.com/samgregoire">samgregoire</a>) did his master’s
project in the CBIO lab and is now, since September 2022, a PhD
student in biomedical sciences. He is working on single-cell biology
in AML, including single-cell proteomics, in collaboration with
Didier Vertommen, who runs the institute’s mass spectrometry core
facility.</p>
  </li>
  <li>
    <p><strong>Andrea Vicini</strong> (Github:
<a href="https://github.com/andreavicini">andreavicini</a>) joined the lab in
October 2022 as a research assistant in bioinformatics. He focuses
on integrative multi-omics approach to study infectious diseases as
part of the large, inter-institute
<a href="https://www.saintluc.be/fr/hygieia">HYGIEIA</a> project.</p>
  </li>
  <li>
    <p><strong>Enes Sefa Ayar</strong> (Github:
<a href="https://github.com/EnesSefaAyar">EnesSefaAyar</a>) joined the lab in
November 2023 as a PhD student working on quantitative analysis of
single-cell proteomics. He did a master’s in computational sciences
and engineering at the Koç University. He is funded by a FNRS
‘Projet de Recherche’ (PDR) awarded to L. Gatto.</p>
  </li>
  <li>
    <p><strong>Jérôme Linden</strong> (GitHub:
<a href="https://github.com/jerlinden">jerlinden</a>) joined the lab in
November 2023 as a Project Manager for
<a href="https://www.eomix.be">e-OMIX</a>, a CBIO-led project aiming at
establishing a collaborative platform for the sharing and analyses
of multiomics data. He has a PhD from ULiège and is funded by the
European Regional Development Fund (ERDF) awarded to L. Gatto and
S. Jodogne.</p>
  </li>
  <li>
    <p><strong>Loïc Guille</strong> (ORCID:
<a href="https://orcid.org/0000-0002-8387-1092">0000-0002-8387-1092</a>,
GitHub: <a href="https://github.com/loguille">loguille</a>) joined the lab in
May 2024 as a bioinformatician for <a href="https://www.eomix.be">e-OMIX</a> a
CBIO-led project.  He has a master’s degree in bioinformatic from
Lyon 1 university.</p>
  </li>
  <li>
    <p><strong>Riya Dogra</strong> (GitHub: <a href="https://github.com/rdogra8">rdogra8</a>), is a
PhD student at the de Duve Institute since June 2024, working on
analysing PTMs using spatial and single cell proteomics
approaches. She has a Master’s in Systems Biology and Bioinformatics
from Panjab University, and worked in the Genomics Research lab at
CSIR-CCMB.</p>
  </li>
  <li>
    <p><strong>Guillaume Deflandre</strong> (GitHub:
<a href="https://github.com/guideflandre">guideflandre</a>), did his Master’s
thesis in the CBIO lab and is now pursuing a PhD in biomedical
sciences at the de Duve institute since October 2024. His project is
centered around the improvement of peptide identifications in
single-cell proteomics. He has a Master’s degree in chemistry and
bio-industries from the UCLouvain.</p>
  </li>
  <li>
    <p><strong>Matthieu Hodeige</strong> is a joint PhD student in biomedical sciences
between the CBIO Lab and the proteomics division of the <a href="https://www.mslab.uliege.be/cms/c_6417994/en/mass-spectrometry-laboratory">Mass
Spectrometry
Laboratory</a>
at the University of Liège, led by Gabriel Mazzucchelli. His
research focuses on developing a cellular-scale spatial proteomics
method by combining single-cell proteomics and spatial proteomics,
enabling the study of tissue heterogeneity in complex biological
systems.</p>
  </li>
   <li>
    <p><strong>Shabnam Zaman</strong> (ORCID:
<a href="https://orcid.org/0000-0001-5399-0130">0000-0001-5399-0130</a>,
Github: <a href="https://github.com/shabnamzaman">shabnamzaman</a>) joined the lab in December 2024 as a researcher focusing on single-cell omics technologies as part of the EPILUNG-CARE consortium. She has a PhD in molecular evolutionary biology from Vrije Universiteit Brussel and is funded by an Action de Recherche Concertée (ARC) project awarded to L. Gatto, C. Pilette and C. De Smet.</p>
  </li>
</ul>

<h2 id="undergraduates">Undergraduates</h2>

<ul>
  <li><strong>Léopold Guyot</strong> (GitHub:
  <a href="https://github.com/leopoldguyot">leopoldguyot</a>), is a final year
  Bioinformatics student at the “Université Libre de Bruxelles”
  (ULB).  He is currently working on his Master’s thesis on
  single-cell proteomics in the CBIO lab under the supervision of
  Laurent Gatto and <a href="https://www.ulb.be/fr/matthieu-defrance-1">Matthieu
  Defrance</a>.</li>
</ul>

<p><img src="/images/CBIO-20240703.jpg" alt="CBIO lab photo" /></p>

<p>From left to right: Samuel, Riya, Enes, Andrea, Guillaume, Philippe,
Axelle, Jérôme, Laurent, Loïc and Lieven (from the
<a href="https://uclouvain.be/en/technology-platforms/smcs.html">SMCS</a>). Julie
is missing. Picture taken at the de Duve Institute on 3 July 2024.</p>

<!-- ## Remote members -->

<h2 id="r-for-mass-spectrometry-core-team"><a href="https://www.rformassspectrometry.org/">R for Mass Spectrometry</a> core team</h2>

<ul>
  <li>
    <p><a href="http://sebastiangibb.de/"><strong>Sebastian Gibb</strong></a>, Department of
Anesthesiology and Intensive Care, University Medicine Greifswald,
Germany. Sebastian visited Laurent’s group in Cambridge 2014 and
since then contributes to various projects, in particular
<a href="http://lgatto.github.io/MSnbase"><code class="language-plaintext highlighter-rouge">MSnbase</code></a>,
<a href="http://bioconductor.org/packages/Pbase"><code class="language-plaintext highlighter-rouge">Pbase</code></a> and
<a href="http://bioconductor.org/packages/synapter"><code class="language-plaintext highlighter-rouge">synapter</code></a>.</p>
  </li>
  <li>
    <p><a href="http://www.eurac.edu/en/research/health/biomed/staff/Pages/staffdetails.aspx?persId=34084"><strong>Johannes
Rainer</strong></a>,
Center for Biomedicine, EURAC, Bolzano, Italy is a contributor to
the <a href="http://lgatto.github.io/MSnbase"><code class="language-plaintext highlighter-rouge">MSnbase</code></a> and
<a href="http://bioconductor.org/packages/Pbase"><code class="language-plaintext highlighter-rouge">Pbase</code></a> packages.</p>
  </li>
</ul>

<h2 id="key-collaborators">Key collaborators</h2>

<!-- - [**Paul -->
<!--   Kirk**](https://www.mrc-bsu.cam.ac.uk/people/in-alphabetical-order/h-to-m/paul-kirk/), -->
<!--   MRC Biostatistics Unit, University of Cambridge, UK. Paul is part of -->
<!--   the team developing principled Bayesian model for spatial -->
<!--   proteomics, such as the T-Augmented Gaussian Mixture (TAGM) model -->
<!--   ([Crook *et al.* -->
<!--   2018](https://doi.org/10.1371/journal.pcbi.1006516)). -->

<!-- - [**Kathyn S. Lilley**](http://www.bioc.cam.ac.uk/people/uto/lilley), -->
<!--   Cambridge Centre for Proteomics, Department of Biochemistry, -->
<!--   University of Cambridge, UK. Kathryn's lab developed the LOPIT -->
<!--   ([Dunkley *et al.* -->
<!--   2006](https://www.ncbi.nlm.nih.gov/pubmed/16618929)) and hyperLOPIT -->
<!--   ([Christoforou *et al.* -->
<!--   2016](https://www.ncbi.nlm.nih.gov/pubmed/26754106)) -->
<!--   technologies. Her lab is at the origin of some of the very best -->
<!--   spatial proteomics datasets in the world, which we help to analyse -->
<!--   and interpret by applying our dedicated software (see for example -->
<!--   [Breckels *et al.* 2016 in -->
<!--   F1000Research](https://f1000research.com/articles/5-2926/)). -->

<!-- - [**Sean Holden**](http://www.cl.cam.ac.uk/~sbh11/), Computer -->
<!--   Laboratory. We collaborate with Sean on the development and -->
<!--   application of machine learning algorithms to solve problems in -->
<!--   spatial proteomics. A recent example is transfer learning -->
<!--   ([Breckels *et al.* 2016](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004920)). -->

<ul>
  <li>
    <p><strong>Manon Martin</strong> (ORCID:
<a href="https://orcid.org/0000-0003-4800-0942">0000-0003-4800-0942</a>,
Github: <a href="https://github.com/ManonMartin">ManonMartin</a>) is a research
assistant in bioinformatics at the LIBST and ELI institues and the
SMCS core facility on the Louvain-la-Neuve campus. She worked in the
CBIO group from September 2020 to June 2022. She still collaborates
with the lab as part of the bioinformatics core facility.</p>
  </li>
  <li>
    <p><a href="https://statomics.github.io/"><strong>Lieven Clement</strong></a>, statOmics group,
UGent, Belgium. We collaborate on statistical analyses of
quantitative proteomics data, in particular the
<a href="https://bioconductor.org/packages/release/bioc/html/msqrob2.html">msqrob2</a>
R/Bioconductor package.</p>
  </li>
  <li>
    <p><a href="https://sites.google.com/site/thomasburgerswebpage/"><strong>Thomas
Burger</strong></a>,
BIG-BGE (Université Grenoble Alpes, CNRS, CEA, INSERM), France. We
collaborate with Thomas and his group on various proteomics data
analysis (for example missing data in <a href="https://www.ncbi.nlm.nih.gov/pubmed/26906401">Lazar <em>et al</em>
2016</a>) and software
(see <a href="https://www.ncbi.nlm.nih.gov/pubmed/27605098">Wieczorek <em>et al.</em>
2016</a>). Thomas
contributed the <em>perTurbo</em> algorithms to
<a href="https://bioconductor.org/packages/pRoloc"><em>pRoloc</em></a>.</p>
  </li>
  <li>
    <p><a href="https://github.com/cvanderaa"><strong>Christophe Vanderaa</strong></a>, statOmics group,
UGent, Belgium. We pursue our collaboration on the development of
computational solutions for the analysis of single-cell proteomics
data.</p>
  </li>
</ul>

<h2 id="alumni">Alumni</h2>

<ul>
  <li>
    <p><strong>Christophe Vanderaa</strong> (ORCID:
<a href="https://orcid.org/0000-0001-7443-5427">0000-0001-7443-5427</a>,
Github: <a href="https://github.com/cvanderaa">cvanderaa</a>) is a former PhD
student working on single-cell assay bioinformatics. He did a
masters in biomedical sciences at the UCLouvain and followed up with
a masters in bioinformatics at the KULeuven before conducting his
PhD training in the lab funded by an FNRS fellowship. Chris then
enrolled as a postdoc at the UGent in the statOmics group (headed by
Lieven Clement) to develop statistical frameworks for the analysis
of single-cell proteomics data.</p>
  </li>
  <li>
    <p><strong>Badr-Ali Mouaden</strong> finished his engineering degree in 2023 and did
his master’s project under the co-supervision of Laurent Gatto,
Philippe Hauchamps and <a href="https://mlg.info.ucl.ac.be/Members/MichelVerleysen">Prof Michel
Versleysen</a> from
the <a href="https://uclouvain.be/fr/facultes/epl">Louvain Engineering School
(EPL)</a>. He worked on the
development and application of deep convolution neural network on
flow cytometry data.</p>
  </li>
  <li>
    <p><strong>Kristína Gömöryová</strong>
(Github:<a href="https://github.com/KristinaGomoryova">KristinaGomoryova</a>)
joined the lab in September 2022 as an OpenMS intern. She is doing
her PhD in Animal Physiology and Immunology in the lab of
<a href="https://www.sci.muni.cz/ofiz/en/bryja/">prof. Bryja</a> at Masaryk
University. In CBIO lab she focuses on the analysis of DIA
proteomics data within the HYGIEIA project.</p>
  </li>
  <li>
    <p><strong>Charlotte Léonard</strong> (Github:
<a href="https://github.com/chachaleo">chachaleo</a>) did her Master’s project
in applied mathematics, working on single-cell multi-modal data
integration. She is co-supervised by <a href="https://mlg.info.ucl.ac.be/Members/MichelVerleysen">Michel
Verleysen</a> from
the Machine Learning Group at UCLouvain.</p>
  </li>
  <li>
    <p><strong>Manojkumar Selvaraju</strong> (ORCID:
<a href="https://orcid.org/0000-0001-6439-5652">0000-0001-6439-5652</a>,
Github: <a href="https://github.com/manogenome">manogenome</a>) worked from
September 2020 to May 2022 on software infrastructure for the
integration of genomics, transcriptomics and proteomics data.</p>
  </li>
  <li>
    <p><strong>Chong Tang</strong> (ORCID:
<a href="https://orcid.org/0000-0002-0482-8361">0000-0002-0482-8361</a>,
Github: <a href="https://github.com/plantton">plantton</a>) was part of the
CBIO lab from the very beginning in 2019 to 2021. He worked on mass
spectrometry-based proteomics software development with a special
focus on using databases to store and retrieve massive MS datasets.</p>
  </li>
  <li>
    <p><strong>Aayush Grover</strong> (ORCID:
<a href="https://orcid.org/0000-0002-3716-2850">0000-0002-3716-2850</a>,
Github: <a href="https://github.com/AayushGrover">AayushGrover</a>) is a PhD
student at the <a href="https://ml.inf.ethz.ch/">Institute for Machine Learning of ETH Zurich</a>
where he is working on computational epigenetics of cancer.
He was a master thesis student at the CBIO lab after completing his summer
internship (2020) at the CBIO lab. He worked on predicting protein subcellular
location using deep learning techniques during his internship and focused on
the task of survival prediction in cancer using multi-omics integration and
few-shot learning.</p>
  </li>
  <li>
    <p><strong>Kilian Maes</strong> (Github:
<a href="https://github.com/KilianMaes">KilianMaes</a>) did his Master’s
project in software engineering in the lab working on the clustering
of mass spectrometry data. He was co-supervised by <a href="https://www.info.ucl.ac.be/~pdupont/">Prof Pierre
Dupont</a> from the Machine
Learning Group. He now works part-time as a software engineer and
continues his medical degree at the UCLouvain.</p>
  </li>
  <li>
    <p><a href="https://olivercrook.co.uk/"><strong>Oliver Crook</strong></a>, was a Wellcome Trust
PhD student at the University of Cambridge, jointly supervised with
<a href="http://www.mrc-bsu.cam.ac.uk/people/in-alphabetical-order/h-to-m/paul-kirk/">Paul
Kirk</a>
from MRC Biostatistics Unit and <a href="http://www.bioc.cam.ac.uk/people/uto/lilley">Kathryn
Lilley</a>. Olly worked on
new Bayesian methods for spatial proteomics (see for example <a href="https://doi.org/10.1371/journal.pcbi.1006516">Crook
<em>et al.</em> 2018</a>),
formalising the theoretical foundations and implementing the
algorithms in the <a href="https://lgatto.github.io/pRoloc/"><code class="language-plaintext highlighter-rouge">pRoloc</code></a>
package. He is now pursuing post-doctoral research at the <a href="http://opig.stats.ox.ac.uk/">Protein
Informatics Group</a> in the Department of
Statistics at the University of Oxford.</p>
  </li>
  <li>
    <p><a href="https://lmsimp.github.io/"><strong>Lisa Breckels</strong></a>, Post-Doctoral
Research Associa in the Cambridge Centre for Proteomics, at the
University of Cambridge (UK). Her work focuses on the application
and development of machine learning and pattern recognition methods
to enable robust analysis of high-throughput quantitative proteomics
data.  All her methods and algorithms form part of the
<a href="http://bioconductor.org/packages/pRoloc"><code class="language-plaintext highlighter-rouge">pRoloc</code></a> framework for
protein localisation prediction.</p>
  </li>
  <li>
    <p><a href="https://github.com/tfkillian"><strong>Theo Killian</strong></a> was a
bioinformatician and research programmer in the CBIO group at the de
Duve Institute. Theo developes and maintains the
<a href="http://bioconductor.org/packages/release/data/experiment/html/depmap.html">depmap</a>
package. He now works as a Bioinformatics scientist at the
Laboratory of Translational Genetics at the KULeuven.</p>
  </li>
  <li>
    <p><a href="https://github.com/Kohze"><strong>Robin Kohze</strong></a> comes from the Radboud
University Nijmegen and visits the group in the frame of his
master’s internship. He developed a new infrastructure to facilitate
the dissemination of spatial proteomics data. He currently pursues a
PhD in the Department of Genetics at the University of Cambridge,
UK.</p>
  </li>
  <li>
    <p><a href="https://tnaake.github.io/"><strong>Thomas Naake</strong></a>, visited the group in
Cambridge in Spring 2014 as an ERASMUS student affiliated to the
University of Freiburg. He developed the first version of
<a href="https://bioconductor.org/packages/release/bioc/html/pRolocGUI.html"><code class="language-plaintext highlighter-rouge">pRolocGUI</code></a>,
an interactive visualisation tools for organelle proteomics
data. Thomas has since earner a PhD from the Max Planck Institute of
Molecular Plant Physiology.</p>
  </li>
  <li>
    <p><a href="https://orcid.org/0000-0002-3038-8929"><strong>Victoria Carr</strong></a>,
was a Part III student in Systems Biology at the University of
Cambridge, applying graph-based methods to study protein
sub-cellular localisation. Vicky went on doing a PhD at King’s
College London and is now working as a Scientific Programmer in the
Pathogen Informatics Unit at the Wellcome Sanger Institute.</p>
  </li>
</ul>

        
      </section>

      <footer class="page__meta">
        
        


        
          <p class="page__date"><strong><i class="fa fa-fw fa-calendar" aria-hidden="true"></i> Updated:</strong> <time datetime="2022-01-14">January 14, 2022</time></p>
        
      </footer>

      

      
    </div>

    
  </article>

  
  
</div>

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
      <li><strong>Follow:</strong></li>
    
    
    
    
      <li><a href="http://github.com/lgatto"><i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li>
    
    
    <li><a href="https://lgatto.github.io/feed.xml"><i class="fa fa-fw fa-rss-square" aria-hidden="true"></i> Feed</a></li>
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2025 Laurent Gatto CC-BY. Powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    <script src="https://lgatto.github.io/assets/js/main.min.js"></script>





  </body>
</html>
