<!doctype html>
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin SEO -->









<title>            Notes on the CZI workshop on single cell proteomics and metabolomics      ~/      </title>




<meta name="description" content="Here’s a summary of my recent short intro talk for the recent virtualCZI workshop on single cell proteomics and metabolomics, 8 April 2022.">




<meta property="og:locale" content="en_GB">
<meta property="og:site_name" content="~/">
<meta property="og:title" content="Notes on the CZI workshop on single cell proteomics and metabolomics">


  <link rel="canonical" href="https://lgatto.github.io/CZI-scMS-workshop/">
  <meta property="og:url" content="https://lgatto.github.io/CZI-scMS-workshop/">



  <meta property="og:description" content="Here’s a summary of my recent short intro talk for the recent virtualCZI workshop on single cell proteomics and metabolomics, 8 April 2022.">





  

  





  <meta property="og:type" content="article">
  <meta property="article:published_time" content="2022-06-13T00:00:00+02:00">








  <script type="application/ld+json">
    {
      "@context" : "http://schema.org",
      "@type" : "Person",
      "name" : "Laurent Gatto",
      "url" : "https://lgatto.github.io",
      "sameAs" : null
    }
  </script>






<!-- end SEO -->


<link href="https://lgatto.github.io/feed.xml" type="application/atom+xml" rel="alternate" title="~/ Feed">

<!-- http://t.co/dKP3o1e -->
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="https://lgatto.github.io/assets/css/main.css">

<meta http-equiv="cleartype" content="on">
    <!-- start custom head snippets -->

<!-- insert favicons. use http://realfavicongenerator.net/ -->

<!-- end custom head snippets -->
  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->
    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <button><div class="navicon"></div></button>
        <ul class="visible-links">
          <li class="masthead__menu-item masthead__menu-item--lg"><a href="https://lgatto.github.io/">~/</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/about/">About</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/year-archive/">Posts</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/cbio-lab/">Lab</a></li>
          
        </ul>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>

    



<div id="main" role="main">
  
  <div class="sidebar sticky">
  

<div itemscope itemtype="http://schema.org/Person">

  
    <div class="author__avatar">
      
        <img src="https://lgatto.github.io/images/lgatto2.jpg" class="author__avatar" alt="Laurent Gatto" itemprop="image">
      
    </div>
  

  <div class="author__content">
    <h3 class="author__name" itemprop="name">Laurent Gatto</h3>
    
      <p class="author__bio" itemprop="description">
        Open science, reproducible research, data champion, computational biology, proteomics, more omics, emacs, a lot of R, quite a bit of running, and parenting.
      </p>
    
  </div>

  <div class="author__urls-wrapper">
    <button class="btn btn--inverse">Follow</button>
    <ul class="author__urls social-icons">
      

      

      

      

      

      

      

      

      

      

      

      

      
        <li>
          <a href="https://github.com/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      

      

      

      

      <!--
  <li>
    <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs">
      <i class="fa fa-fw" aria-hidden="true"></i> Custom Social Profile Link
    </a>
  </li>
-->


  <!-- <li> -->
  <!--   <a href="http://dissem.in/r/6231/laurent-gatto" itemprop="sameAs"> -->
  <!--     <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> dissem.in -->
  <!--   </a> -->
  <!-- </li> -->

  <li>
    <a href="https://codeberg.org/lgatto" itemprop="sameAs">
      <i class="fa fa-code" aria-hidden="true"></i> codeberg
    </a>
  </li>


  <li>
    <a href="https://orcid.org/0000-0002-1520-2268" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> ORCID
    </a>
  </li>

  <li>
    <a href="https://scholar.google.co.uk/citations?user=k5DrB74AAAAJ&hl=en" itemprop="sameAs">
      <i class="fa fa-fw fa-google" aria-hidden="true"></i> scholar
    </a>
  </li>


  <!-- <li> -->
  <!--   <a href="https://impactstory.org/u/0000-0002-1520-2268" itemprop="sameAs"> -->
  <!--     <i class="fa fa-fw fa-sun-o" aria-hidden="true"></i> Impactstory -->
  <!--   </a> -->
  <!-- </li> -->
    </ul>
  </div>
</div>

  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="Notes on the CZI workshop on single cell proteomics and metabolomics">
    <meta itemprop="description" content="Here’s a summary of my recent short intro talk for the recent virtualCZI workshop on single cell proteomics and metabolomics, 8 April 2022.">
    <meta itemprop="datePublished" content="June 13, 2022">
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 class="page__title" itemprop="headline">Notes on the CZI workshop on single cell proteomics and metabolomics
</h1>
          
            <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  3 minute read
</p>
          
        </header>
      

      <section class="page__content" itemprop="text">
        <p>Here’s a summary of my recent short intro talk for the recent virtual
CZI workshop on single cell proteomics and metabolomics, 8 April 2022.</p>

<p><img src="/images/202204-czi-slide0.png" alt="Slide 0" /></p>

<h2 id="overview">Overview</h2>

<p>As an example of state-of-the-art infrastructure, I would like to
present the <a href="http://rformassspectrometry.org/">R for Mass
Spectrometry</a> initiative and
illustrate some of our approaches to tackle software and analysis
related bottlenecks in mass spectrometry-based proteomics and
metabolomics.</p>

<p><img src="/images/202204-czi-slide1.png" alt="Slide 1" /></p>

<p>R for Mass Spectrometry is formed by over 20 tightly integrated and
interoperable software packages, tackling the whole spectrum of data
and their analyses, from raw mass spectrometry data, peptide and
metabolite identification, quantitative data analysis, up to
single-cell proteomics data processing. Each package is thoroughly
tested, documented and peer-reviewed. Once a package reaches maturity,
it get submitted to and reviewed by the Bioconductor project. There’s
a project-level web page in addition to each package’s specific page
and a <a href="https://rformassspectrometry.github.io/docs/">book/tutorial</a>.</p>

<p>There are currently 14 contributors. The 3 main developers and
initiators of the project have over 30 years of cumulative experience
in software development and data analysis in mass spectrometry-based
proteomics and metabolomics.</p>

<p>A trend worth highlighting is the move away from large, monolithic,
one-size-fits-all software (especially important for fast-evolving
technologies that require flexibility) and the
integration/re-unification of these individual components/tools at the
workflow level.</p>

<h2 id="bottlenecks">Bottlenecks</h2>

<p>I would like to mention a couple of high-level challenges, beyond
those that are at our reach, such as the handling of missing data (be
it through technological means such as DIA, software solution, and/or
both), or batch effect (through better experimental designs and/or
software means) and the interaction between these two (see
<a href="https://www.tandfonline.com/doi/full/10.1080/14789450.2021.1988571">here</a>).</p>

<p><img src="/images/202204-czi-slide2.png" alt="Slide 2" /></p>

<ol>
  <li>Repeatability, reproducibility, replicability and re-use: we are
still struggling at the whole range of “doing things again”, be it
with one’s own data some time later, re-analysing someone’s else
data, using either the exact software or different ones.</li>
  <li>This struggle is in great part related to poor data, meta-data and
analysis reporting.</li>
</ol>

<p>These are particularly problematic at an early stage, when a
technology and its applications are developed and mature, such as
single-cell proteomics/metabolomics. The example of the
<a href="https://uclouvain-cbio.github.io/SCP.replication/">SCP.replication</a>
package illustrates the reproduction and replication of some published
studies, some of which we were able to reproduce and replicate, others
not at all (due for example to wrong supplementary files, missing
information about how the data were analysed, or completely missing
meta-data).</p>

<ol>
  <li>
    <p>The need for integration between software and languages is
inevitable - it provides access to more tools, even if at the cost
of increased complexity in code. Here’s <a href="https://uclouvain-cbio.github.io/SCP.replication/articles/schoof2021.html#batch-correction-1">short
example</a>
from one of the replications mentioned above, where we use
<a href="https://rstudio.github.io/reticulate/">reticulate</a> to call the
<a href="https://github.com/bfurtwa/SCeptre">SCeptre</a> Python package. Julia is another language that provides
unique libraries and features, that can be extremely useful for the
analysis of multi-condition single cell data.</p>
  </li>
  <li>
    <p>Some of these challenges and bottlenecks could be addressed, at
least to some extend, through better benchmarking. Benchmarking is
however a very difficult exercise that also required better
reporting of data, meta-data and analyses details.</p>
  </li>
</ol>

<h2 id="opportunities">Opportunities</h2>

<p>In terms of community, I think that the CZI Single Cell Biology Program
has demonstrated some notable successes. Related to some of the more
technical points I have mentioned above, interoperability between the
R and Python ecosystems of single-cell RNA sequencing and common data
structures have emerged.</p>

<p><img src="/images/202204-czi-slide3.png" alt="Slide 3" /></p>

<p>From a software point of view, the Essential Open Source Software for
Science has gotten a lot of visibility in genomics, less so in
proteomics and metabolomics. I see an opportunity to aim for a common
corpus of re-usable and interoperable software and methods as well as
better integration between software across platforms and operating
systems. The
<a href="https://rformassspectrometry.github.io/SpectriPy/">SpectriPy/</a> is one
effort in the R for Mass Spectrometry initiative.</p>

<p>I mentioned repeatability, reproducibility, replicability and better
reporting. We need better benchmarking, not necessarily as a way to
identify “the best software” (which is hardly the best, by merely the
one that performed better in a given data/user setting), but more as a
means to integrate and unify common experiences and expertise into a
workflow to address more difficult tasks.</p>

<p>In terms of single cell biology, we have a lot to learn and be
inspired by what in done in genomics, including their experience in
data integration. But in terms of applications, it’s important to
differentiate from genomics. I don’t think that aiming for the same
application, or at least no only or mainly the definition of cell
types and states (including dimensionality reduction, clustering, …)
is the best approach. Indeed, we currently lack scalability
(i.e. tens, hundreds of thousands or even millions of cells) and it
will be difficult to reach the level of genomics at a sensible
cost. We should focus for example on the <a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3001512">inference of regulatory
interactions with minimal
assumptions</a>,
notable in the light of post translational modifications. And to do
this, we need open and well-annotated data and software.</p>

        
      </section>

      <footer class="page__meta">
        
        
  


  
  
  

  <p class="page__taxonomy">
    <strong><i class="fa fa-fw fa-tags" aria-hidden="true"></i> Tags: </strong>
    <span itemprop="keywords">
    
      
      
      <a href="https://lgatto.github.io/tags/#bioconductor" class="page__taxonomy-item" rel="tag">Bioconductor</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#proteomics" class="page__taxonomy-item" rel="tag">proteomics</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#r" class="page__taxonomy-item" rel="tag">R</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#single-cell" class="page__taxonomy-item" rel="tag">single cell</a>
    
    </span>
  </p>




        
          <p class="page__date"><strong><i class="fa fa-fw fa-calendar" aria-hidden="true"></i> Updated:</strong> <time datetime="2022-06-13T00:00:00+02:00">June 13, 2022</time></p>
        
      </footer>

      

      
  <nav class="pagination">
    
      <a href="https://lgatto.github.io/cbio-jobs-2022/" class="pagination--pager" title="Open positions in the CBIO lab
">Previous</a>
    
    
      <a href="https://lgatto.github.io/running-watch/" class="pagination--pager" title="Tribute to my first running watch
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
    <div class="page__related">
      
        <h4 class="page__related-title">You May Also Enjoy</h4>
      
      <div class="grid__wrapper">
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/EuroBioc-Barcelona/" rel="permalink">CBIO’s EuroBioc2025 posters and talks
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  7 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">Before I forget posting this year’s lab EuroBioc contribution, here
are the abstract of the work we will present at the EuroBioc2025 in
Barcelona

</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/EuroBioc-Oxford/" rel="permalink">CBIO’s EuroBioc2024 posters and talks
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  11 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">The lab is preparing for EuroBioc2025 in
Barcelona, and I realise that
I forgot to post our contributions to EuroBioc2024 in
Oxford. So here they are
(with p...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/neil-postman-six-question-technology/" rel="permalink">6 Questions We Should Ask Before Adopting a New Technology
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  1 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">I stumbled across this
message
on mastodon, that directed me to 6 questions what one should ask
before adopting a new technology. These questions where posed...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/ia-irss/" rel="permalink">On cats, farts and parrots.
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  12 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">I was invited to contribute to a
seminar/discussion
on AI, and using language models in research. These are the notes I
prepared my short presentation. I am ...</p>
  </article>
</div>
        
      </div>
    </div>
  
</div>

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
      <li><strong>Follow:</strong></li>
    
    
    
    
      <li><a href="http://github.com/lgatto"><i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li>
    
    
    <li><a href="https://lgatto.github.io/feed.xml"><i class="fa fa-fw fa-rss-square" aria-hidden="true"></i> Feed</a></li>
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2025 Laurent Gatto CC-BY. Powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    <script src="https://lgatto.github.io/assets/js/main.min.js"></script>





  </body>
</html>
