<!doctype html>
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin SEO -->









<title>            MSnbase contributors 2010 - 2016 2017      ~/      </title>




<meta name="description" content="MSnbase(or here on GitHub) (Gatto andLilley, 2010) is one ofmy main software projects. I started working on the package when Imoved to Cambridge in 2010. It offers a way to import, manipulate andprocess raw mass spectrometry and quantitative proteomics data in R.">




<meta property="og:locale" content="en_GB">
<meta property="og:site_name" content="~/">
<meta property="og:title" content="MSnbase contributors 2010 - 2016 2017">


  <link rel="canonical" href="https://lgatto.github.io/msnbase-contribs/">
  <meta property="og:url" content="https://lgatto.github.io/msnbase-contribs/">



  <meta property="og:description" content="MSnbase(or here on GitHub) (Gatto andLilley, 2010) is one ofmy main software projects. I started working on the package when Imoved to Cambridge in 2010. It offers a way to import, manipulate andprocess raw mass spectrometry and quantitative proteomics data in R.">





  

  





  <meta property="og:type" content="article">
  <meta property="article:published_time" content="2016-11-27T00:00:00+01:00">








  <script type="application/ld+json">
    {
      "@context" : "http://schema.org",
      "@type" : "Person",
      "name" : "Laurent Gatto",
      "url" : "https://lgatto.github.io",
      "sameAs" : null
    }
  </script>






<!-- end SEO -->


<link href="https://lgatto.github.io/feed.xml" type="application/atom+xml" rel="alternate" title="~/ Feed">

<!-- http://t.co/dKP3o1e -->
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="https://lgatto.github.io/assets/css/main.css">

<meta http-equiv="cleartype" content="on">
    <!-- start custom head snippets -->

<!-- insert favicons. use http://realfavicongenerator.net/ -->

<!-- end custom head snippets -->
  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->
    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <button><div class="navicon"></div></button>
        <ul class="visible-links">
          <li class="masthead__menu-item masthead__menu-item--lg"><a href="https://lgatto.github.io/">~/</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/about/">About</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/year-archive/">Posts</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/cbio-lab/">Lab</a></li>
          
        </ul>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>

    



<div id="main" role="main">
  
  <div class="sidebar sticky">
  

<div itemscope itemtype="http://schema.org/Person">

  
    <div class="author__avatar">
      
        <img src="https://lgatto.github.io/images/lgatto2.jpg" class="author__avatar" alt="Laurent Gatto" itemprop="image">
      
    </div>
  

  <div class="author__content">
    <h3 class="author__name" itemprop="name">Laurent Gatto</h3>
    
      <p class="author__bio" itemprop="description">
        Open science, reproducible research, data champion, computational biology, proteomics, more omics, emacs, a lot of R, quite a bit of running, and parenting.
      </p>
    
  </div>

  <div class="author__urls-wrapper">
    <button class="btn btn--inverse">Follow</button>
    <ul class="author__urls social-icons">
      

      

      

      
        <li>
          <a href="https://keybase.io/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-key" aria-hidden="true"></i> Keybase
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      
        <li>
          <a href="https://github.com/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      

      

      

      

      <!--
  <li>
    <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs">
      <i class="fa fa-fw" aria-hidden="true"></i> Custom Social Profile Link
    </a>
  </li>
-->


  <li>
    <a href="http://dissem.in/r/6231/laurent-gatto" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> dissem.in
    </a>
  </li>

  <li>
    <a href="https://orcid.org/0000-0002-1520-2268" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> ORCID
    </a>
  </li>

  <li>
    <a href="https://scholar.google.co.uk/citations?user=k5DrB74AAAAJ&hl=en" itemprop="sameAs">
      <i class="fa fa-fw fa-google" aria-hidden="true"></i> scholar
    </a>
  </li>

  <!-- <li> -->
  <!--   <a href="https://impactstory.org/u/0000-0002-1520-2268" itemprop="sameAs"> -->
  <!--     <i class="fa fa-fw fa-sun-o" aria-hidden="true"></i> Impactstory -->
  <!--   </a> -->
  <!-- </li> -->

    </ul>
  </div>
</div>

  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="MSnbase contributors 2010 - 2016 2017">
    <meta itemprop="description" content="MSnbase(or here on GitHub) (Gatto andLilley, 2010) is one ofmy main software projects. I started working on the package when Imoved to Cambridge in 2010. It offers a way to import, manipulate andprocess raw mass spectrometry and quantitative proteomics data in R.">
    <meta itemprop="datePublished" content="November 27, 2016">
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 class="page__title" itemprop="headline">MSnbase contributors 2010 - <del>2016</del> 2017
</h1>
          
            <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  4 minute read
</p>
          
        </header>
      

      <section class="page__content" itemprop="text">
        <p><a href="http://bioconductor.org/packages/release/bioc/html/MSnbase.html">MSnbase</a>
(or <a href="https://github.com/lgatto/MSnbase/">here</a> on GitHub) (<a href="https://www.ncbi.nlm.nih.gov/pubmed/22113085">Gatto and
Lilley, 2010</a>) is one of
my main software projects. I started working on the package when I
moved to Cambridge in 2010. It offers a way to import, manipulate and
process raw mass spectrometry and quantitative proteomics data in R.</p>

<p>Since then, I benefited from quite a few contributions (which I
already briefly highlighted
<a href="http://lgatto.github.io/cpu-spat-prot-2015">here</a>). In this post, I
want to give a few more details about and credit to the contributors.</p>

<!--more-->

<p>The figure below summarises the contributors over time.</p>

<p><img src="/images/msnbase-contributors-2018-01-04.png" alt="MSnbase contributors" /></p>

<p>The first data point on the plot is Mon Oct 4 23:35:23 2010, and
corresponds to the very first git commit (with typo) in the
<a href="https://github.com/lgatto/MSnbase/">GitHub repository</a>:</p>

<figure class="highlight"><pre><code class="language-text" data-lang="text">commit 7cb6b1a598d0b2ed55234f75229b925ceb26afaa
Author: Laurent Gatto &lt;laurent.gatto@gmail.com&gt;
Date:   Mon Oct 4 23:35:23 2010 +0100

    Inital commit - package dir structure and R code
    
    All the R code files, some with roxygen incode documentation
    as well as the R package structure are committed. Included
    are also DESCRIPTION, NAMESPACE, NEWS (empty) and README.org.
    A dataset in mzXML format, dummyiTRAQ.mzXML in inst/extdata
    will serve as testing file.</code></pre></figure>

<p>Since then, the commits have been quite regular, except for 2012.</p>

<p><img src="/images/msnbase-contributors-github-2016-11-27.png" alt="GitHub commits" /></p>

<h3 id="guangchuang-yu">Guangchuang Yu</h3>

<p>Guangchuang contributed the <code class="language-plaintext highlighter-rouge">plotMzDelta</code> function in June 2011. The
function produces a figure used as a quality for MS2 spectra, as
detailed in
<a href="https://www.ncbi.nlm.nih.gov/pubmed/21538885">Foster et al. 2011</a>. All
differences between neighbouring peaks in MS2 spectra are calculated
and plotted as a histogram. Assuming good peptide fragmentation and
absence of contamination, the histogram should feature peaks
corresponding to amino acids.</p>

<p><img src="/images/msnbase-plotMzDelta.png" alt="plotMzDelta output" /></p>

<p>He is also the author of the first implementation of the <code class="language-plaintext highlighter-rouge">readMgfData</code>
that, as the name implies, reads <code class="language-plaintext highlighter-rouge">mgf</code> data (thanks for the
<a href="https://twitter.com/guangchuangyu/status/803414524657221632?s=09">reminder</a>).</p>

<h3 id="samuel-wieczorek-and-cosmin-lazar">Samuel Wieczorek and Cosmin Lazar</h3>

<p>The contributions of Sam and Cosmin in February 2014 stem from our
work on missing value imputation in quantitative proteomics
(<a href="https://www.ncbi.nlm.nih.gov/pubmed/26906401">Lazar et al. 2016</a>)
and have materialised in improvements in the <code class="language-plaintext highlighter-rouge">impute</code> function.</p>

<h3 id="vlad-petyuk">Vlad Petyuk</h3>

<p>Vladâ€™s main contrition was in the <code class="language-plaintext highlighter-rouge">combineFeatures</code> function that
aggregates low level features, in March 2014. He contributed the
<em>redundancy handler</em>, that defines how to handle peptides that can be
associated to multiple higher-level features (proteins).</p>

<p>The <code class="language-plaintext highlighter-rouge">ggplot2</code>-based implementation of <code class="language-plaintext highlighter-rouge">image</code>, that produces a simple
heatmap of the quantitative data also come from him, and is based on
his own <code class="language-plaintext highlighter-rouge">vp.misc::image_msnset</code> implementation.</p>

<p><img src="/images/msnbase-image-facetBy.png" alt="Facetted image" /></p>

<h3 id="thomas-naake">Thomas Naake</h3>

<p>Thomas visited the group as an Erasmus student from April to June 2014
and implemented the first version of the
<a href="http://www.bioconductor.org/packages/release/bioc/html/pRolocGUI.html"><code class="language-plaintext highlighter-rouge">pRolocGUI</code></a>
package (and
<a href="https://github.com/ComputationalProteomicsUnit/pRolocGUI">here</a> on
GitHub). During this work, we discussed about features that would be
needed for the interactive visualisation that ended up being
implemented/added to <code class="language-plaintext highlighter-rouge">MSnbase</code> and then used in the GUIs. The main one
I can remember is the <code class="language-plaintext highlighter-rouge">FreaturesOfInterest</code> class, that stores an
arbitrary set of features (proteins) that can then conveniently
highlighted on a PCA plot using the <code class="language-plaintext highlighter-rouge">highlightOnPlot</code> function from
the <code class="language-plaintext highlighter-rouge">pRoloc</code> package.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="s2">"pRolocdata"</span><span class="p">)</span><span class="w">
</span><span class="n">data</span><span class="p">(</span><span class="s2">"tan2009r1"</span><span class="p">)</span><span class="w">
</span><span class="n">x</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">FeaturesOfInterest</span><span class="p">(</span><span class="n">description</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"A test set of features of interest"</span><span class="p">,</span><span class="w">
                        </span><span class="n">fnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">featureNames</span><span class="p">(</span><span class="n">tan2009r1</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">],</span><span class="w">
                        </span><span class="n">object</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">tan2009r1</span><span class="p">)</span><span class="w">
</span><span class="n">plot2D</span><span class="p">(</span><span class="n">tan2009r1</span><span class="p">)</span><span class="w">
</span><span class="n">highlightOnPlot</span><span class="p">(</span><span class="n">tan2009r1</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="p">)</span><span class="w">
</span><span class="n">highlightOnPlot</span><span class="p">(</span><span class="n">tan2009r1</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">pos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span></code></pre></figure>

<p><img src="/images/fois-highlightOnPlot.png" alt="Highlight features of interest" /></p>

<h3 id="sebastian-gibb">Sebastian Gibb</h3>

<p>Sebastian visited the group for 3 months in 2014. He did a lot of work
on <code class="language-plaintext highlighter-rouge">MSnbase</code> and
<a href="http://bioconductor.org/packages/release/bioc/html/synapter.html"><code class="language-plaintext highlighter-rouge">synapter</code></a>
(<a href="https://github.com/lgatto/synapter">here</a>) and is still
active. Among his many contributions are the <code class="language-plaintext highlighter-rouge">addIdentificationData</code>,
that adds identification data from <code class="language-plaintext highlighter-rouge">mzid</code> files to raw (<code class="language-plaintext highlighter-rouge">MSnExp</code>
objects) and quantitative (<code class="language-plaintext highlighter-rouge">MSnSet</code> objects) data. He also added
various raw data processing functions (such as smoothing, peak
picking) by leveraging existing code in his
<a href="https://cran.r-project.org/web/packages/MALDIquant/"><code class="language-plaintext highlighter-rouge">MALDIQuant</code></a>
package and support for label-free MS2 quantitation. He also worked on
spectra comparison, annotation and visualisation, as illustrated
below.</p>

<p><img src="/images/msnbase-spectrum-plots.png" alt="Spectum annotation and comparison" /></p>

<h3 id="richie-cotton">Richie Cotton</h3>

<p>Richie contributed in supporting <code class="language-plaintext highlighter-rouge">mzTab</code> version 1.0, as described in
<a href="https://github.com/lgatto/MSnbase/issues/41">issue #41</a> from
June 2015. I updated his code to fit into the <code class="language-plaintext highlighter-rouge">MSnbase</code> infrastructure
and annotate some the ontology controlled parameters (using the
<a href="http://bioconductor.org/packages/release/bioc/html/rols.html"><code class="language-plaintext highlighter-rouge">rols</code></a>
package).</p>

<h3 id="martina-fischer">Martina Fischer</h3>

<p>In June 2015, Martina contributed a whole new method for feature
aggregation, termed <em>iPQF</em> (for Isobaric Protein Quantification based
on Features)). <em>iPQF</em> is a new peptide-to-protein summarisation method
using peptide spectra characteristics to improve protein
quantification. All details in
<a href="https://www.ncbi.nlm.nih.gov/pubmed/26589272">Fischer and Renard, 2016</a>,
readily available using <code class="language-plaintext highlighter-rouge">combineFeatures(..., method = "iPQF")</code>.</p>

<h3 id="johannes-rainer">Johannes Rainer</h3>

<p>Johannes has been instrumental in the recent (October 2016) release of
<code class="language-plaintext highlighter-rouge">MSnbase</code> version 2.0. During summer 2016, we worked on a new backend
for raw data. Instead of loading spectra into memory, as in the
original <code class="language-plaintext highlighter-rouge">MSnExp</code> implementation, the alternative implementation
accesses the raw data from the hard drive on-the-fly only when it is
needed. This is made possible by the fast on-disk access provided by
the
<a href="http://bioconductor.org/packages/release/bioc/html/mzR.html"><code class="language-plaintext highlighter-rouge">mzR</code></a>
package (<a href="https://github.com/sneumann/mzR/">here</a> on GitHub) that uses
the <a href="http://proteowizard.sourceforge.net/">proteowizard</a> C/C++ code
base under the hood. For more details and a direct comparison, see the
<a href="http://bioconductor.org/packages/release/bioc/vignettes/MSnbase/inst/doc/benchmarking.html">benchmarking</a>
vignette.</p>

<p>This joint work with Johannes aims at providing a common and efficient
infrastructure for mass spectrometry data that can be used by the
proteomics and metabolomics developers.</p>

<h3 id="arne-smits">Arne Smits</h3>

<p>Arne developes the
<a href="https://bioconductor.org/packages/release/bioc/html/DEP.html"><code class="language-plaintext highlighter-rouge">DEP</code></a>
package for differential enrichment analysis of proteomics data and
has <a href="https://github.com/lgatto/MSnbase/pull/284">contributed</a> <code class="language-plaintext highlighter-rouge">MSnSet</code>
to/from <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> methods, to facilitate the
inter-operability between his package and <code class="language-plaintext highlighter-rouge">MSnbase</code>.</p>

<p>Seeing that <code class="language-plaintext highlighter-rouge">MSnbase</code> is used and attracts attention from other
developers is a great reward for me. Thank you all for your valuable
contributions!</p>

<p><strong>Updates</strong> Since its first publication (2016-11-27), this post has
been ammended to add Arne Smitsâ€™ contribution (2018-01-04).</p>

        
      </section>

      <footer class="page__meta">
        
        
  


  
  
  

  <p class="page__taxonomy">
    <strong><i class="fa fa-fw fa-tags" aria-hidden="true"></i> Tags: </strong>
    <span itemprop="keywords">
    
      
      
      <a href="https://lgatto.github.io/tags/#bioconductor" class="page__taxonomy-item" rel="tag">Bioconductor</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#cpu" class="page__taxonomy-item" rel="tag">CPU</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#msnbase" class="page__taxonomy-item" rel="tag">MSnbase</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#r" class="page__taxonomy-item" rel="tag">R</a>
    
    </span>
  </p>




        
          <p class="page__date"><strong><i class="fa fa-fw fa-calendar" aria-hidden="true"></i> Updated:</strong> <time datetime="2016-11-27T00:00:00+01:00">November 27, 2016</time></p>
        
      </footer>

      

      
  <nav class="pagination">
    
      <a href="https://lgatto.github.io/brief-data-analysis-advice/" class="pagination--pager" title="Some brief advice on data analysis
">Previous</a>
    
    
      <a href="https://lgatto.github.io/branch-specific-gitignore/" class="pagination--pager" title="Branch specific gitignores for Bioconductor package documentation
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
    <div class="page__related">
      
        <h4 class="page__related-title">You May Also Enjoy</h4>
      
      <div class="grid__wrapper">
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/ia-irss/" rel="permalink">On cats, farts and parrots.
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  12 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">I was invited to contribute to a
seminar/discussion
on AI, and using language models in research. These are the notes I
prepared my short presentation. I am ...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/oublions-les-notes/" rel="permalink">Podcasts du LLL: Et si on oubliait les notes?
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  less than 1 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">DeuxiÃ¨me billet en franÃ§ais rÃ©fÃ©renÃ§ant un podcast du Louvain
Learning Lab (le
premier est ici). Cette
fois, je fais un peu dâ€™auto-promotion:

</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/ecr-recognition/" rel="permalink">HUPO ECR Online Panel Discussion - Getting recognised for your work
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  8 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">The HUPO Early Career Researcher (ECR) committee has organised a
discussion panel on Getting recognised for your work and have asked
me to participate - than...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/the-grid-in-r/" rel="permalink">The Grid poster, in R
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  3 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">The MuseeL is the
UCLouvain University museum in
Louvain-la-Neuve. Highly recommended. Itâ€™s located to the lively
place des sciences, in a nice brutalist sty...</p>
  </article>
</div>
        
      </div>
    </div>
  
</div>

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
      <li><strong>Follow:</strong></li>
    
    
    
    
      <li><a href="http://github.com/lgatto"><i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li>
    
    
    <li><a href="https://lgatto.github.io/feed.xml"><i class="fa fa-fw fa-rss-square" aria-hidden="true"></i> Feed</a></li>
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2025 Laurent Gatto CC-BY. Powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    <script src="https://lgatto.github.io/assets/js/main.min.js"></script>





  </body>
</html>
