<!doctype html>
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin SEO -->









<title>            New feature: annotated chromatograms      ~/      </title>




<meta name="description" content="Today, I’m going to introduce a recent feature from theSpectra package,namely annotated chromatograms. I’ll start by showing the result andthem explain how to produce it.">




<meta property="og:locale" content="en_GB">
<meta property="og:site_name" content="~/">
<meta property="og:title" content="New feature: annotated chromatograms">


  <link rel="canonical" href="https://lgatto.github.io/annotchrom/">
  <meta property="og:url" content="https://lgatto.github.io/annotchrom/">



  <meta property="og:description" content="Today, I’m going to introduce a recent feature from theSpectra package,namely annotated chromatograms. I’ll start by showing the result andthem explain how to produce it.">





  

  





  <meta property="og:type" content="article">
  <meta property="article:published_time" content="2021-12-24T00:00:00+01:00">








  <script type="application/ld+json">
    {
      "@context" : "http://schema.org",
      "@type" : "Person",
      "name" : "Laurent Gatto",
      "url" : "https://lgatto.github.io",
      "sameAs" : null
    }
  </script>






<!-- end SEO -->


<link href="https://lgatto.github.io/feed.xml" type="application/atom+xml" rel="alternate" title="~/ Feed">

<!-- http://t.co/dKP3o1e -->
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="https://lgatto.github.io/assets/css/main.css">

<meta http-equiv="cleartype" content="on">
    <!-- start custom head snippets -->

<!-- insert favicons. use http://realfavicongenerator.net/ -->

<!-- end custom head snippets -->
  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->
    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <button><div class="navicon"></div></button>
        <ul class="visible-links">
          <li class="masthead__menu-item masthead__menu-item--lg"><a href="https://lgatto.github.io/">~/</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/about/">About</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/year-archive/">Posts</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/cbio-lab/">Lab</a></li>
          
        </ul>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>

    



<div id="main" role="main">
  
  <div class="sidebar sticky">
  

<div itemscope itemtype="http://schema.org/Person">

  
    <div class="author__avatar">
      
        <img src="https://lgatto.github.io/images/lgatto2.jpg" class="author__avatar" alt="Laurent Gatto" itemprop="image">
      
    </div>
  

  <div class="author__content">
    <h3 class="author__name" itemprop="name">Laurent Gatto</h3>
    
      <p class="author__bio" itemprop="description">
        Open science, reproducible research, data champion, computational biology, proteomics, more omics, emacs, a lot of R, quite a bit of running, and parenting.
      </p>
    
  </div>

  <div class="author__urls-wrapper">
    <button class="btn btn--inverse">Follow</button>
    <ul class="author__urls social-icons">
      

      

      

      

      

      

      

      

      

      

      

      

      
        <li>
          <a href="https://github.com/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      

      

      

      

      <!--
  <li>
    <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs">
      <i class="fa fa-fw" aria-hidden="true"></i> Custom Social Profile Link
    </a>
  </li>
-->


  <!-- <li> -->
  <!--   <a href="http://dissem.in/r/6231/laurent-gatto" itemprop="sameAs"> -->
  <!--     <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> dissem.in -->
  <!--   </a> -->
  <!-- </li> -->

  <li>
    <a href="https://codeberg.org/lgatto" itemprop="sameAs">
      <i class="fa fa-code" aria-hidden="true"></i> codeberg
    </a>
  </li>


  <li>
    <a href="https://orcid.org/0000-0002-1520-2268" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> ORCID
    </a>
  </li>

  <li>
    <a href="https://scholar.google.co.uk/citations?user=k5DrB74AAAAJ&hl=en" itemprop="sameAs">
      <i class="fa fa-fw fa-google" aria-hidden="true"></i> scholar
    </a>
  </li>


  <!-- <li> -->
  <!--   <a href="https://impactstory.org/u/0000-0002-1520-2268" itemprop="sameAs"> -->
  <!--     <i class="fa fa-fw fa-sun-o" aria-hidden="true"></i> Impactstory -->
  <!--   </a> -->
  <!-- </li> -->
    </ul>
  </div>
</div>

  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="New feature: annotated chromatograms">
    <meta itemprop="description" content="Today, I’m going to introduce a recent feature from theSpectra package,namely annotated chromatograms. I’ll start by showing the result andthem explain how to produce it.">
    <meta itemprop="datePublished" content="December 24, 2021">
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 class="page__title" itemprop="headline">New feature: annotated chromatograms
</h1>
          
            <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  4 minute read
</p>
          
        </header>
      

      <section class="page__content" itemprop="text">
        <p>Today, I’m going to introduce a recent feature from the
<a href="https://rformassspectrometry.github.io/Spectra/">Spectra</a> package,
namely annotated chromatograms. I’ll start by showing the result and
them explain how to produce it.</p>

<h2 id="an-annotated-chromatogram">An annotated chromatogram</h2>

<p>You will immediately recognise a chromatogram on the figure below,
showing MS1 scan total ion current over the experiment’s retention
time. Each MS1 event is highlighted by a dot which is colour-coded
based on the number of MS2 offspring scans that lead to
identifications. Grey dots are MS1 scans without any identified
precursor peaks while coloured dots (from dark to light blue)
represent MS1 scans have 1 to 10 (in one case, see details below)
identified precursor peaks.</p>

<p><img src="https://rformassspectrometry.github.io/docs/R4MS_files/figure-html/nSequencePlot-1.png" alt="Annotated chromatogram" /></p>

<p>You can also watch a video illustrating this post <a href="https://youtu.be/skOfnVeJc_U">here</a>.</p>

<h2 id="the-data">The data</h2>

<p>The generate the data for an annotated chromatogram, we need raw and
identification data and join them together. We will use the
<em>TMT_Erwinia</em> data available in the
<a href="http://bioconductor.org/packages/release/data/experiment/html/msdata.html">msdata</a>. Below,
we get the raw and identification data files.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="s2">"msdata"</span><span class="p">)</span><span class="w">
</span><span class="n">basename</span><span class="p">(</span><span class="n">mzml_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">proteomics</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"20141210.mzML"</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## [1] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML.gz"</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">basename</span><span class="p">(</span><span class="n">mzid_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ident</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"20141210.mzid"</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## [1] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzid"</code></pre></figure>

<p>Here, we load the raw data into R as a <code class="language-plaintext highlighter-rouge">Spectra</code> object:</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="s2">"Spectra"</span><span class="p">)</span><span class="w">
</span><span class="n">rw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Spectra</span><span class="p">(</span><span class="n">mzml_file</span><span class="p">)</span><span class="w">
</span><span class="n">rw</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## MSn data (Spectra) with 7534 spectra in a MsBackendMzR backend:
##        msLevel     rtime scanIndex
##      &lt;integer&gt; &lt;numeric&gt; &lt;integer&gt;
## 1            1    0.4584         1
## 2            1    0.9725         2
## 3            1    1.8524         3
## 4            1    2.7424         4
## 5            1    3.6124         5
## ...        ...       ...       ...
## 7530         2   3600.47      7530
## 7531         2   3600.83      7531
## 7532         2   3601.18      7532
## 7533         2   3601.57      7533
## 7534         2   3601.98      7534
##  ... 33 more variables/columns.
##
## file(s):
## TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML.gz</code></pre></figure>

<p>Below, we load the identification data into R as a <code class="language-plaintext highlighter-rouge">PSM</code> object and
filter the PSMs:</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="s2">"PSMatch"</span><span class="p">)</span><span class="w">
</span><span class="n">id</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">PSM</span><span class="p">(</span><span class="n">mzid_file</span><span class="p">)</span><span class="w"> </span><span class="o">|&gt;</span><span class="w">
   </span><span class="n">filterPSMs</span><span class="p">()</span><span class="w">
</span><span class="n">id</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## PSM with 2666 rows and 35 columns.
## Spectra: 2646 unique
##   db: 2666 target, 0 decoy
##   ranks: 1:2666
## Peptides: 2324 unique, 0 multiple
## Proteins: 1466
## names(35): sequence spectrumID ... subReplacementResidue subLocation</code></pre></figure>

<p>We can now annotate the spectra with the identification data. For
details about these steps, see the details and example of the
<a href="https://rformassspectrometry.github.io/Spectra/reference/Spectra.html#data-subsetting-filtering-and-merging-1">joinSpectraData()</a>
function.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">rw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">joinSpectraData</span><span class="p">(</span><span class="n">rw</span><span class="p">,</span><span class="w"> </span><span class="n">id</span><span class="p">,</span><span class="w">
                </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"spectrumId"</span><span class="p">,</span><span class="w">
                </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"spectrumID"</span><span class="p">)</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">na.omit</span><span class="p">(</span><span class="n">rw</span><span class="o">$</span><span class="n">sequence</span><span class="p">))</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## [1] "ERKLIKKIAKTLVK" "IVILVLIVLFIQKR" "THSQEEMQHMQR"   "IDHRSIRVQYK"
## [5] "THSQEEMQHMQR"   "RNYKTNNLYLKLK"</code></pre></figure>

<h2 id="the-countidentifications-function">The <code class="language-plaintext highlighter-rouge">countIdentifications()</code> function</h2>

<p>The function that permit to produce the data for the figure above is
<a href="https://rformassspectrometry.github.io/Spectra/reference/countIdentifications.html">countIdentifications()</a>.</p>

<p>The function is going to tally the number of identifications (i.e
non-missing characters in the <code class="language-plaintext highlighter-rouge">sequence</code> spectra variable) for each
scan. In the case of MS2 scans, these will be either 1 or 0, depending
the presence of a sequence. For MS1 scans, the function will count the
number of sequences for the descendant MS2 scans, i.e. those produced
from precursor ions from each MS1 scan.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">rw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">countIdentifications</span><span class="p">(</span><span class="n">rw</span><span class="p">)</span></code></pre></figure>

<p>Below, we see on the second line that 3457 MS2 scans lead to no PSM,
while 2546 lead to an identification. Among all MS1 scans, 833 lead to
no MS2 scans with PSMs. 30 MS1 scans generated one MS2 scan that lead
to a PSM, 45 lead to two PSMs, …</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">table</span><span class="p">(</span><span class="n">msLevel</span><span class="p">(</span><span class="n">rw</span><span class="p">),</span><span class="w"> </span><span class="n">rw</span><span class="o">$</span><span class="n">countIdentifications</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">##
##        0    1    2    3    4    5    6    7    8    9   10
##   1  833   30   45   97  139  132   92   42   17    3    1
##   2 3457 2646    0    0    0    0    0    0    0    0    0</code></pre></figure>

<p>We can now use this new <code class="language-plaintext highlighter-rouge">countIdentifications</code> variable to generate
our annotated chromatogram. The code chunk below filters MS1 level
data and then extracts the spectra variable, in particular the
retention time <code class="language-plaintext highlighter-rouge">rtime</code>, the total ion current <code class="language-plaintext highlighter-rouge">totIonCurrent</code> and the
newly created <code class="language-plaintext highlighter-rouge">countIdentifications</code> to produce a figure with
<a href="https://ggplot2.tidyverse.org/">ggplot2</a>.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span><span class="w">
</span><span class="n">rw</span><span class="w"> </span><span class="o">|&gt;</span><span class="w">
</span><span class="n">filterMsLevel</span><span class="p">(</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">|&gt;</span><span class="w">
</span><span class="n">spectraData</span><span class="p">()</span><span class="w"> </span><span class="o">|&gt;</span><span class="w">
</span><span class="n">as_tibble</span><span class="p">()</span><span class="w"> </span><span class="o">|&gt;</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rtime</span><span class="p">,</span><span class="w">
           </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">totIonCurrent</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">geom_line</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.25</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">geom_point</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ifelse</span><span class="p">(</span><span class="n">countIdentifications</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w">
                                   </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">countIdentifications</span><span class="p">)),</span><span class="w">
               </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w">
               </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">labs</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Number of ids"</span><span class="p">)</span></code></pre></figure>

<p><img src="/images/countIdentificationsFig-1.png" alt="plot of chunk countIdentificationsFig" /></p>

<h2 id="more-details-and-session-information">More details and session information</h2>

<p>For details about the <code class="language-plaintext highlighter-rouge">Spectra</code> package, see the <a href="https://rformassspectrometry.github.io/Spectra/">package web
page</a> and the <a href="https://rformassspectrometry.github.io/docs/"><em>R for
Mass Spectrometry</em>
tutorial</a>.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">sessionInfo</span><span class="p">()</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text">## R version 4.1.2 (2021-11-01)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
##
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/atlas/libblas.so.3.10.3
## LAPACK: /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3.10.3
##
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_US.UTF-8
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_US.UTF-8
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C
##
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods
## [8] base
##
## other attached packages:
##  [1] forcats_0.5.1       stringr_1.4.0       dplyr_1.0.7
##  [4] purrr_0.3.4         readr_2.1.1         tidyr_1.1.4
##  [7] tibble_3.1.6        ggplot2_3.3.5       tidyverse_1.3.1
## [10] PSMatch_0.7.1       Spectra_1.5.2       ProtGenerics_1.26.0
## [13] BiocParallel_1.28.2 S4Vectors_0.32.3    BiocGenerics_0.40.0
## [16] msdata_0.34.0       knitr_1.36
##
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7                matrixStats_0.61.0
##  [3] fs_1.5.1                    lubridate_1.8.0
##  [5] httr_1.4.2                  GenomeInfoDb_1.30.0
##  [7] tools_4.1.2                 backports_1.4.0
##  [9] utf8_1.2.2                  R6_2.5.1
## [11] DBI_1.1.1                   lazyeval_0.2.2
## [13] colorspace_2.0-2            withr_2.4.3
## [15] tidyselect_1.1.1            compiler_4.1.2
## [17] cli_3.1.0                   rvest_1.0.2
## [19] Biobase_2.54.0              xml2_1.3.3
## [21] DelayedArray_0.20.0         labeling_0.4.2
## [23] scales_1.1.1                digest_0.6.29
## [25] XVector_0.34.0              pkgconfig_2.0.3
## [27] MatrixGenerics_1.6.0        highr_0.9
## [29] dbplyr_2.1.1                rlang_0.4.12
## [31] readxl_1.3.1                rstudioapi_0.13
## [33] farver_2.1.0                generics_0.1.1
## [35] jsonlite_1.7.2              RCurl_1.98-1.5
## [37] magrittr_2.0.1              GenomeInfoDbData_1.2.7
## [39] Matrix_1.4-0                Rcpp_1.0.7
## [41] munsell_0.5.0               fansi_0.5.0
## [43] MsCoreUtils_1.6.0           lifecycle_1.0.1
## [45] stringi_1.7.6               MASS_7.3-54
## [47] SummarizedExperiment_1.24.0 zlibbioc_1.40.0
## [49] grid_4.1.2                  parallel_4.1.2
## [51] crayon_1.4.2                lattice_0.20-45
## [53] haven_2.4.3                 hms_1.1.1
## [55] mzR_2.28.0                  pillar_1.6.4
## [57] igraph_1.2.9                GenomicRanges_1.46.1
## [59] QFeatures_1.4.0             codetools_0.2-18
## [61] reprex_2.0.1                glue_1.5.1
## [63] evaluate_0.14               modelr_0.1.8
## [65] MultiAssayExperiment_1.20.0 vctrs_0.3.8
## [67] tzdb_0.2.0                  cellranger_1.1.0
## [69] gtable_0.3.0                clue_0.3-60
## [71] assertthat_0.2.1            xfun_0.28
## [73] broom_0.7.10                AnnotationFilter_1.18.0
## [75] ncdf4_1.18                  IRanges_2.28.0
## [77] cluster_2.1.2               ellipsis_0.3.2</code></pre></figure>


        
      </section>

      <footer class="page__meta">
        
        
  


  
  
  

  <p class="page__taxonomy">
    <strong><i class="fa fa-fw fa-tags" aria-hidden="true"></i> Tags: </strong>
    <span itemprop="keywords">
    
      
      
      <a href="https://lgatto.github.io/tags/#bioconductor" class="page__taxonomy-item" rel="tag">Bioconductor</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#proteomics" class="page__taxonomy-item" rel="tag">proteomics</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#r" class="page__taxonomy-item" rel="tag">R</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#software" class="page__taxonomy-item" rel="tag">software</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#spectra" class="page__taxonomy-item" rel="tag">Spectra</a>
    
    </span>
  </p>




        
          <p class="page__date"><strong><i class="fa fa-fw fa-calendar" aria-hidden="true"></i> Updated:</strong> <time datetime="2021-12-24T00:00:00+01:00">December 24, 2021</time></p>
        
      </footer>

      

      
  <nav class="pagination">
    
      <a href="https://lgatto.github.io/rdmhk/" class="pagination--pager" title="How to Manage Data: Data Stewardship and FAIR Skills
">Previous</a>
    
    
      <a href="https://lgatto.github.io/running-shoes/" class="pagination--pager" title="These shoes are made for running
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
    <div class="page__related">
      
        <h4 class="page__related-title">You May Also Enjoy</h4>
      
      <div class="grid__wrapper">
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/EuroBioc-Oxford/" rel="permalink">CBIO’s EuroBioc2024 posters and talks
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  11 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">The lab is preparing for EuroBioc2025 in
Barcelona, and I realise that
I forgot to post our contributions to EuroBioc2024 in
Oxford. So here they are
(with p...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/neil-postman-six-question-technology/" rel="permalink">6 Questions We Should Ask Before Adopting a New Technology
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  1 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">I stumbled across this
message
on mastodon, that directed me to 6 questions what one should ask
before adopting a new technology. These questions where posed...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/ia-irss/" rel="permalink">On cats, farts and parrots.
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  12 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">I was invited to contribute to a
seminar/discussion
on AI, and using language models in research. These are the notes I
prepared my short presentation. I am ...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/oublions-les-notes/" rel="permalink">Podcasts du LLL: Et si on oubliait les notes?
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  less than 1 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">Deuxième billet en français référençant un podcast du Louvain
Learning Lab (le
premier est ici). Cette
fois, je fais un peu d’auto-promotion:

</p>
  </article>
</div>
        
      </div>
    </div>
  
</div>

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
      <li><strong>Follow:</strong></li>
    
    
    
    
      <li><a href="http://github.com/lgatto"><i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li>
    
    
    <li><a href="https://lgatto.github.io/feed.xml"><i class="fa fa-fw fa-rss-square" aria-hidden="true"></i> Feed</a></li>
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2025 Laurent Gatto CC-BY. Powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    <script src="https://lgatto.github.io/assets/js/main.min.js"></script>





  </body>
</html>
