<!doctype html>
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin SEO -->









<title>            How to succeed in my exam      ~/      </title>




<meta name="description" content="This short post describes how to succeed and how to fail in answeringa typical exam question in one of my bachelor courses. I even believeit is applicable later when tackling more complex data analysisquestions in a research project.">




<meta property="og:locale" content="en_GB">
<meta property="og:site_name" content="~/">
<meta property="og:title" content="How to succeed in my exam">


  <link rel="canonical" href="https://lgatto.github.io/exam-qst/">
  <meta property="og:url" content="https://lgatto.github.io/exam-qst/">



  <meta property="og:description" content="This short post describes how to succeed and how to fail in answeringa typical exam question in one of my bachelor courses. I even believeit is applicable later when tackling more complex data analysisquestions in a research project.">





  

  





  <meta property="og:type" content="article">
  <meta property="article:published_time" content="2021-09-23T00:00:00+02:00">








  <script type="application/ld+json">
    {
      "@context" : "http://schema.org",
      "@type" : "Person",
      "name" : "Laurent Gatto",
      "url" : "https://lgatto.github.io",
      "sameAs" : null
    }
  </script>






<!-- end SEO -->


<link href="https://lgatto.github.io/feed.xml" type="application/atom+xml" rel="alternate" title="~/ Feed">

<!-- http://t.co/dKP3o1e -->
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="https://lgatto.github.io/assets/css/main.css">

<meta http-equiv="cleartype" content="on">
    <!-- start custom head snippets -->

<!-- insert favicons. use http://realfavicongenerator.net/ -->

<!-- end custom head snippets -->
  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->
    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <button><div class="navicon"></div></button>
        <ul class="visible-links">
          <li class="masthead__menu-item masthead__menu-item--lg"><a href="https://lgatto.github.io/">~/</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/about/">About</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/year-archive/">Posts</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/cbio-lab/">Lab</a></li>
          
        </ul>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>

    



<div id="main" role="main">
  
  <div class="sidebar sticky">
  

<div itemscope itemtype="http://schema.org/Person">

  
    <div class="author__avatar">
      
        <img src="https://lgatto.github.io/images/lgatto2.jpg" class="author__avatar" alt="Laurent Gatto" itemprop="image">
      
    </div>
  

  <div class="author__content">
    <h3 class="author__name" itemprop="name">Laurent Gatto</h3>
    
      <p class="author__bio" itemprop="description">
        Open science, reproducible research, data champion, computational biology, proteomics, more omics, emacs, a lot of R, quite a bit of running, and parenting.
      </p>
    
  </div>

  <div class="author__urls-wrapper">
    <button class="btn btn--inverse">Follow</button>
    <ul class="author__urls social-icons">
      

      

      

      
        <li>
          <a href="https://keybase.io/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-key" aria-hidden="true"></i> Keybase
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      
        <li>
          <a href="https://github.com/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      

      

      

      

      <!--
  <li>
    <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs">
      <i class="fa fa-fw" aria-hidden="true"></i> Custom Social Profile Link
    </a>
  </li>
-->


  <li>
    <a href="http://dissem.in/r/6231/laurent-gatto" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> dissem.in
    </a>
  </li>

  <li>
    <a href="https://orcid.org/0000-0002-1520-2268" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> ORCID
    </a>
  </li>

  <li>
    <a href="https://scholar.google.co.uk/citations?user=k5DrB74AAAAJ&hl=en" itemprop="sameAs">
      <i class="fa fa-fw fa-google" aria-hidden="true"></i> scholar
    </a>
  </li>

  <!-- <li> -->
  <!--   <a href="https://impactstory.org/u/0000-0002-1520-2268" itemprop="sameAs"> -->
  <!--     <i class="fa fa-fw fa-sun-o" aria-hidden="true"></i> Impactstory -->
  <!--   </a> -->
  <!-- </li> -->

    </ul>
  </div>
</div>

  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="How to succeed in my exam">
    <meta itemprop="description" content="This short post describes how to succeed and how to fail in answeringa typical exam question in one of my bachelor courses. I even believeit is applicable later when tackling more complex data analysisquestions in a research project.">
    <meta itemprop="datePublished" content="September 23, 2021">
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 class="page__title" itemprop="headline">How to succeed in my exam
</h1>
          
            <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  6 minute read
</p>
          
        </header>
      

      <section class="page__content" itemprop="text">
        <p>This short post describes how to succeed and how to fail in answering
a typical exam question in one of my bachelor courses. I even believe
it is applicable later when tackling more complex data analysis
questions in a research project.</p>

<aside class="sidebar__right">
<nav class="toc">
    <header><h4 class="nav__title"><i class="fa fa-file-text"></i> On This Page</h4></header>
<ul class="toc__menu" id="markdown-toc">
  <li><a href="#the-question" id="markdown-toc-the-question">The question</a></li>
  <li><a href="#pre-requisites" id="markdown-toc-pre-requisites">Pre-requisites</a></li>
  <li><a href="#its-a-trap" id="markdown-toc-its-a-trap">It’s a trap</a></li>
  <li><a href="#a-conceptual-solution" id="markdown-toc-a-conceptual-solution">A conceptual solution</a></li>
  <li><a href="#bringing-it-all-together" id="markdown-toc-bringing-it-all-together">Bringing it all together</a></li>
  <li><a href="#conclusions" id="markdown-toc-conclusions">Conclusions</a></li>
  <li><a href="#some-context" id="markdown-toc-some-context">Some context</a></li>
</ul>

  </nav>
</aside>

<h2 id="the-question">The question</h2>

<p>The student is provided with an object of class
<a href="http://www.biocondoctor.org/packages/SummarizedExperiment"><code class="language-plaintext highlighter-rouge">SummarizedExperiment</code></a>,
called <code class="language-plaintext highlighter-rouge">se</code>, that contains (part of the) data from <em>The effect of
upper-respiratory infection on transcriptomic changes in the CNS</em> by
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5544260/">Blackmore <em>et al.</em>
(2017)</a>:</p>

<blockquote>
  <p><strong>Purpose</strong>: The goal of this study was to determine the effect of an
upper-respiratory infection on changes in RNA transcription occurring
in the cerebellum and spinal cord post infection.</p>

  <p>(The provided data here only contain the cerebellum samples.)</p>

  <p><strong>Methods</strong>: Gender matched eight week old C57BL/6 mice were
inoculated saline or with Influenza A by intranasal route and
transcriptomic changes in the cerebellum and spinal cord tissues were
evaluated by RNA-seq at days 0 (non-infected), 4 and 8.</p>
</blockquote>

<p>The student is asked to visualise the gene expression distributions in
each sample using violin plots and to colour these based on the
infection statues, <em>Influenza A</em> or <em>Non-Infected</em>.</p>

<h2 id="pre-requisites">Pre-requisites</h2>

<p>There are several ways to produce this visualisation. Here, the
students have learnt to use the
<a href="https://www.tidyverse.org/">tidyverse</a> and about
<code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> objects, and not (yet) with base plotting. The
goal of this question is for them to demonstrate they can extract
different sub-parts from a <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code>, know how to
transform and combine them using standard <code class="language-plaintext highlighter-rouge">dplyr</code> functions and
eventually visualise them using <code class="language-plaintext highlighter-rouge">ggplot2</code>.</p>

<h2 id="its-a-trap">It’s a trap</h2>

<p>The main trap many students fall into is to start writing code before
having thought about how to answer the question. They have access to
the lecture notes and code as well as past exam questions. They often
start by copying code that apparently (at least) partially addresses
the question, and try to modify it until “it works”. Student often end
up wasting precious time with this random trial and error
strategy. This trial-and-error approach equates to randomly taking
buses until one brings them to they expected destination by sheer
luck… which is extremely unlikely when several buses are needed to
reach that destination.</p>

<h2 id="a-conceptual-solution">A conceptual solution</h2>

<p>My suggestion is to first think about how to answer the question
before actually answering. As a way of focusing on the how, let’s
first sketch out an plan, starting from the end.</p>

<ol>
  <li>
    <p>Based on the question, I anticipate to have a figure with samples
along the x axis and expression values along the y axis, with
violin plots of different colours based on the infection status. I
typically recommend students to have a go with pen and paper and
draw such a hypothetical figure.</p>
  </li>
  <li>
    <p>We can already easily write code that produces such a figure,
assuming that we have a long <code class="language-plaintext highlighter-rouge">data.frame</code> or <code class="language-plaintext highlighter-rouge">tibble</code> named <code class="language-plaintext highlighter-rouge">x</code>,
that contains three columns, namely <code class="language-plaintext highlighter-rouge">expression</code> with the
expression values, <code class="language-plaintext highlighter-rouge">sample</code> containing the sample names and
<code class="language-plaintext highlighter-rouge">infection</code>, with the <em>‘Influenza A</em> and <em>Non-Infected</em> sample
infection status.</p>

    <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sample</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">expression</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">infection</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">geom_violin</span><span class="p">()</span><span class="w">
</span></code></pre></div>    </div>
  </li>
  <li>
    <p>From the previous step, we know we will need the expression data,
the sample names, and their respective infection status. Even
without looking at the actual data, we know how to extract these:
<code class="language-plaintext highlighter-rouge">assay(se)</code> gives us the expression data, with column names
referring to the sample, and <code class="language-plaintext highlighter-rouge">colData(se)$infection</code> (or simply
<code class="language-plaintext highlighter-rouge">se$infection</code>) provides the infection status (assuming that it’s
encoded by the <code class="language-plaintext highlighter-rouge">infection</code> column variable).</p>
  </li>
  <li>
    <p>We also know that <code class="language-plaintext highlighter-rouge">assay(se)</code> returns the data in a wide format,
with genes along the rows and samples along the columns. Given that
we want to produce the figure sketched out in step 1 using
<code class="language-plaintext highlighter-rouge">ggplot()</code> in step 2, we need to convert it into a long table,
which we can do use with <code class="language-plaintext highlighter-rouge">pivot_longer()</code>.</p>

    <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pivot_longer</span><span class="p">(</span><span class="n">assay</span><span class="p">(</span><span class="n">se</span><span class="p">),</span><span class="w">
             </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"sample"</span><span class="p">,</span><span class="w">
             </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"expression"</span><span class="p">,</span><span class="w">
             </span><span class="n">everything</span><span class="p">())</span><span class="w">
</span></code></pre></div>    </div>

    <p>The code above will fail because <code class="language-plaintext highlighter-rouge">assay(se)</code> returns a matrix, but
<code class="language-plaintext highlighter-rouge">pivot_longer()</code> expects a <code class="language-plaintext highlighter-rouge">data.frame</code>, as mentioned above. We
thus need to convert the matrix as shown below:</p>

    <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pivot_longer</span><span class="p">(</span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">assay</span><span class="p">(</span><span class="n">se</span><span class="p">)),</span><span class="w">
             </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"sample"</span><span class="p">,</span><span class="w">
             </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"expression"</span><span class="p">,</span><span class="w">
             </span><span class="n">everything</span><span class="p">())</span><span class="w">
</span></code></pre></div>    </div>

    <p>This will generate something quite close to what we need, a table
with the <code class="language-plaintext highlighter-rouge">expression</code> and <code class="language-plaintext highlighter-rouge">sample</code> columns. We are still missing
the <code class="language-plaintext highlighter-rouge">infection</code> status.</p>
  </li>
  <li>
    <p>We know from step 3 that the infection status is available from the
<code class="language-plaintext highlighter-rouge">colData</code>, that we need to merge to our long two-column table from
step 4. This can be done with <code class="language-plaintext highlighter-rouge">full_join()</code>. Below, I use the pipe
operator to continue the chunk written above.</p>

    <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pivot_longer</span><span class="p">(</span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">assay</span><span class="p">(</span><span class="n">se</span><span class="p">)),</span><span class="w">
             </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"sample"</span><span class="p">,</span><span class="w">
             </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"expression"</span><span class="p">,</span><span class="w">
             </span><span class="n">everything</span><span class="p">())</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">full_join</span><span class="p">(</span><span class="n">colData</span><span class="p">(</span><span class="n">se</span><span class="p">))</span><span class="w">
</span></code></pre></div>    </div>

    <p>There are two errors in the code above. First, we need to convert
the <code class="language-plaintext highlighter-rouge">DataFrame</code> returned by <code class="language-plaintext highlighter-rouge">colData(se)</code> to a <code class="language-plaintext highlighter-rouge">data.frame</code> (or
<code class="language-plaintext highlighter-rouge">tibble</code>). This is easy. The second (likely) error lies in the
variable names used to join the tables (<code class="language-plaintext highlighter-rouge">full_join()</code>’s <code class="language-plaintext highlighter-rouge">by</code>
argument). If we have a <code class="language-plaintext highlighter-rouge">sample</code> variable in the <code class="language-plaintext highlighter-rouge">colData</code> that
contains sample names, the join will succeed. Otherwise, we will
need to defined the vector of variables to join by using <code class="language-plaintext highlighter-rouge">by</code>. We
can also create that variable on the fly using
<code class="language-plaintext highlighter-rouge">tibble::rownames_to_column()</code> knowing that the rownames of
<code class="language-plaintext highlighter-rouge">colData</code> match the sample names (by definition), and set it to
<code class="language-plaintext highlighter-rouge">sample</code>.</p>

    <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pivot_longer</span><span class="p">(</span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">assay</span><span class="p">(</span><span class="n">se</span><span class="p">)),</span><span class="w">
             </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"sample"</span><span class="p">,</span><span class="w">
             </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"expression"</span><span class="p">,</span><span class="w">
             </span><span class="n">everything</span><span class="p">())</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">full_join</span><span class="p">(</span><span class="n">rownames_to_column</span><span class="p">(</span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">colData</span><span class="p">(</span><span class="n">se</span><span class="p">)),</span><span class="w">
                                 </span><span class="n">var</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"sample"</span><span class="p">))</span><span class="w">
</span></code></pre></div>    </div>
  </li>
</ol>

<h2 id="bringing-it-all-together">Bringing it all together</h2>

<p>In the previous section, I already wrote syntactically correct code,
ironing out smaller issues such as conversion to <code class="language-plaintext highlighter-rouge">data.frame</code> and
variable matching. I haven’t run the code, but I’m quite confident it
should work with my <code class="language-plaintext highlighter-rouge">se</code> object, as long as the infection status is
encoded as <code class="language-plaintext highlighter-rouge">infection</code> in the <code class="language-plaintext highlighter-rouge">colData</code>. A conceptual solution without
these details would already earn students a very decent grade. The
effort to produce the actual figure and earn full marks is relatively
minor, once the path to the answer is mapped out. Referring back to
the travelling by bus illustration above, this conceptual solution is
similar to having the bus names, their respective stop, and possibly
even times mapped out. All that is left is to execute the actual
travel plan.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pivot_longer</span><span class="p">(</span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">assay</span><span class="p">(</span><span class="n">se</span><span class="p">)),</span><span class="w">
             </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"sample"</span><span class="p">,</span><span class="w">
             </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"expression"</span><span class="p">,</span><span class="w">
             </span><span class="n">everything</span><span class="p">())</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">full_join</span><span class="p">(</span><span class="n">rownames_to_column</span><span class="p">(</span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">colData</span><span class="p">(</span><span class="n">se</span><span class="p">)),</span><span class="w">
                                 </span><span class="n">var</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"sample"</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">ggplot</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sample</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">expression</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">infection</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">geom_violin</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>

<h2 id="conclusions">Conclusions</h2>

<p>There is of course a much more direct solution, using <code class="language-plaintext highlighter-rouge">boxplot()</code> on
the assay matrix and setting the <code class="language-plaintext highlighter-rouge">col</code> argument with the infection
state vector. This solution is one that student would most likely
implement after following the second year of the bioinformatics
curriculum. In the <a href="http://bit.ly/WSBIM1207">first year</a>, we start
with general data manipulation and visualisation using the
tidyverse. Omics data and <code class="language-plaintext highlighter-rouge">SummarizedExperiment</code> only come later, as a
preparation for the <a href="http://bit.ly/WSBIM1322">second</a> and
<a href="http://bit.ly/WSBIM2122">third</a> years. Other examples won’t simplify
that nicely though. But the important fact remains that elaborating a
solution will always reply on a good understanding of the basic data
structure and their manipulation and, using that understanding,
conceptually sketch an initial path to the solution.</p>

<p>While I do emphasise the reasoning and concepts behind the code when
introducing new material and solving exercises in class, I appreciate
that when learning a programming language for the first time, one
tends to focus on the syntax (the interpreter isn’t forgiving, after
all) and it can be difficult to step back and look at the bigger
picture. This is probably something I will need to pay special
attention to from now on.</p>

<h2 id="some-context">Some context</h2>

<p>I came to write this post following several discussions with students
during office hours. These discussions led invariably to the same
conclusion: those that struggle in class and during the exams struggle
to conceptualise the course material. Despite honest efforts, every
exercise comes across as a new beast to tame - they fail to visualise
the common patterns and start from scratch at every attempt, which is
not only very inefficient, but also leads to considerable stress,
which itself amplifies perceived difficulty of the task.</p>

        
      </section>

      <footer class="page__meta">
        
        
  


  
  
  

  <p class="page__taxonomy">
    <strong><i class="fa fa-fw fa-tags" aria-hidden="true"></i> Tags: </strong>
    <span itemprop="keywords">
    
      
      
      <a href="https://lgatto.github.io/tags/#education" class="page__taxonomy-item" rel="tag">education</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#r" class="page__taxonomy-item" rel="tag">R</a><span class="sep">, </span>
    
      
      
      <a href="https://lgatto.github.io/tags/#teaching" class="page__taxonomy-item" rel="tag">teaching</a>
    
    </span>
  </p>




        
          <p class="page__date"><strong><i class="fa fa-fw fa-calendar" aria-hidden="true"></i> Updated:</strong> <time datetime="2021-09-23T00:00:00+02:00">September 23, 2021</time></p>
        
      </footer>

      

      
  <nav class="pagination">
    
      <a href="https://lgatto.github.io/circle-u/" class="pagination--pager" title="Open and reproducible management and dissemination of knowledge
">Previous</a>
    
    
      <a href="https://lgatto.github.io/rdmhk/" class="pagination--pager" title="How to Manage Data: Data Stewardship and FAIR Skills
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
    <div class="page__related">
      
        <h4 class="page__related-title">You May Also Enjoy</h4>
      
      <div class="grid__wrapper">
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/ia-irss/" rel="permalink">On cats, farts and parrots.
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  12 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">I was invited to contribute to a
seminar/discussion
on AI, and using language models in research. These are the notes I
prepared my short presentation. I am ...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/oublions-les-notes/" rel="permalink">Podcasts du LLL: Et si on oubliait les notes?
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  less than 1 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">Deuxième billet en français référençant un podcast du Louvain
Learning Lab (le
premier est ici). Cette
fois, je fais un peu d’auto-promotion:

</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/ecr-recognition/" rel="permalink">HUPO ECR Online Panel Discussion - Getting recognised for your work
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  8 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">The HUPO Early Career Researcher (ECR) committee has organised a
discussion panel on Getting recognised for your work and have asked
me to participate - than...</p>
  </article>
</div>
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="http://schema.org/CreativeWork">
    
      <div class="archive__item-teaser">
        <img src=
          
            "https://lgatto.github.io/assets/images/slide-code.png"
          
          alt="">
      </div>
    
    <h2 class="archive__item-title" itemprop="headline">
      
        <a href="https://lgatto.github.io/the-grid-in-r/" rel="permalink">The Grid poster, in R
</a>
      
    </h2>
    
      <p class="page__meta"><i class="fa fa-clock-o" aria-hidden="true"></i> 




  3 minute read
</p>
    
    <p class="archive__item-excerpt" itemprop="description">The MuseeL is the
UCLouvain University museum in
Louvain-la-Neuve. Highly recommended. It’s located to the lively
place des sciences, in a nice brutalist sty...</p>
  </article>
</div>
        
      </div>
    </div>
  
</div>

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
      <li><strong>Follow:</strong></li>
    
    
    
    
      <li><a href="http://github.com/lgatto"><i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li>
    
    
    <li><a href="https://lgatto.github.io/feed.xml"><i class="fa fa-fw fa-rss-square" aria-hidden="true"></i> Feed</a></li>
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2025 Laurent Gatto CC-BY. Powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    <script src="https://lgatto.github.io/assets/js/main.min.js"></script>





  </body>
</html>
