<!doctype html>
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin SEO -->









<title>            About the author      ~/      </title>




<meta name="description" content="Laurent Gatto’s online home">




<meta property="og:locale" content="en_GB">
<meta property="og:site_name" content="~/">
<meta property="og:title" content="About the author">


  <link rel="canonical" href="https://lgatto.github.io/about/">
  <meta property="og:url" content="https://lgatto.github.io/about/">







  

  



  <meta property="og:image" content="https://lgatto.github.io/images/slide-code2.png">










  <script type="application/ld+json">
    {
      "@context" : "http://schema.org",
      "@type" : "Person",
      "name" : "Laurent Gatto",
      "url" : "https://lgatto.github.io",
      "sameAs" : null
    }
  </script>






<!-- end SEO -->


<link href="https://lgatto.github.io/feed.xml" type="application/atom+xml" rel="alternate" title="~/ Feed">

<!-- http://t.co/dKP3o1e -->
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="https://lgatto.github.io/assets/css/main.css">

<meta http-equiv="cleartype" content="on">
    <!-- start custom head snippets -->

<!-- insert favicons. use http://realfavicongenerator.net/ -->

<!-- end custom head snippets -->
  </head>

  <body class="layout--single">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->
    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <button><div class="navicon"></div></button>
        <ul class="visible-links">
          <li class="masthead__menu-item masthead__menu-item--lg"><a href="https://lgatto.github.io/">~/</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/about/">About</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/year-archive/">Posts</a></li>
          
            
            <li class="masthead__menu-item"><a href="https://lgatto.github.io/cbio-lab/">Lab</a></li>
          
        </ul>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>

    
  







<div class="page__hero--overlay"
  style=" background-image: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url('https://lgatto.github.io/images/slide-code2.png');"
>
  
    <div class="wrapper">
      <h1 class="page__title" itemprop="headline">
        
          About the author

        
      </h1>
      
      
      
    </div>
  
  
</div>




<div id="main" role="main">
  
  <div class="sidebar sticky">
  

<div itemscope itemtype="http://schema.org/Person">

  
    <div class="author__avatar">
      
        <img src="https://lgatto.github.io/images/lgatto2.jpg" class="author__avatar" alt="Laurent Gatto" itemprop="image">
      
    </div>
  

  <div class="author__content">
    <h3 class="author__name" itemprop="name">Laurent Gatto</h3>
    
      <p class="author__bio" itemprop="description">
        Open science, reproducible research, data champion, computational biology, proteomics, more omics, emacs, a lot of R, quite a bit of running, and parenting.
      </p>
    
  </div>

  <div class="author__urls-wrapper">
    <button class="btn btn--inverse">Follow</button>
    <ul class="author__urls social-icons">
      

      

      

      
        <li>
          <a href="https://keybase.io/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-key" aria-hidden="true"></i> Keybase
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      
        <li>
          <a href="https://github.com/lgatto" itemprop="sameAs">
            <i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub
          </a>
        </li>
      

      

      

      

      

      

      

      

      

      

      

      

      

      <!--
  <li>
    <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs">
      <i class="fa fa-fw" aria-hidden="true"></i> Custom Social Profile Link
    </a>
  </li>
-->


  <li>
    <a href="http://dissem.in/r/6231/laurent-gatto" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> dissem.in
    </a>
  </li>

  <li>
    <a href="https://orcid.org/0000-0002-1520-2268" itemprop="sameAs">
      <i class="fa fa-fw fa-pencil" aria-hidden="true"></i> ORCID
    </a>
  </li>

  <li>
    <a href="https://scholar.google.co.uk/citations?user=k5DrB74AAAAJ&hl=en" itemprop="sameAs">
      <i class="fa fa-fw fa-google" aria-hidden="true"></i> scholar
    </a>
  </li>

  <!-- <li> -->
  <!--   <a href="https://impactstory.org/u/0000-0002-1520-2268" itemprop="sameAs"> -->
  <!--     <i class="fa fa-fw fa-sun-o" aria-hidden="true"></i> Impactstory -->
  <!--   </a> -->
  <!-- </li> -->

    </ul>
  </div>
</div>

  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="About the author">
    
    
    <meta itemprop="dateModified" content="November 05, 2022">

    <div class="page__inner-wrap">
      

      <section class="page__content" itemprop="text">
        <aside class="sidebar__right">
<nav class="toc">
    <header><h4 class="nav__title"><i class="fa fa-file-text"></i> On This Page</h4></header>
<ul class="toc__menu" id="markdown-toc">
  <li><a href="#short-bio" id="markdown-toc-short-bio">Short bio</a></li>
  <li><a href="#research" id="markdown-toc-research">Research</a>    <ul>
      <li><a href="#mass-spectrometry-and-proteomics" id="markdown-toc-mass-spectrometry-and-proteomics">Mass spectrometry and proteomics</a></li>
      <li><a href="#single-cell-proteomics" id="markdown-toc-single-cell-proteomics">Single-cell proteomics</a></li>
      <li><a href="#spatial-proteomics" id="markdown-toc-spatial-proteomics">Spatial proteomics</a></li>
    </ul>
  </li>
  <li><a href="#teaching" id="markdown-toc-teaching">Teaching</a></li>
  <li><a href="#publications" id="markdown-toc-publications">Publications</a>    <ul>
      <li><a href="#research-articles" id="markdown-toc-research-articles">Research articles</a></li>
      <li><a href="#book-chapters" id="markdown-toc-book-chapters">Book chapters</a></li>
      <li><a href="#technical-notes" id="markdown-toc-technical-notes">Technical Notes</a></li>
    </ul>
  </li>
  <li><a href="#webpress-coverage" id="markdown-toc-webpress-coverage">Web/press coverage</a></li>
  <li><a href="#software" id="markdown-toc-software">Software</a></li>
  <li><a href="#talks" id="markdown-toc-talks">Talks</a></li>
  <li><a href="#get-in-touch" id="markdown-toc-get-in-touch">Get in touch</a></li>
</ul>

  </nav>
</aside>

<h2 id="short-bio">Short bio</h2>

<p>I am an Full Professor of Bioinformatics at the
<a href="https://uclouvain.be/en/index.html">UCLouvain</a>, in Belgium, and
director of the Computational Biology and Bioinformatics (CBIO)
group. I am located in the <a href="http://www.deduveinstitute.be/">de Duve
Institute</a>, on the medical campus in
Brussels, where I run a research group and teach at the faculty of
pharmacy and biomedical sciences (FASB).</p>

<p>I am an avid open research advocate and make every possible effort to
make my research reproducible and openly available. I am a <a href="https://www.software.ac.uk/fellowship-programme">Software
Sustainability Institute
fellow</a>, a
<a href="http://www.datacarpentry.org/">Data</a> and <a href="https://software-carpentry.org/team/">Software
Carpentry</a> instructor, and an
affiliated <a href="http://bioconductor.org/about/core-team/">member</a> of the
<a href="http://bioconductor.org">Bioconductor</a> project (see below).</p>

<p>I am also involved in the <a href="http://bulliedintobadscience.org/">Bullied Into Bad
Science</a> campaign, an initiative by
and for early career researchers who aim for a fairer, more open and
ethical research and publication environment. In 2017-2018, I was also
part of the eLife <a href="https://elifesciences.org/about/people/early-career">Early-career advisory
group</a> and, since
2017, am a <a href="http://asapbio.org/">#ASAPbio</a> ambassador.</p>

<p><a href="http://bioconductor.org">Bioconductor</a> (<a href="http://www.ncbi.nlm.nih.gov/pubmed/25633503">Huber et al.,
2015</a>) is a project that
has had a great impact on many aspects of research and development in
bioinformatics, computational statistics, bio-medical sciences and
many other fields. It also had a big impact on my vision of rigorous
research software development as the foundation of reliable, open and
reproducible research. Since my very first software contribution (the
<a href="https://bioconductor.org/packages/2.2/bioc/html/yaqcaffy.html">yaqcaffy</a>
package, accepted in Bioconductor 2.2, since then deprecated), I have
become involved in multiple capacity in the project. I have
contributed many packages (mostly focused on mass spectrometry and
proteomics), I am a member of the <a href="https://bioconductor.org/about/technical-advisory-board/">technical advisory
board</a>,
deputy-chair of the <a href="https://bioconductor.org/about/european-bioconductor-society/">European Bioconductor
Society</a>,
I co-lead the <a href="https://bioconductor.org/help/education-training/">Teaching and education
committee</a>, I’m a
member of the <a href="https://bioconductor.org/about/code-of-conduct/">Code of
conduct</a> committee
since it was set up, I contribute to various <a href="http://workinggroups.bioconductor.org/">working
groups</a>, and am an occasional
<a href="https://bioconductor.org/about/package-reviewers/">package
reviewer</a>. In am
one of the winners of the 2021 <a href="https://bioconductor.org/about/awards/">Bioconductor Community
Awards</a>, one of my proudest
professional achievements. In addition to delivering great science and
software, Bioconductor is a <a href="https://bioconductor.org/about/community-advisory-board/">welcoming and supportive
community</a>.</p>

<p>Before joining UCLouvain in September 2018, I was a Senior Research
Associate in the Department of Biochemistry at the University of
Cambridge. It’s at the <a href="http://proteomics.bio.cam.ac.uk/">Cambridge Centre for
Proteomics</a> and the Computational
Proteomics Unit (my former research group in Cambridge) that I started
working on various aspects of quantitative and spatial proteomics,
developing new methods and implementing computational tools with a
strong emphasis on rigorous and reproducible data analysis. During my
time in Cambridge, I have also a been a visiting scientist in the
<a href="http://www.ebi.ac.uk/pride/">PRIDE</a> team at the <a href="http://www.ebi.ac.uk/pride/">European
Bioinformatics Institute</a> and in the
<a href="http://www.damtp.cam.ac.uk/">Department of Applied Mathematics and Theoretical Physics
(DAMTP)</a>.</p>

<p>And even before that, during my MSc and <a href="http://difusion.ulb.ac.be/vufind/Record/ULB-DIPOT:oai:dipot.ulb.ac.be:2013/210855/Holdings">PhD
work</a>,
I studied micro-evolutionary genetic patterns of the Broom leaf beetle
<a href="http://www.biol.uni.wroc.pl/cassidae/European%20Chrysomelidae/gonioctena%20variabilis.htm"><em>Gonioctena</em></a>
<a href="http://davesgarden.com/guides/bf/go/6185/"><em>variabilis</em></a> in Southern
Europe (<a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1095-8312.2008.00972.x/abstract">Gatto el al.,
2008</a>),
the application of short interspersed mobile elements (SINEs) to study
the evolution of cetaceans, and the applicability of the General Time
Reversible nucleotide substitution model in the light of differential
lineage sorting (<a href="http://www.ncbi.nlm.nih.gov/pubmed/19455208">Gatto el al.,
2006</a>). I also spend 3
years in industry working on genomic and transcriptomics data, in
particular the microarray quality control (<a href="http://www.ncbi.nlm.nih.gov/pubmed/20676074">Shi et al.,
2010</a>).</p>

<h2 id="research">Research</h2>

<p>As pointed out by D. Donoho, <em>An article about computational science
in a scientific publication is not the scholarship itself, it is
merely advertising of the scholarship. The actual scholarship is the
complete software development environment and the complete set of
instructions which generated the figures</em>. This directly applies to
high throughput biology data analysis and I strongly believe that
being able to reproduce the complete set of results, replicate an
analysis with new data and track the evolution of the work that lead
to the scientific novelty are essential aspects of the process of doing
research. Hence, I regard the development of <strong>scientific software</strong>
as well as agile and robust analysis methodologies that facilitate
reproducible research, an important aspect of my scientific activity.</p>

<p>Clarity and traceability of the data and the analysis methodology
enable us to better understand what we do, how and why we do it and
consequently exploit the data and comprehend the biology. While not
sufficient, these are nevertheless necessary requirements for
effective <strong>data-driven science</strong>.</p>

<p>The <strong>collaborative</strong> and <strong>interdisciplinarity</strong> nature of much of
the research in biology calls for a open approaches, influenced by the
<strong>open source</strong> and <strong>free/libre</strong> software movements, from communication
between stake holders, <strong>open research</strong> and development to open
dissemination of all research outputs.</p>

<h3 id="mass-spectrometry-and-proteomics">Mass spectrometry and proteomics</h3>

<p>My work on the design and implementation of reproducible mass
spectrometry-based proteomics data analysis pipeline has materialised
in the development of the
<a href="http://bioconductor.org/packages/MSnbase"><code class="language-plaintext highlighter-rouge">MSnbase</code></a> (<a href="http://www.ncbi.nlm.nih.gov/pubmed/22113085">Gatto et al.
2012</a> and
<a href="https://pubs.acs.org/doi/10.1021/acs.jproteome.0c00313">2021</a>)
package to manipulate, process and analyse quantitative proteomics
data. The <code class="language-plaintext highlighter-rouge">MSnbase</code> infrastructure also supports the work on the
statistical learning applied to spatial proteomics (see below). After
a <a href="https://lgatto.github.io/msnbase-contribs-2/">decade of <code class="language-plaintext highlighter-rouge">MSnbase</code>
development</a>, Johannes
Rainer, Sebastien Gibb and I have started the <a href="https://www.rformassspectrometry.org/">R for Mass Spectrometry
initiative</a>. Base on our
decade-long experiences, we have set ourselves the aim of designing
and implementing efficient, thoroughly documented, tested and flexible
R software for the analysis and interpretation of high throughput mass
spectrometry assays, including proteomics and metabolomics
experiments.</p>

<h3 id="single-cell-proteomics">Single-cell proteomics</h3>

<p>In a Nature Methods Technology Feature in August 2019, Vivien Marx
<a href="https://doi.org/10.1038/s41592-019-0540-6">dreamt of single-cell
proteomics</a>. In 2021,
thanks to some pioneers’ groundbreaking efforts, we can confidently
claim that single-cell proteomics is here. The CBIO lab has
contributed the <a href="https://bioconductor.org/packages/scp"><code class="language-plaintext highlighter-rouge">scp</code></a> package
(<a href="https://www.tandfonline.com/doi/abs/10.1080/14789450.2021.1988571?journalCode=ieru20">Vanderaa and Gatto,
2021</a>
and
<a href="https://www.biorxiv.org/content/10.1101/2021.04.12.439408v2">pre-print</a>,
<a href="https://doi.org/10.1002/cpz1.658">2023a</a> and
<a href="https://arxiv.org/abs/2210.01020">pre-print</a>,
<a href="https://pubs.acs.org/doi/10.1021/acs.jproteome.3c00227">2023b</a> and
<a href="https://arxiv.org/abs/2304.06654">pre-print</a>) to these efforts. <code class="language-plaintext highlighter-rouge">scp</code>
relies of <a href="https://www.rformassspectrometry.org/">R for Mass
Spectrometry</a>’s
<a href="https://bioconductor.org/packages/QFeatures"><code class="language-plaintext highlighter-rouge">QFeatures</code></a> package and
dedicated functions to enable extensive single-cell proteomics data
processing, analysis and interpretation. In 2023, we also contributed
to initial recommendations for performing, benchmarking and reporting
single-cell proteomics experiments (<a href="https://www.nature.com/articles/s41592-023-01785-3">Gatto et
al. 2023</a> and
<a href="https://arxiv.org/abs/2210.01020">pre-print</a>). Our latest
<a href="https://doi.org/10.1101/2023.12.14.571792">work</a> proposes the
<em>scplainer</em> method, based linear models to understand mass
spectrometry-based single-cell proteomics data, that tackles many of
the issues described above.</p>

<h3 id="spatial-proteomics">Spatial proteomics</h3>

<p>In biology, localisation is function: knowledge of the localisation of
proteins is of paramount importance to assess and study their
function, and spatial proteomics is the systematic study of the
sub-cellular localisation of proteins and changes thereof (<a href="http://www.ncbi.nlm.nih.gov/pubmed/21080489">Gatto et
al. 2010</a> and
<a href="http://www.mcponline.org/content/13/8/1937.long">2014</a>). Since 2010,
my team has developed novel software and machine learning approaches
enabling more reliable and systematic inference of protein
localisations using quantitative proteomics. This work has
materialised in the
<a href="http://bioconductor.org/packages/pRoloc"><code class="language-plaintext highlighter-rouge">pRoloc</code></a> package (<a href="http://www.ncbi.nlm.nih.gov/pubmed/24413670">Gatto et
al. 2014</a>) that
implements various established classification algorithms, effective
visualisation techniques (<a href="http://www.ncbi.nlm.nih.gov/pubmed/25690415">Gatto el al.
2015</a>) as well as novelty
detection (<a href="http://www.ncbi.nlm.nih.gov/pubmed/23523639">Breckels et al.
2013</a>, <a href="https://doi.org/10.1371/journal.pcbi.1008288">Crook et al.
2020</a>) and transfer
learning (<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004920">Breckels et al.
2016</a>),
harvesting GO annotations of microscopy-based methods to improve the
spatial resolution of experimental spatial proteomics data. More
recently, we proposed a Bayesian framework to quantify uncertainty and
probabilisatically address multi-localsiation of proteins (<a href="https://doi.org/10.1371/journal.pcbi.1006516">Crook et
al. 2018</a>) and study
comparative spatial proteomics to infer protein re-localisation
(<a href="https://doi.org/10.1038/s41467-022-33570-9">Crook et al. 2022</a>).</p>

<h2 id="teaching">Teaching</h2>

<p>My current course at the UCLouvain are</p>

<ul>
  <li><a href="http://bit.ly/WSBIM1207">Introduction to bioinformatics</a> (since 2018)</li>
  <li><a href="https://uclouvain.be/cours-2023-wmds1114">Eléments de statistiques médicales</a> (since 2018)</li>
  <li><a href="http://bit.ly/WSBIM1322">Bioinformatics</a> (since 2019)</li>
  <li><a href="http://bit.ly/WSBIM2122">Omics data analysis</a> (since 2019)</li>
</ul>

<p>Over the years I have been involved in many teaching activities,
ranging from beginners and advanced R courses,
<a href="http://www.huber.embl.de/csama/">genome biology</a>,
<a href="http://www.wellcome.ac.uk/Education-resources/Courses-and-conferences/Advanced-Courses-and-Scientific-Conferences/Workshops/WTX054153.htm">proteomics bioinformatics</a>,
<a href="http://www.ebi.ac.uk/training/course/introduction-integrative-omics">integrative omics</a>,
scientific computing as part of the
<a href="http://www.ccbi.cam.ac.uk/Education/MPhil/">MPhil in Computational Biology</a>
in Cambridge, as well as several Software and Data Carpentry
bootcamps. All my teaching material is available in the
<a href="http://lgatto.github.io/TeachingMaterial/"><strong>TeachingMaterial</strong></a>
repository.</p>

<p>Please do get in touch if you are interested in running workshops.</p>

<h2 id="publications">Publications</h2>

<blockquote>
  <p>See also my <a href="https://scholar.google.co.uk/citations?user=k5DrB74AAAAJ&amp;hl=en">Google scholar
profile</a>
and publications in
<a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=gatto+laurent">PubMed</a>.</p>
</blockquote>

<h3 id="research-articles">Research articles</h3>

<p>Kohrs FE, Kazezian V, Bagley R, Boisgontier MP, Brod S, Carneiro CFD,
…, Weissgerber TL . <em>Collaborating with early career researchers to
enhance the future of scholarly communication: A guide for
publishers</em>. (2024)
<a href="https://doi.org/10.31219/osf.io/pbvfz">https://doi.org/10.31219/osf.io/pbvfz</a>.</p>

<p>Jenny, Drnevich, J. Tan Frederick, Almeida-Silva Fabricio, Castelo
Robert, C. Culhane Aedin, Davis Sean, A. Doyle Maria, et
al. . <em>Learning and Teaching Biological Data Science in the
Bioconductor Community</em> (2024) arXiv [Cs.CY]
doi:<a href="http://arxiv.org/abs/2410.01351">arXiv:2410.01351</a>.</p>

<p>Pirenne S, Manzano-Núñez F, Loriot A, Cordi S, Desmet L, Aydin S,
Hubert C, Toffoli S, Limaye N, Sempoux C, Komuta M, Gatto L, Lemaigre
FP. <em>Spatial transcriptomics profiling of gallbladder adenocarcinoma:
a detailed two-case study of progression from precursor lesions to
cancer</em> BMC Cancer 24, 1025 (2024)
doi:<a href="https://doi.org/10.1186/s12885-024-12770-0">10.1186/s12885-024-12770-0</a>
(pre-print:
<a href="https://doi.org/10.1101/2024.02.27.582232">10.1101/2024.02.27.582232</a>).</p>

<p>Ward B et al. <em>Deep Plasma Proteomics with Data-Independent
Acquisition: Clinical Study Protocol Optimization with a COVID-19
Cohort</em>. (2024) J. Proteome Res. 23, 9, 3806–3822
doi:<a href="https://pubs.acs.org/doi/10.1021/acs.jproteome.4c00104">10.1021/acs.jproteome.4c00104</a>
(pre-print
<a href="https://doi.org/10.1101/2024.02.23.581160">10.1101/2024.02.23.581160</a>).</p>

<p>Hauchamps P, Delandre S, Temmerman ST, Lin D, Gatto L. <em>Visual quality
control with CytoMDS, a Bioconductor package for low dimensional
representation of cytometry sample distances</em> bioRxiv
2024.07.01.601465; doi:
<a href="https://doi.org/10.1101/2024.07.01.601465">10.1101/2024.07.01.601465</a>.</p>

<p>Lynch MP, Wang Y, Gatto L, Culhane AC. <em>demuxSNP: supervised
demultiplexing scRNAseq using cell hashing and SNPs</em>. bioRxiv
2024.04.22.590526;
doi:<a href="https://doi.org/10.1101/2024.04.22.590526">10.1101/2024.04.22.590526</a>.</p>

<p>Hauchamps P, Bayat B, Delandre S, Hamrouni M, Toussaint M, Temmerman
S, Lin D, Gatto L <em>CytoPipeline and CytoPipelineGUI: a Bioconductor R
package suite for building and visualizing automated pre-processing
pipelines for flow cytometry data</em> BMC Bioinformatics 25, 80
(2024). doi:<a href="https://doi.org/10.1186/s12859-024-05691-z">10.1186/s12859-024-05691-z</a>
(pre-print:
<a href="https://doi.org/10.1101/2023.10.10.561699">10.1101/2023.10.10.561699</a>).</p>

<p>Lizcano-Perret B, Vertommen Didier, Herinckx G, Calabrese V, Gatto L,
Roux P and Michiels T <em>Identification of RSK substrates using an
analog-sensitive kinase approach</em> (2024)
J. Biol. Chem. doi:<a href="https://doi.org/10.1016/j.jbc.2024.105739">10.1016/j.jbc.2024.105739</a>.</p>

<p>Vanderaa C and Gatto L <em>scplainer: using linear models to understand
mass spectrometry-based single-cell proteomics data</em> (2023) bioRxiv
2023.12.14.571792;
doi:<a href="https://doi.org/10.1101/2023.12.14.571792">10.1101/2023.12.14.571792</a>.</p>

<p>Harvengt AA, Polle OG, Martin M, van Maanen A, Gatto L, et al.
<em>Post-Hypoglycemic hyperglycemia are highly relevant markers for
stratification of glycemic variability and partial remission status of
pediatric patients with new-onset type 1 diabetes</em> (2023) PLOS ONE
18(11): e0294982
doi:<a href="https://doi.org/10.1371/journal.pone.0294982">10.1371/journal.pone.0294982</a>.</p>

<p>Cloos AS, Pollet H, Stommen A, Maja M, Lingurski M, Brichard B,
Lambert C, Henriet P, Pierreux C, Pyr Dit Ruys S, Van Der Smissen P,
Vikkula M, Gatto L, Martin M, Brouillard P, Vertommen D, Tyteca
D. <em>Splenectomy improves erythrocyte functionality in spherocytosis
based on septin abundance, but not maturation defects</em>. Blood
Adv. 2023 Sep
12;7(17):4705-4720. doi:<a href="https://doi.org/10.1182/bloodadvances.2022009114">10.1182/bloodadvances.2022009114</a>.</p>

<p>Kelly Eckenrode, Dario Righelli, Marcel Ramos, Ricard Argelaguet,
Christophe Vanderaa, Ludwig Geistlinger, Aedin Culhane, Laurent Gatto,
Vincent J. Carey, Martin Morgan, Davide Risso, Levi D Waldron <em>Curated
single cell multimodal landmark datasets for R/Bioconductor</em> PLOS
Computational Biology 19(8):
e1011324. doi:<a href="https://doi.org/10.1371/journal.pcbi.1011324">10.1371/journal.pcbi.1011324</a>
(preprint
bioRxiv:<a href="https://doi.org/10.1101/2021.10.27.466079">10.1101/2021.10.27.466079</a>).</p>

<p>Gannoun L, De Schrevel C, Belle M, Dauguet N, Achouri Y, Loriot A,
Vanderaa C, Cordi S, Dili A, Heremans Y, Rooman I, Leclercq IA,
Jacquemin P, Gatto L, Lemaigre FP. <em>Axon guidance genes control
hepatic artery development</em> Development. 2023 Aug
15;150(16):dev201642. doi:<a href="https://doi.org/10.1242/dev.201642">10.1242/dev.201642</a>. Epub
2023 Aug 17.</p>

<p>Vanderaa C and Gatto L. <em>Revisiting the Thorny Issue of Missing Values
in Single-Cell Proteomics</em> J. Proteome Res. (2023)
<a href="https://pubs.acs.org/doi/10.1021/acs.jproteome.3c00227">doi:10.1021/acs.jproteome.3c00227</a>
(preprint <a href="https://arxiv.org/abs/2304.06654">arXiv:2304.06654v2</a>).</p>

<p>Halbout M, Bury M, Nanet A, Gerin I, Graff J, Killian T, Gatto L,
Vertommen D, Bommer G. <em>SUZ domain-containing proteins have multiple
effects on nonsense-mediated decay target transcripts</em> (2023) Journal
of Biological Chemistry, July 2023;
<a href="https://doi.org/10.1016/j.jbc.2023.105095">doi:10.1016/j.jbc.2023.105095</a>.</p>

<p>Degraeve AL, Haufroid V, Loriot A, et al. <em>Gut microbiome modulates
tacrolimus pharmacokinetics through the transcriptional regulation of
ABCB1</em>. Microbiome 11, 138 (2023)
<a href="https://doi.org/10.1186/s40168-023-01578-y">doi:10.1186/s40168-023-01578-y</a>.</p>

<p>Jannone G, Bonaccorsi Riani E, de Magnée C, Tambucci R, Evraerts J,
Ravau J, Baldin P, et al. (2023). <em>Senescence and Senotherapies in
Biliary Atresia and Biliary Cirrhosis</em> Aging 15 (11): 4576–99
<a href="https://doi.org/10.18632/aging.204700">10.18632/aging.204700</a>.</p>

<p>Zhu J, Naulaerts S, Boudhan L, Martin M, Gatto L, and Van den Eynde
BJ. <em>Tumour Immune Rejection Triggered by Activation of
alpha2-Adrenergic Receptors</em> (2023) Nature, June, 1-9;
<a href="https://doi.org/10.1038/s41586-023-06110-8">doi:10.1038/s41586-023-06110-8</a>.</p>

<p>Colson A, Depoix CL, Lambert I, Leducq C, Bedin M, De Beukelaer M,
Gatto L, Loriot A, Peers de Nieuwburgh M, Bouhna K, Baldin P, Hubinont
C, Sonveaux P, Debiève F <em>Specific HIF-2α (Hypoxia-Inducible Factor-2)
Inhibitor PT2385 Mitigates Placental Dysfunction In Vitro and in a Rat
Model of Preeclampsia (RUPP)</em> (2023) Hypertension, 80:1011–1023
<a href="https://doi.org/10.1161/hypertensionaha.122.20739">doi:10.1161/HYPERTENSIONAHA.122.20739</a>.</p>

<p>Gatto L, Aebersold R, Cox J <em>et al.</em>, <em>Initial recommendations for
performing, benchmarking, and reporting single-cell proteomics
experiments</em> (2023) Nat Methods 20, 375–386
<a href="https://doi.org/10.1038/s41592-023-01785-3">doi:10.1038/s41592-023-01785-3</a>
(preprint: <a href="https://arxiv.org/abs/2207.10815">arXiv:2207.10815</a>).</p>

<p>Vanderaa C and Gatto L <em>The current state of single-cell proteomics
data analysis</em> (2023), Current Protocols 3 (1):e658,
<a href="https://doi.org/10.1002/cpz1.658">10.1002/cpz1.658</a> (preprint:
<a href="https://arxiv.org/abs/2210.01020">arXiv:2210.01020</a>).</p>

<p>Crook OM, Davies CTR, Breckels LM, Christopher JA, Gatto L, Kirk PDW,
Lilley KS <em>Inferring differential subcellular localisation in
comparative spatial proteomics using BANDLE</em>, Nature Communications
13, 5948 (2022)
<a href="https://doi.org/10.1038/s41467-022-33570-9">doi:10.1038/s41467-022-33570-9</a>
(Pre-print: <a href="https://doi.org/10.1101/2021.01.04.425239">biorXiv</a>).</p>

<p>Lizcano-Perret B., Lardinois C., Wavreil F., Hauchamps P., Herinckx
G., Sorgeloos F., Vertommen D., Gatto L., Michiels T. <em>Cardiovirus
leader proteins retarget RSK kinases toward alternative substrates to
perturb nucleocytoplasmic traffic</em>. (2022) PLoS Pathogens
18(2):e1011042. <a href="https://doi.org/10.1371/journal.ppat.1011042">doi:10.1371/journal.ppat.1011042</a>.</p>

<p>Pollé O.G, Delfosse A., Martin M., Louis J., Gies I., den Brinker M.,
Seret N., Lebrethon M., Mouraux T., Gatto L., Lysy P.A. on behalf of
the DIATAG Working Group <em>Glycemic Variability Patterns Strongly
Correlate with Partial Remission Status in Children with Newly
Diagnosed Type 1 Diabetes</em>. Diabetes Care 45(10):2360–2368
(2022), doi:<a href="https://doi.org/10.2337/dc21-2543">10.2337/dc21-2543</a>.</p>

<p>Moulis M., Runser S.V.M., Glorieux L. <em>et al.</em> <em>Identification and
implication of tissue-enriched ligands in epithelial–endothelial
crosstalk during pancreas development</em>. Sci Rep 12, 12498 (2022),
doi:<a href="https://doi.org/10.1038/s41598-022-16072-y">10.1038/s41598-022-16072-y</a>.</p>

<p>Ward B, Yombi JC, Balligand J-L, Cani PD, Collet J-F, <em>et al.</em>
<em>HYGIEIA: HYpothesizing the Genesis of Infectious Diseases and
Epidemics through an Integrated Systems Biology
Approach</em>. Viruses. 2022; 14(7):1373,
doi:<a href="https://doi.org/10.3390/v14071373">10.3390/v14071373</a>.</p>

<p>Dewulf JP, Martin M, Marie S, Oguz F, Belkhir L, De Greef J, <em>et
al</em>. <em>Urine metabolomics links dysregulation of the
tryptophan-kynurenine pathway to inflammation and severity of
COVID-19</em>. Sci Rep. 12: 1–8 (2022),
doi:<a href="https://doi.org/10.1038/s41598-022-14292-w">10.1038/s41598-022-14292-w</a>.</p>

<p>De Berdt P, Vanvarenberg K, Ucakar B, Bouzin C, Paquot A, Gratpain V,
Loriot A, Payen V, Bearzatto B, Muccioli GG, Gatto L, Diogenes A, des
Rieux A <em>The human dental apical papilla promotes spinal cord repair
through a paracrine mechanism</em> Cell. Mol. Life Sci. 79, 252
(2022), doi:<a href="https://doi.org/10.1007/s00018-022-04210-8">10.1007/s00018-022-04210-8</a>.</p>

<p>Dontaine J, Bouali A, Daussin F <em>et al.</em> <em>The intra-mitochondrial
O-GlcNAcylation system rapidly modulates OXPHOS function and ROS
release in the heart</em>. Commun Biol 5, 349
(2022), doi:<a href="https://doi.org/10.1038/s42003-022-03282-3">10.1038/s42003-022-03282-3</a>.</p>

<p>Delcorte O, Spourquet C, Lemoine P, Degosserie J, Van Der Smissen P,
Dauguet N, Loriot A, Knauf JA, Gatto L, Marbaix E, Fagin JA, Pierreux
CE. <em>BRAFV600E Induction in Thyrocytes Triggers Important Changes in
the miRNAs Content and the Populations of Extracellular Vesicles
Released in Thyroid Tumor Microenvironment</em>. Biomedicines. 2022;
10(4):755,
doi:<a href="https://www.mdpi.com/2227-9059/10/4/755">10.3390/biomedicines10040755</a>.</p>

<p>Rainer J, Vicini A, Salzer L, Stanstrup J, Badia JM, Neumann S, Stravs
MA, Verri Hernandes V, Gatto L, Gibb S, Witting M. A Modular and
Expandable Ecosystem for Metabolomics Data Annotation in
R. Metabolites. 2022;
12(2):173, doi:<a href="https://doi.org/10.3390/metabo12020173">10.3390/metabo12020173</a>.</p>

<p>Aayush Grover and Laurent Gatto <em>ProtFinder: finding subcellular
locations of proteins using protein interaction networks</em> bioRxiv
2022.01.11.475836,
doi:<a href="https://doi.org/10.1101/2022.01.11.475836">10.1101/2022.01.11.475836</a>
(<a href="https://github.com/UCLouvain-CBIO/ProtFinder">software repo</a>).</p>

<p>Dechamps M, De Poortere, Martin M, Gatto L, et
al. <em>Inflammation-Induced Coagulopathy Substantially Differs Between
COVID-19 and Septic Shock: A Prospective Observational Study</em>
Frontiers in Medicine (2022),
doi:<a href="https://www.frontiersin.org/article/10.3389/fmed.2021.780750">10.3389/fmed.2021.780750</a>.</p>

<p>Christophe Vanderaa, Laurent Gatto <em>Replication of single-cell
proteomics data reveals important computational challenges</em> Expert
Review of Proteomics (2021),
doi:<a href="https://www.tandfonline.com/doi/full/10.1080/14789450.2021.1988571">10.1080/14789450.2021.1988571</a>
(pre-print:
<a href="https://www.biorxiv.org/content/10.1101/2021.04.12.439408v2">10.1101/2021.04.12.439408v2</a>).</p>

<p>Mulvey CM, Breckels LM, Crook O, Sanders D, Ribeiro A, Geladaki A,
Christoforou A, Britovsek NK, Hurrell T, Deery MJ, Gatto L, Smith A,
and Lilley KS. <em>Spatiotemporal proteomic profiling of the
pro-inflammatory response to lipopolysaccharide in the THP-1 human
leukaemia cell line</em> Nat Commun 12, 5773 (2021),
doi:<a href="https://doi.org/10.1038/s41467-021-26000-9">10.1038/s41467-021-26000-9</a>.</p>

<p>Killian T and Gatto L. Exploiting the DepMap cancer dependency data
using the <code class="language-plaintext highlighter-rouge">depmap</code> R package [version 1; peer review: 1 approved with
reservations]. F1000Research 2021, 10:416,
doi:<a href="https://doi.org/10.12688/f1000research.52811.1">10.12688/f1000research.52811.1</a>.</p>

<p>Jérôme Ambroise, Laurent Gatto, Julie Hurel, Bertrand Bearzatto,
Jean-Luc Gala <em>On the many advantages of using the VariantExperiment
class to store, exchange and analyze SARS-CoV-2 genomic data and
associated metadata</em> bioRxiv 2021.04.05.438328, doi:
<a href="https://doi.org/10.1101/2021.04.05.438328">https://doi.org/10.1101/2021.04.05.438328</a>.</p>

<p>Crook OM, Geladaki A, Nightingale DJH, Vennard O, Lilley KS, Gatto L,
Kirk PDW <em>A semi-supervised Bayesian approach for simultaneous protein
sub-cellular localisation assignment and novelty detection</em> PLOS
Computational Biology 2020 16(11):e1008288,
doi:<a href="https://doi.org/10.1371/journal.pcbi.1008288">10.1371/journal.pcbi.1008288</a>
(<a href="https://doi.org/10.1101/2020.05.05.078345">preprint</a>).</p>

<p>Pollet H <em>et al.</em>, Aberrant Membrane Composition and Biophysical
Properties Impair Erythrocyte Morphology and Functionality in
Elliptocytosis, Biomolecules 2020, 10(8), 1120,
doi:<a href="https://www.mdpi.com/2218-273X/10/8/1120">10.3390/biom10081120</a>.</p>

<p>Gatto L, Gibb S, Rainer J <em>MSnbase, efficient and elegant R-based
processing and visualisation of raw mass spectrometry data</em>
J. Proteome Res. 2021, 20, 1, 1063–1069,
doi:<a href="https://doi.org/10.1021/acs.jproteome.0c00313">10.1021/acs.jproteome.0c00313</a>
(<a href="https://doi.org/10.1101/2020.04.29.067868">preprint</a>).</p>

<p>Silverman EK <em>et al.</em>, <em>Molecular networks in Network Medicine:
Development and applications</em>, WIREs Systems Biology and Medecine,
2020, doi:<a href="https://dx.doi.org/10.1002/wsbm.1489">10.1002/wsbm.1489</a>.</p>

<p>Crook PM, Gatto L, Kirk PDW <em>Fast approximate inference for variable
selection in Dirichlet process mixtures, with an application to
pan-cancer proteomics</em> Statistical Applications in Genetics and
Molecular Biology, Volume 18, Issue 6, 1544-6115,
doi:<a href="https://doi.org/10.1515/sagmb-2018-0065">10.1515/sagmb-2018-0065</a>
2019 (pre-print:
<a href="https://arxiv.org/abs/1810.05450">arXiv:1810.05450</a>).</p>

<p>Baers LL, Breckels LM, Mills LA, Gatto L, Deery M, Stevens TJ, Howe
CH, Lilley KS, Lea-Smith DJ <em>Proteome mapping of a cyanobacterium
reveals distinct compartment organisation and cell-dispersed
metabolism</em>, Plant Physiology 2019,
doi:<a href="https://doi.org/10.1104/pp.19.00897">10.1104/pp.19.00897</a>.</p>

<p>Petyuk VA, Gatto L and Payne SH <em>Reproducibility and Transparency by
Design</em>, Molecular &amp; Cellular Proteomics July 4, 2019,
mcp.IP119.001567,
doi:<a href="https://doi.org/10.1074/mcp.IP119.001567">10.1074/mcp.IP119.001567</a>.</p>

<p>Crook OM, Breckels LM, Lilley KS, Kirk PDW and Gatto L (2019) <em>A
Bioconductor workflow for the Bayesian analysis of spatial proteomics</em>
[version 1; peer review: 1 approved, 2 approved with reservations]
F1000Research 2019, 8:446,
doi:<a href="https://doi.org/10.12688/f1000research.18636.1">10.12688/f1000research.18636.1</a>.</p>

<p>Crook OM, Lilley KS, Gatto L, Kirk PDW (2019). <em>Semi-Supervised
Non-Parametric Bayesian Modelling of Spatial Proteomics.</em> arXiv
[stat.AP]. <a href="http://arxiv.org/abs/1903.02909">http://arxiv.org/abs/1903.02909</a>.</p>

<p>Johannes Rainer, Laurent Gatto, Christian X. Weichenberger;
<em><code class="language-plaintext highlighter-rouge">ensembldb</code>: an R package to create and use Ensembl-based annotation
resources</em>, Bioinformatics, btz031,
doi:<a href="https://doi.org/10.1093/bioinformatics/btz031">10.1093/bioinformatics/btz031</a>.</p>

<p>Laurent Gatto, Lisa M Breckels, Kathryn S Lilley <em>Assessing
sub-cellular resolution in spatial proteomics experiments</em> Current
Opinion in Chemical Biology, 48 123-149
(2019), doi:<a href="https://doi.org/10.1016/j.cbpa.2018.11.015">10.1016/j.cbpa.2018.11.015</a>.</p>

<p>Crook OM, Mulvey CM, Kirk PDW, Lilley KS, Gatto L (2018) <em>A Bayesian
mixture modelling approach for spatial proteomics</em>. PLoS Comput Biol
14(11):
e1006516. <a href="https://doi.org/10.1371/journal.pcbi.1006516">https://doi.org/10.1371/journal.pcbi.1006516</a>.</p>

<p>Eglen, S., Mounce, R., Gatto, L., Currie, A., &amp; Nobis, Y. <em>Recent
developments in scholarly publishing: a view from the life
sciences</em>. Emerging Topics in Life Sciences, Dec 21, 2018, 2 (6)
775-778,
doi:<a href="http://www.emergtoplifesci.org/content/2/6/775">10.1042/ETLS20180172</a>
(pre-print
<a href="https://doi.org/10.31219/osf.io/57hcs">doi:10.31219/osf.io/57hcs</a>).</p>

<p>Aikaterini Geladaki, Nina Kocevar Britovsek, Lisa M. Breckels, Tom
S. Smith, Claire M. Mulvey, Oliver M. Crook, Laurent Gatto, Kathryn
S. Lilley <a href="https://www.nature.com/articles/s41467-018-08191-w"><em>Combining LOPIT with differential ultracentrifugation for
high-resolution spatial
proteomics</em></a>
Nature Communications volume 10, Article number:331 (2019) (pre-print
<a href="https://doi.org/10.1101/378364">10.1101/378364</a>).</p>

<p>Segeritz CP, Rashid ST, Cardoso de Brito M, Paola MS, Ordonez A,
Morell CM, Kaserman JE, Madrigal P, Hannan N, Gatto L, Tan L, Wilson
AA, Lilley K, Marciniak SJ, Gooptu B, Lomas DA, Vallier L. <em>hiPSC
hepatocyte model demonstrates the role of unfolded protein response
and inflammatory networks in α(1)-antitrypsin deficiency.</em> J
Hepatol. 2018 Jun 4. pii:
S0168-8278(18)32113-5. doi:<a href="https://dx.doi.org/10.1016/j.jhep.2018.05.028">10.1016/j.jhep.2018.05.028</a>.</p>

<p>Nett I, Mulas C, Gatto L, Lilley KS, Smith A. <em>Negative feedback via
RSK modulates Erk‐dependent progression from naïve pluripotency</em> EMBO
reports (2018) e45642,
doi:<a href="http://dx.doi.org/10.15252/embr.201745642">10.15252/embr.201745642</a>.</p>

<p>Thul PJ, <em>et al.</em> <em>A subcellular map of the human
proteome</em>. Science. 2017 May 11. pii: eaal3321,
doi:<a href="http://science.sciencemag.org/content/early/2017/05/10/science.aal3321">10.1126/science.aal3321</a>.
PubMed <a href="https://www.ncbi.nlm.nih.gov/pubmed/28495876">PMID:28495876</a>.</p>

<p>Mulvey CM, Breckels LM, Geladaki A, Kocevar Britovsek N, Nightingale
DJH, Christoforou A , Elzek M, Deery MJ, Gatto L, Lilley KS. <em>Using
HyperLOPIT to perform high-resolution mapping of the spatial
proteome</em>. Nature Protocols, 12, 1110–1135 (2017)
<a href="https://www.nature.com/nprot/journal/v12/n6/full/nprot.2017.026.html">doi:10.1038/nprot.2017.026</a>
(See the
<a href="https://f1000research.com/articles/5-2926/">F1000Research workflow</a>
for details on the computational side of the protocol.)</p>

<p>Leprevost FD, et al. <em><a href="http://biocontainers.pro/">BioContainers</a>: An
open-source and community-driven framework for software
standardization.</em> Bioinformatics. 2017 Mar 30,
doi:<a href="https://academic.oup.com/bioinformatics/article/doi/10.1093/bioinformatics/btx192/3096437/BioContainers-An-open-source-and-community-driven">10.1093/bioinformatics/btx192</a>. [Epub
ahead of print] <a href="https://www.ncbi.nlm.nih.gov/pubmed/28379341">PubMed
PMID:28379341</a>.</p>

<p>Breckels LM, Mulvey CM, Lilley KS and Gatto L. <em>A Bioconductor
workflow for processing and analysing spatial proteomics data</em>
F1000Research 2016, 5:2926
(doi:<a href="https://f1000research.com/articles/5-2926/">10.12688/f1000research.10411.1</a>).
[Software: <a href="http://bioconductor.org/packages/MSnbase">MSnbase</a>,
<a href="http://bioconductor.org/packages/pRoloc">pRoloc</a>,
<a href="http://bioconductor.org/packages/pRolocGUI">pRolocGUI</a>]</p>

<p>Wieczorek S, Combes F, Lazar C, Giai Gianetto Q, Gatto L, Dorffer A,
Hesse A, Coute Y, Ferro M, Bruley C, and Burger T. <em>DAPAR &amp; ProStaR:
software to perform statistical analyses in quantitative discovery
proteomics</em> Bioinformatics 2016,
doi:<a href="http://bioinformatics.oxfordjournals.org/content/33/1/135">10.1093/bioinformatics/btw580</a>.</p>

<p>Perez-Riverol Y, Gatto L, Wang R, Sachsenberg T, Uszkoreit J,
Leprevost Fda V, Fufezan C, Ternent T, Eglen SJ, Katz DS, Pollard TJ,
Konovalov A, Flight RM, Blin K, Vizcaino JA. <em>Ten Simple Rules for
Taking Advantage of Git and GitHub</em>. PLoS Comput Biol. 2016 Jul
14;12(7):e1004947,
doi:<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004947">10.1371/journal.pcbi.1004947</a>
<a href="http://www.ncbi.nlm.nih.gov/pubmed/27415786">PMID:27415786</a>.</p>

<p>Breckels LM, Holden S, Wonjar D, Mulvey CM, Christoforou A, Groen AJ,
Kohlbacher O, Lilley KS, Gatto L. <em>Learning from heterogeneous data
sources: an application in spatial proteomics</em>. PLoS Comput Biol. 2016
May 13;12(5):e1004920,
doi:<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004920">10.1371/journal.pcbi.1004920</a>
(<a href="http://bioconductor.org/packages/release/bioc/html/pRoloc.html">Software</a>)</p>

<p>Fabre B, Korona D, Groen A, Vowinckel J, Gatto L, Deery MJ, Ralser M,
Russell S, Lilley KS. <em>Analysis of the Drosophila melanogaster
proteome dynamics during the embryo early development by a combination
of label-free proteomics approaches</em>,
Proteomics, 2016 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/27029218">PMID:27029218</a>,
<a href="http://dx.doi.org/10.1002/pmic.201500482">Publisher</a>)</p>

<p>Lazar C, Gatto L, Ferro M, Bruley C, Burger T. <em>Accounting for the
Multiple Natures of Missing Values in Label-Free Quantitative
Proteomics Data Sets to Compare Imputation Strategies</em>. J Proteome
Res. 2016 Apr
1;15(4):1116-25. (<a href="http://pubsdc3.acs.org/doi/abs/10.1021/acs.jproteome.5b00981">Publisher</a>,
<a href="http://www.ncbi.nlm.nih.gov/pubmed/26906401">PMID:26906401</a>,
Software:
<a href="https://cran.r-project.org/web/packages/imputeLCMD/index.html">CRAN</a>
and
<a href="http://bioconductor.org/packages/release/bioc/html/MSnbase.html">Bioconductor</a>)</p>

<p>Christoforou A, Mulvey CM, Breckels LM, Geladaki A, Hurrell T, Hayward
PC, Naake T, Gatto L, Viner R, Arias AM, Lilley KS. A draft map of the
mouse pluripotent stem cell spatial proteome. Nat Commun. 2016 Jan
12;7:9992,
doi:<a href="http://www.nature.com/ncomms/2016/160112/ncomms9992/abs/ncomms9992.html">10.1038/ncomms9992</a>
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/26754106">PMID:26754106</a>,
<a href="http://www.bioconductor.org/packages/release/data/experiment/html/pRolocdata.html">data</a>,
<a href="http://www.ebi.ac.uk/pride/archive/projects/PXD001279">PRIDE</a>,
<a href="https://lgatto.shinyapps.io/christoforou2015/">resource</a>).</p>

<p>Gatto L, Hansen KD, Hoopmann MR, Hermjakob H, Kohlbacher O and Beyer,
A <em>Testing and validation of computational methods for mass
spectrometry</em>. J Proteome Res. 2015,
doi:<a href="http://pubs.acs.org/doi/abs/10.1021/acs.jproteome.5b00852">10.1002/stem.2067</a>
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/26549429">PubMed</a>).</p>

<p>Mulvey CM, Schröter C, Gatto L, Dikicioglu D, Baris Fidaner I,
Christoforou A, Deery MJ, Cho LT, Niakan KK, Martinez-Arias A, Lilley
KS. <em>Dynamic proteomic profiling of extra-embryonic endoderm
differentiation in mouse embryonic stem cells</em>. Stem Cells. 2015
Jun 8. doi: 10.1002/stem.2067
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/26059426">PubMed</a>).</p>

<p>Gatto L, Breckels LM, Naake T and Gibb S <em>Visualisation of proteomics
data using R and Bioconductor</em>. Proteomics. 2015
Feb 18. doi:10.1002/pmic.201400392
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/25690415">PubMed</a>,
<a href="http://onlinelibrary.wiley.com/doi/10.1002/pmic.201400392/abstract">Publisher</a>
and software:
<a href="http://bioconductor.org/packages/devel/data/experiment/html/RforProteomics.html">Bioconductor</a>,
<a href="https://github.com/lgatto/RforProteomics">github</a>).</p>

<p>Huber W et al. <em>Orchestrating high-throughput genomic analysis with
Bioconductor</em>. Nat Methods. 2015 Jan 29;12(2):115-21
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/25633503">PubMed</a>,
<a href="http://www.nature.com/doifinder/10.1038/nmeth.3252">Publisher</a>).</p>

<p>Hiemstra TF et al. <em>Human urinary exosomes as innate immune
effectors</em>, J Am Soc Nephrol. 2014
Sep;25(9):2017-27. (<a href="http://www.ncbi.nlm.nih.gov/pubmed/24700864">PubMed</a>,<a href="http://jasn.asnjournals.org/content/25/9/2017">Publisher</a>).</p>

<p>Nikolovski N, Shliaha PV, Gatto L, Dupree P and Lilley KS <em>Label free
protein quantification for plant Golgi protein localisation and
abundance</em>, Plant Physiol. pp.114.245589; First Published on August
13, 2014; doi:10.1104/pp.114.245589 (<a href="http://www.plantphysiol.org/content/early/2014/08/13/pp.114.245589.short">Publisher</a>,
<a href="http://www.ncbi.nlm.nih.gov/pubmed/25122472">PubMed</a>)</p>

<p>Griss J, <em>et al.</em> <em>The mzTab Data Exchange Format: communicating
MS-based proteomics and metabolomics experimental results to a wider
audience</em>, Mol Cell Proteomics. 2014 June 30. (<a href="http://www.mcponline.org./content/early/2014/06/30/mcp.O113.036681.abstract">Publisher</a>)</p>

<p>Tomizioli M, <i>et al.</i> <em>Deciphering thylakoid sub-compartments
using a mass spectrometry-based approach</em>, Mol Cell Proteomics. 2014
May 28. (<a href="http://mcponline.org/content/early/2014/05/28/mcp.M114.040923.abstract">Publisher</a>,
<a href="http://www.ncbi.nlm.nih.gov/pubmed/24872594">PubMed</a>)</p>

<p>Gatto L, <i>et al.</i> <em>A foundation for reliable spatial proteomics
data analysis</em>, Mol Cell Proteomics. 2014 Aug;13(8):1937-52. (<a href="http://www.mcponline.org/content/13/8/1937.long">Publisher</a>,
<a href="http://www.ncbi.nlm.nih.gov/pubmed/24846987">PubMed</a>, <a href="http://www.bioconductor.org/packages/devel/bioc/html/pRoloc.html">software</a>,
<a href="http://www.genomeweb.com/proteomics/cambridge-researchers-outline-best-practices-analysis-organelle-proteomics-data">press
coverage</a>)</p>

<p>Walzer M, <i>et al.</i> <em>qcML: an exchange format for quality control
metrics from mass spectrometry experiments</em>, Mol Cell Proteomics. 2014
Apr 23 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/24760958">PubMed</a>).</p>

<p>Vizcaíno J.A. <i>et al.</i> <em>ProteomeXchange: globally co-ordinated
proteomics data submission and dissemination</em>, Nature Biotechnology
2014, 32, 223–226 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/24727771">PubMed</a>)</p>

<p>Gatto L., Breckels L.M, Burger T, Wieczorek S. and Lilley
K.S. <em>Mass-spectrometry based spatial proteomics data analysis using
pRoloc and pRolocdata</em>, Bioinformatics, 2014 May
1;30(9):1322-4. doi:10.1093/bioinformatics/btu013 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/24413670">PubMed</a>, <a href="http://bioinformatics.oxfordjournals.org/content/30/9/1322">publisher</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/pRoloc.html">software</a>
and <a href="http://www.bioconductor.org/packages/release/data/experiment/html/pRolocdata.html">data</a>).</p>

<p>Groen A., Sancho-Andrés G., Breckels LM., Gatto L., Aniento
F. and Lilley K.S. <em>Identification of Trans Golgi Network proteins in
Arabidopsis thaliana root tissue</em> Journal of Proteome Research, 2013
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/24344820">PubMed</a>, <a href="http://pubs.acs.org/doi/abs/10.1021/pr4008464">publisher</a>).</p>

<p>Wilf N.M. <i>et al.</i> <em>RNA-seq reveals the RNA binding proteins, Hfq
and RsmA, play various roles in virulence, antibiotic production and
genomic flux in Serratia sp. ATCC 39006</em> <a href="http://www.biomedcentral.com/1471-2164/14/822/abstract">BMC
Genomics 2013, 14:822</a>.</p>

<p>Gatto L. and Christoforou A. <em>Using R and Bioconductor for proteomics
data analysis</em>, Biochim Biophys Acta - Proteins and Proteomics, 2013.
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/23692960">PubMed</a>,
<a href="http://arxiv.org/abs/1305.6559">pre-print</a> and software:
<a href="http://bioconductor.org/packages/devel/data/experiment/html/RforProteomics.html">Bioconductor</a>,
<a href="https://github.com/lgatto/RforProteomics">github</a>).</p>

<p>Bond N.J., Shliaha P.V, Lilley K.S., and Gatto L. <em>Improving
qualitative and quantitative performance for MS<sup>E</sup>-based
label free proteomics</em>, J. Proteome Res., 2013 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/23510225">PubMed</a>, <a href="http://pubs.acs.org/doi/abs/10.1021/pr300776t">publisher</a>, <a href="http://www.bioconductor.org/packages/release/bioc/html/synapter.html">software</a>).</p>

<p>Shliaha P.V, Bond N.J., Gatto L. and Lilley K.S. <em>The Effects of
Travelling Wave Ion Mobility Separation on Data Independent
Acquisition in Proteomics Studies</em>, J. Proteome Res., 2013
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/23514362">PubMed</a>,
<a href="http://pubs.acs.org/doi/abs/10.1021/pr300775k">publisher</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/synapter.html">software</a>).</p>

<p>Breckels L.M., Gatto L., Christoforou A., Groen A.J., Lilley K.S. and
Trotter M.W.B.  <em>The Effect of Organelle Discovery upon Sub-Cellular
Protein Localisation</em>, Journal of Proteomics, 2013 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/23523639">PubMed</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/pRoloc.html">software</a>).</p>

<p>Chambers M. <i>et al.</i> <em>A Cross-platform Toolkit for Mass
Spectrometry and Proteomics</em>, <a href="http://www.nature.com/nbt/index.html">Nature Biotechnology</a>
30, 918–920, 2012 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/23051804">PubMed</a>,
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3471674/pdf/nihms-374482.pdf">pdf</a>,
software [<a href="http://proteowizard.sourceforge.net/">1</a>|<a href="http://www.bioconductor.org/packages/release/bioc/html/mzR.html">2</a>]).</p>

<p>Gatto L. and Lilley K.S. <em>MSnbase - an R/Bioconductor package for
isobaric tagged mass spectrometry data visualisation, processing and
quantitation</em>, <a href="http://bioinformatics.oxfordjournals.org/">Bioinformatics</a>,
28(2), 288-289, 2012 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/22113085">PubMed</a>,
<a href="http://bioinformatics.oxfordjournals.org/content/28/2/288.full.pdf">pdf</a>,
<a href="http://www.bioconductor.org/packages/release/bioc/html/MSnbase.html">software</a>).</p>

<p>Capuano F., Bond N.J., Gatto L., Beaudoin F., Napier J., Benvenuto E.,
Lilley K.S, and Baschieri S. <em>LC-MS/MS methods for absolute
quantification and identification of proteins associated to chimeric
plant oil bodies</em>, <a href="http://pubs.acs.org/journal/ancham">Analytical Chemistry</a>,
Dec 15;83(24):9267-72, 2011 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/22017570">PubMed</a> - <a href="http://www.ebi.ac.uk/pride/simpleSearch.do?simpleSearchValue=19642">data</a>).</p>

<p>Foster J.M., Degroeve S., Gatto L., Visser, M., Wang, R., Griss J.,
Apweiler R. and Martens L. <em>A posteriori quality control for the
curation and reuse of public proteomics data</em>,
<a href="http://www3.interscience.wiley.com/journal/76510741/home">Proteomics</a>
11(11):2182-94, 2011
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/21538885">PubMed</a>,
<a href="http://onlinelibrary.wiley.com/doi/10.1002/pmic.201000602/pdf">pdf</a>).</p>

<p>Lilley K.S., Deery M.J. and Gatto L. <em>Challenges for Proteomics Core
Facilities</em>,
<a href="http://www3.interscience.wiley.com/journal/76510741/home">Proteomics</a>
11: 1017–1025, 2011
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/21360676">PubMed</a>,
<a href="http://onlinelibrary.wiley.com/doi/10.1002/pmic.201000693/pdf">pdf</a>).</p>

<p>Gatto L., Vizcaíno J.A., Hermjakob H., Huber W. and Lilley
K.S. <em>Organelle proteomics experimental designs and analysis</em>
<a href="http://www3.interscience.wiley.com/journal/76510741/home">Proteomics</a>,
10:22, 3957-3969, 2010
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/21080489">PubMed</a>,
<a href="http://onlinelibrary.wiley.com/doi/10.1002/pmic.201000244/pdf">pdf</a>).</p>

<p><a href="http://www.nature.com/nbt/journal/v28/n8/abs/nbt.1665.html#/group-1">MAQC
Consortium</a> <em>The MicroArray Quality Control (MAQC)-II study of
common practices for the development and validation of
microarray-based predictive models</em> <a href="http://www.nature.com/nbt/index.html">Nature Biotechnology</a>
28, 827–838 2010
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/20676074">PubMed</a>,
<a href="http://www.nature.com/nbt/journal/v28/n8/pdf/nbt.1665.pdf">pdf</a>).</p>

<p>Gatto L., Mardulyn P. and Pasteels J.M. <em>Morphological and
mitochondrial DNA analysis suggest the presence of a hybrid zone
between two species of the leaf beetle Gonioctena variabilis species
complex in southern Spain</em>, <a href="http://www.wiley.com/bw/journal.asp?ref=0024-4066&amp;site=1">Biological
Journal of the Linnean Society</a>, 2008, 94(1), 105-114
(<a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1095-8312.2008.00972.x/abstract">abstract</a>,
<a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1095-8312.2008.00972.x/pdf">pdf</a>).</p>

<p>Danis B., George T.C., Goriely S., Dutta B., Renneson J., Gatto L.,
Fitzgerald-Bocarsly P., Marchant A., Goldman M., Willems F. and De Wit
D. <em>Interferon regulatory factor 7-mediated responses are defective in
cord blood plasmacytoid dendritic cells.</em> <a href="http://onlinelibrary.wiley.com/journal/10.1002/(ISSN)1521-4141">Eur
J Immunol.</a> 2008 Feb;38(2):507-17.
(<a href="http://www.ncbi.nlm.nih.gov/pubmed/18200500">PubMed</a>,
<a href="http://onlinelibrary.wiley.com/doi/10.1002/eji.200737760/pdf">pdf</a>).</p>

<p>Gatto L., Catanzaro D. and Milinkovitch M.C. <em>Assessing the
Applicability of the GTR Nucleotide Substitution Model Through
Simulations</em> <a href="http://www.la-press.com/evolutionary-bioinformatics-journal-j17">Evolutionary
Bioinformatics</a> 2006:2 (<a href="http://www.ncbi.nlm.nih.gov/pubmed/19455208">PubMed</a>,
<a href="http://www.la-press.com/redirect_file.php?fileId=648&amp;filename=EBO-2-Milinkovitch-et-al-(2)&amp;fileType=pdf">pdf</a>).</p>

<h3 id="book-chapters">Book chapters</h3>

<p>Grégoire S, Vanderaa C, Pyr dit Ruys S, Mazzucchelli G, Kune C,
Vertommen D and Gatto L <a href="https://doi.org/10.1007/978-1-0716-3934-4_14">Standardised workflow for mass
spectrometry-based single-cell proteomics data processing and analysis
using the <code class="language-plaintext highlighter-rouge">scp</code> package</a>
In: Vegvari, A., Teppo, J., Zubarev, R.A. (eds) Mass Spectrometry
Based Single Cell Proteomics. Methods in Molecular Biology,
vol 2817. Humana, New York, NY (<a href="https://arxiv.org/abs/2310.13598">arXiv
pre-print</a>), April 2024.</p>

<p>Breckels LM, Gibb S, Petyuk V and Gatto L <a href="http://pubs.rsc.org/en/content/chapter/bk9781782624288-00321/978-1-78262-428-8#!divabstract">R for
Proteomics</a>.
In: <em>Proteome Informatics</em>, The Royal Society of Chemistry, November
2016.</p>

<p>Christoforou A., Mulvey C., Breckels LM., Gatto L. and Lilley
KS. <a href="http://pubs.rsc.org/en/content/chapter/bk9781849738088-00185/978-1-84973-808-8#!divabstract">Spatial Proteomics: Practical Considerations for Data Acquisition
and Analysis in Protein Subcellular Localisation
Studies</a>.
In <em>Quantitative Proteomics</em>, 185-210, The Royal Society of Chemistry,
2014.</p>

<h3 id="technical-notes">Technical Notes</h3>

<p>Gatto L. <em>Data Management Plan for a Biotechnology and Biological
Sciences Research Council (BBSRC) Tools and ResourcesDevelopment Fund
(TRDF) Grant</em>, Research Ideas and Outcomes (2017),
<a href="http://riojournal.com/articles.php?id=11624">doi:10.3897/rio.3.e11624</a>.</p>

<p>Gatto, L. and Schretter, C. <em>Designing Primer Pairs and Oligos with
OligoFaktorySE</em>. <a href="http://journal.embnet.org/index.php/embnetnews/article/view/38">EMBnet.news</a>
North America, 15, oct. 2009
(<a href="http://journal.embnet.org/index.php/embnetnews/article/download/38/58">pdf</a>,<a href="http://www.bioinformatics.org/oligofaktory/">software</a>).</p>

<p>Schretter, C. and Gatto, L. <em>A Tiny Queuing System for Blast Servers</em>
December, 2005
(<a href="http://proteome.sysbiol.cam.ac.uk/lgatto/files/tiny_queuing_system.pdf">short</a>
and slighly
<a href="http://proteome.sysbiol.cam.ac.uk/lgatto/files/OligoFaktory_queuing_system.pdf">longer</a>
versions).</p>

<h2 id="webpress-coverage">Web/press coverage</h2>

<p><em>News in Proteomics Research</em> blog posts about <a href="https://proteomicsnews.blogspot.co.uk/2013/07/r-for-proteomics.html">R for
proteomics</a>,
<a href="https://proteomicsnews.blogspot.co.uk/2015/02/bioconductor-for-proteomics.html">R and
Bioconductor</a>,
<a href="https://proteomicsnews.blogspot.co.uk/2016/12/bioconductor-workflow-for-spatial.html
and
https://proteomicsnews.blogspot.co.uk/2017/05/hyperlopit-high-resolution-spatial.html">spatial
proteomics</a>
and <a href="https://proteomicsnews.blogspot.co.uk/2016/02/imputation-strategies-in-label-free.html">handling of missing
values</a>.</p>

<p><em>Genome web</em> <a href="https://www.genomeweb.com/proteomics-protein-research/statistical-design-remains-sticking-point-proteomics-experiments">Statistical Design Remains Sticking Point for Proteomics
Experiments</a>
Jul 27, 2017.</p>

<p><a href="http://www.researchresearch.com/news/article/?articleId=1370722">Publishers put the squeeze on
ResearchGate</a>,
11 Oct 2017.</p>

<p><em>Nature News &amp; Comment</em> Correspondance <a href="https://www.nature.com/articles/d41586-018-06055-3">Preprints help journalism, not
hinder it</a>, , 29
Aug 2018.</p>

<h2 id="software">Software</h2>

<p>I have developed and have contributed to many open source
R/<a href="http://www.bioconductor.org">Bioconductor</a> packages, in particular
<a href="http://bioconductor.org/packages/release/BiocViews.html#___Proteomics">proteomics</a>
software and data packages, all of which are available on my
<a href="http://www.github.com/lgatto">own</a> and my
<a href="http://www.github.com/ComputationalProteomicsUnit">group’s</a> GitHub
pages. See
<a href="http://www.ncbi.nlm.nih.gov/pubmed/23692960">Gatto and Christoforou, 2014</a>,
<a href="http://www.ncbi.nlm.nih.gov/pubmed/25690415">Gatto et al., 2014</a> and
the
<a href="http://www.bioconductor.org/packages/release/data/experiment/html/RforProteomics.html"><code class="language-plaintext highlighter-rouge">RforProteomics</code> vignettes</a>
for an overview of the R/Bioconductor infrastructure for mass
spectrometry and proteomics.</p>

<h2 id="talks">Talks</h2>

<p>This is an incomplete list of talks I have given in the past.</p>

<p>Principled and Reproducible Mass Spectrometry-based Single Cell
Proteomics Data Analysis, USHUPO, 08 March 2020.</p>

<p><a href="http://bit.ly/2020SCP">Standardised and reproducible analysis of mass spectrometry-based
single-cell proteomics data</a>, <a href="http://single-cell.net/proteomics/scp2020">Single Cell
Proteomics meeting</a>, 18
August 2020.</p>

<p><a href="https://lgatto.github.io/2020_07_15_ISMB/">MSnbase, efficient and elegant R-based processing and visualisation
of raw mass spectrometry
data</a>, ISCB ISMB 2020
virtual conference, 15 July 2020.</p>

<p><a href="https://lgatto.github.io/2020_05_07_MayInstitute/">MSnbase design principles and code
walk</a>, Online May
Institute for Computation and statistics for mass spectrometry and
proteomics, 7 May 2020.</p>

<p><a href="http://bit.ly/ABLS2020">Probabilistic mapping of the sub-cellular
proteome</a>, VIB Applied Bioinformatics in Life
Sciences (3rd edition), 13-14 February 2020, Leuven Belgium.</p>

<p>Invited talk at the Belgian Biophysical Society Symposium <a href="https://www.sciences.uliege.be/cms/c_4944765/en/protein-folding-and-stability">Protein
folding and
stability</a>,
30 August 2019, Liège.</p>

<p>Invited talk at the <em>The Alan Turing Institute Workshop on Statistical
Data Science for Proteomics and Metabolomics</em>, 13th June 2019.</p>

<p>Invited talk at the <a href="https://kuleuvencongres.be/iscb40">International Society for Clinical Biostatistics
BePA mini-symposium</a> at the
KULeuven, 18 July 2019.</p>

<p><a href="https://lgatto.github.io/rr-publ/">Becoming a better scientist with open and reproducible
research</a>, <em>Are you ready for
publishing reproducible research?</em> meeting, 16 May 2019, TU Delft,
Netherlands.</p>

<p><a href="http://bit.ly/20180109eubic">Open source and open development proteomics software</a> at
the
<a href="http://uahost.uantwerpen.be/eubic18/">EuBIC 2018 developer’s meeting</a>,
9 - 12 January 2018, Ghent, Belgium.</p>

<p>A longer talk about <em>the <a href="http://bulliedintobadscience.org/">Bullied Into Bad
Science</a> campaign</em> at the
<a href="http://www.opencon2017.org/lgatt0/opencon_2017_cambridge">OpenConCam</a>
2017 conference on the 16 November 2017. The content and slides are
available
<a href="https://lgatto.github.io/BulliedIntoBadScience-OpenConCam/">here</a>.</p>

<p>Short <a href="http://bit.ly/20171112OpenCon">I won’t be #BulliedIntoBadScience!</a> as
part of the panel discussion <em>Next-generation initiatives advancing
open</em>, at <a href="http://www.opencon2017.org/">OpenCon</a> 11 - 13 November
2017, Berlin.</p>

<p><a href="http://bit.ly/20171108FLI">Mapping the sub-cellular proteome</a>, 8
November 2017, Leibniz Institut for Aging, Jena, Germany.</p>

<p><a href="https://osip2017.epfl.ch/">Open Science in Practice</a>, 25 September
2017, Lausanne,
Switzerland. <a href="https://lgatto.github.io/EPFL-open-science/">An early career researcher’s view on modern and open scholarship</a>.</p>

<p><a href="https://www.proteomicsmethodsforum.org.uk/">Proteomics Method Forum</a>,
Oxford, UK, 22-23
June 2017. <a href="https://bit.ly/20170623pmf">The Bioconductor project - analysis and comprehension of high-throughput proteomics data</a>.</p>

<p><a href="http://www.dcc.ac.uk/events/research-data-management-forum-rdmf/rdmf17">Research Data Management Forum</a>,
London, UK, 9th
June 2017. <a href="https://bit.ly/20170609rdmf">An early career researcher’s view on modern and open scholarship … and careers</a>.</p>

<p>Office of Scholarly Communication
Training -
<a href="https://www.training.cam.ac.uk/osc/event/2080517">How to Get the Most Out of Modern Peer Review</a>,
Cambridge, UK, 30
Mar 2017. <a href="http://bit.ly/20170329peerrev">The role of peer-reviewers in promoting open science</a>.</p>

<p><a href="http://scicore.ch/events/eurobioc2016/">European Bioconductor Developer Meeting</a>,
Zurich, Switzerland, 6 - 7
December 2016. <a href="https://github.com/lgatto/EuroBioc2016-Basel-MSnbase2"><code class="language-plaintext highlighter-rouge">MSnbase2</code> - disk access is the limit</a>.</p>

<p>Cambridge Computational Biology Institute, UK, 16
November 2016. <a href="http://goo.gl/SZRMjg">Mapping the sub-cellular proteome: Computational analyses of high-throughput mass spectrometry-based spatial proteomics data</a>.</p>

<p>Dialogue on methods for ecology, Cambridge, UK, 15 November 2016,
Learning from heterogeneous data in spatial proteomics.</p>

<p><a href="https://www.biochemistry.org/Events/tabid/379/View/Programme/MeetingNo/TD007/Default.aspx">Quantitative Proteomics and Data Analysis</a>,
Chester, UK, 4 - 5 April 2016. Inspection, visualisation and analysis
of quantitative proteomics data
(<a href="http://bit.ly/qprotda">slides</a>,
<a href="http://bit.ly/qprotdavig">vignette</a>).</p>

<p><a href="http://tinyurl.com/Intro-Integ-Omics-Prot">Introduction to Integrative Omics: proteomics</a>,
European Bioinformatics Institute, Hinxton, UK, 8 March 2016.</p>

<h2 id="get-in-touch">Get in touch</h2>

<p>On the fediverse at <a rel="me" href="https://fosstodon.org/@lgatto">lgatto@fosstodon.org</a>,
by email at <code class="language-plaintext highlighter-rouge">name.surname&lt;at&gt;institution.be</code>, via a
<a href="https://github.com/lgatto">Github</a> issue on a relevant repo or on the
<a href="https://bioc-community.herokuapp.com/">Bioconductor community slack</a>.</p>

        
      </section>

      <footer class="page__meta">
        
        


        
          <p class="page__date"><strong><i class="fa fa-fw fa-calendar" aria-hidden="true"></i> Updated:</strong> <time datetime="2022-11-05">November 05, 2022</time></p>
        
      </footer>

      

      
    </div>

    
  </article>

  
  
</div>

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
      <li><strong>Follow:</strong></li>
    
    
    
    
      <li><a href="http://github.com/lgatto"><i class="fa fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li>
    
    
    <li><a href="https://lgatto.github.io/feed.xml"><i class="fa fa-fw fa-rss-square" aria-hidden="true"></i> Feed</a></li>
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2024 Laurent Gatto CC-BY. Powered by <a href="http://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    <script src="https://lgatto.github.io/assets/js/main.min.js"></script>





  </body>
</html>
