<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Preamble | Introduction to bioinformatics" />
<meta property="og:type" content="book" />

<meta property="og:description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2025-06-16" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="/usr/share/javascript/mathjax/MathJax.js" type="text/javascript"></script>

<meta name="description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain.">

<title>Preamble | Introduction to bioinformatics</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script>
function copy_link(id) {
  var dummy = document.createElement('input'),
  text = window.location.href.split(/[?#]/)[0] + '#' + id;
  document.body.appendChild(dummy);
  dummy.value = text;
  dummy.select();
  document.execCommand('copy');
  document.body.removeChild(dummy);
  
  var tooltip = document.getElementById(id + '-tooltip');
  tooltip.innerHTML = 'Copied!';
}

function reset_tooltip(id) {
  var tooltip = document.getElementById(id);
  tooltip.innerHTML = 'Copy link';
}
</script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }

code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Introduction to bioinformatics<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a id="active-page" href="index.html" id="toc-preamble">Preamble</a><ul class="toc-sections">
<li class="toc"><a href="#motivation">Motivation</a></li>
<li class="toc"><a href="#references-and-credits">References and credits</a></li>
<li class="toc"><a href="#pre-requisites">Pre-requisites</a></li>
<li class="toc"><a href="#about-this-course-material">About this course material</a></li>
<li class="toc"><a href="#citation">Citation</a></li>
<li class="toc"><a href="#license">License</a></li>
<li class="toc"><a href="#setup">Setup</a></li>
</ul>
<a href="sec-dataorg.html" id="toc-sec-dataorg"><span class="toc-section-number">1</span> Data organisation with Spreadsheets</a>
<a href="sec-rrstudio.html" id="toc-sec-rrstudio"><span class="toc-section-number">2</span> R and RStudio</a>
<a href="sec-startr.html" id="toc-sec-startr"><span class="toc-section-number">3</span> Introduction to R</a>
<a href="sec-startdata.html" id="toc-sec-startdata"><span class="toc-section-number">4</span> Starting with data</a>
<a href="sec-dplyr.html" id="toc-sec-dplyr"><span class="toc-section-number">5</span> Manipulating and analyzing data with dplyr</a>
<a href="sec-vis.html" id="toc-sec-vis"><span class="toc-section-number">6</span> Data visualization</a>
<a href="sec-join.html" id="toc-sec-join"><span class="toc-section-number">7</span> Joining tables</a>
<a href="sec-rr.html" id="toc-sec-rr"><span class="toc-section-number">8</span> Reproducible research</a>
<a href="sec-bioinfo.html" id="toc-sec-bioinfo"><span class="toc-section-number">9</span> Bioinformatics</a>
<a href="sec-prog.html" id="toc-sec-prog"><span class="toc-section-number">10</span> Additional programming concepts</a>
<a href="sec-ccl.html" id="toc-sec-ccl"><span class="toc-section-number">11</span> Conclusions</a>
<a href="sec-anx.html" id="toc-sec-anx"><span class="toc-section-number">12</span> Annex</a>
<a href="session-information.html" id="toc-session-information"><span class="toc-section-number">13</span> Session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="preamble" class="section level1 unnumbered">
<h1>Preamble</h1>
<p>The <a href="https://uclouvain.be/cours-2022-wsbim1207.html">WSBIM1207</a> course
is an introduction to bioinformatics (and data science) for biology
and biomedical students. It introduces bioinformatics methodology and
technologies without relying on any prerequisites. The aim of this
course is for students to be in a position to understand important
notions of bioinformatics and tackle simple bioinformatics-related
problems in R, in particular to develop simple R analysis scripts and
reproducible analysis reports to interrogate, visualise and understand
data in a tidy tabular format.</p>
<p>The course is followed by <em>Bioinformatics</em>
(<a href="https://github.com/UCLouvain-CBIO/WSBIM1322">WSBIM1322</a>) and <em>Omics
data analysis</em>
(<a href="https://github.com/UCLouvain-CBIO/WSBIM2122">WSBIM2122</a>).</p>
<p>It is interesting to start this course by asking the students, who
have likely not yet been exposed to bioinformatics</p>
<blockquote>
<p>What is bioinformatics?</p>
</blockquote>
<p>While <a href="https://en.wikipedia.org/wiki/Bioinformatics">Wikipedia
defines</a> it as</p>
<blockquote>
<p>Bioinformatics is an interdisciplinary field that develops methods
and software tools for understanding biological data.</p>
</blockquote>
<p>this course won’t try to answer that question by providing an overview
of the many context where methods and software tools are developed in
the frame of biological data. We will focus on the getting hands-on
experience in data manipulation. Hence, the material and how it is
taught will be focused on practice.</p>
<div id="motivation" class="section level2 unnumbered">
<h2>Motivation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('motivation')" onmouseout="reset_tooltip('motivation-tooltip')"><span class="tooltiptext" id="motivation-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>Today, it is difficult to overestimate the very broad importance and
impact of <em>data</em>. Given the abundance of data around us, and the
sophistication of tools for their analysis and interpretation that are
readily available, data has become a tool of profound social
change. Research in general, and biomedical research in particular, is
at the centre of this evolution. And while bioinformatics has been
playing a central role in bio-medical research for many years now,
bioinformatics skills aren’t well integrated in life science
curricula, limiting students in their career prospects and research
horizon <span class="citation">(<label for="tufte-mn-1" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-1" class="margin-toggle">Wilson Sayres et al. 2018<span class="marginnote">Wilson Sayres, M A, C Hauser, M Sierk, S Robic, A G Rosenwald, T M Smith, E W Triplett, et al. 2018. <span>“Bioinformatics Core Competencies for Undergraduate Life Sciences Education.”</span> <em>PLoS One</em> 13 (6): e0196878. <a href="https://doi.org/10.1371/journal.pone.0196878">https://doi.org/10.1371/journal.pone.0196878</a>.</span>)</span>. It is important for young researchers to
acquire quantitative, computational and data skills to address the
challenges that lie ahead.</p>
<p>This first course will focus on the data skills. It will not teach to
use any specific piece of bioinformatics software. Once one has
identified a relevant piece of software, running it shouldn’t be a
major problem<label for="tufte-sn-1" class="margin-toggle sidenote-number">1</label><input type="checkbox" id="tufte-sn-1" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">1</span> Although, in all fairness, some bioinformatics software
are notoriously difficult to install and run.</span>. The important part will be to understand
what it does, why it does it and how to assess if the output can be
trusted. The latter requires to explore and understand the data and
the results, i.e. data skills. As described by <span class="citation">Buffalo (<label for="tufte-mn-2" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-2" class="margin-toggle">2015<span class="marginnote">Buffalo, Vince. 2015. <em>Bioinformatics Data Skills</em>. O’Reilly Media, Inc.</span>)</span>, equating
learning a piece of bioinformatics software to learning bioinformatics
is like learning pipetting as a means to learn molecular biology.</p>
<p>Critical thinking is essential in any aspect of research, and
bioinformatics and data analysis doesn’t escape that rule. Teaching
critical thinking is however very difficult, and arguably impossible
without first possessing the skills and master the tools to be
critical about data. Rather than teaching a limited set of software
tools, this course aims at teaching a set of hands-on skills and
methodologies that allow to interrogate and visualise data, and hence
be in a position to be critical with respect to new data. In addition,
while specific software become obsolete and are replaced, or are
specific to one specific field of research, critical investigation of
data will always be required and will never be replaced.</p>
<p>We will be using the <a href="https://www.R-project.org/">R</a> language and
environment <span class="citation">(<label for="tufte-mn-3" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-3" class="margin-toggle">R Core Team 2019<span class="marginnote">R Core Team. 2019. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</span>)</span> and the <a href="https://www.rstudio.com/products/RStudio/">RStudio integrated development
environment</a> to acquire
these data skills. Other interactive language such as
<a href="https://www.R-project.org/">Python</a> and the interactive <a href="https://jupyter.org/">jupyer
notebooks</a> would also have been a good fit. One
motivation of this choice is the availability of numerous
R/<a href="https://www.bioconductor.org/">Bionductor</a> packages <span class="citation">(<label for="tufte-mn-4" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-4" class="margin-toggle">Huber et al. 2015<span class="marginnote">Huber, W, V J Carey, R Gentleman, S Anders, M Carlson, B S Carvalho, H C Bravo, et al. 2015. <span>“Orchestrating High-Throughput Genomic Analysis with <span>Bioconductor</span>.”</span> <em>Nat Methods</em> 12 (2): 115–21. <a href="https://doi.org/10.1038/nmeth.3252">https://doi.org/10.1038/nmeth.3252</a>.</span>)</span>
for the analysis of high throughput biology data.</p>
<p>Another reason why the focus of the course ought to be on data skills
is that a notable difficulty in modern, multidisciplinary research is
communication. Wetlab biomedical scientists aren’t required to become
bioinformaticians, statisticians, programmers, … to be outstanding
researchers, but they will need to communicate efficiently with these
experts (and vice versa). What most often unites all these experts is
data, and communication around data is critical. The importance of
critical thinking and communication around data becomes more evident
when one realises that, when tracking work in bioinformatics core
facilities, only a minority of projects were purely routine and that
most researchers came to the bioinformatics core seeking customized
analysis, rather than a standardized package <span class="citation">(<label for="tufte-mn-5" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-5" class="margin-toggle">Chang 2015<span class="marginnote">Chang, J. 2015. <span>“Core Services: Reward Bioinformaticians.”</span> <em>Nature</em> 520 (7546): 151–52. <a href="https://doi.org/10.1038/520151a">https://doi.org/10.1038/520151a</a>.</span>)</span>.</p>
<p>To illustrate the reasons why R in general (and in the case of
biomedical sciences, Bioconductor in particular) are worth learning, I
provide here some examples where these software are used. From a
bioinformatics point of view:</p>
<ul>
<li><p>Single cell transcriptomics data: <a href="https://f1000research.com/articles/6-1158/v1">Bioconductor workflow for
single-cell RNA sequencing: Normalization, dimensionality reduction,
clustering, and lineage
inference</a></p></li>
<li><p>Quantitative proteomics data: <a href="https://f1000research.com/articles/5-2926/v2">A Bioconductor workflow for
processing and analysing spatial proteomics
data</a>.</p></li>
<li><p>Epigenetic data: <a href="https://f1000research.com/articles/6-2055/v2">Differential methylation analysis of reduced
representation bisulfite sequencing experiments using
edgeR</a></p></li>
<li><p>…</p></li>
</ul>
<p>In addition, here are three short videos (in French, with subtitles in
multiple languages) by PhD students that assist in the teaching of
this course. The videos illustrate real-work applications of some of
the concepts taught in the two bachelor bioinformatics course (this
one and <a href="https://github.com/UCLouvain-CBIO/WSBIM1322">WSBIM1322</a>):</p>
<ul>
<li>
<a href="https://youtu.be/MOa9fCHqKko">Julie Devis</a> is pursuing a PhD in the
Computational Biology and Bioinformatics Unit with Prof Laurent
Gatto. She uses R and Bioconductor to study the methylation in
cancer germ-line genes.</li>
<li>
<a href="https://youtu.be/XUiOQ0LRKxc">Valentine Robaux</a> is pursuing a PhD
in the Cardiovascular Research Unit with Profs Sandrine Horman and
Christophe Beauloye. She uses R and Bioconductor to investigate
platelet function.</li>
<li>
<a href="https://youtu.be/XC6HDdPzPac">Jean Fain</a> is pursuing a PhD in the
Epigenetics Unit with Prof Charles De Smet. He uses R and
Bioconductor to study DNA methylation and its role in cancer.</li>
</ul>
<p>And more generally, at Google, Pfizer, Merck, GSK, Bank of America,
the InterContinental Hotels Group, Shell, …</p>
<ul>
<li><a href="https://www.nytimes.com/2009/01/07/technology/business-computing/07program.html">Data Analysts Captivated by R’s
Power</a></li>
</ul>
<p>In summary, the overall learning objectives of this course are:</p>
<ul>
<li><p>for students to apply and adapt the general data analysis techniques
and principles that are presented to new data and new contexts;</p></li>
<li><p>establish links between different concepts seen in the course such
as, for example, the importance of tidy data in general, how it
applies to dataframes in R, and how it enables reasoning on the
data;</p></li>
<li><p>become autonomous when being presented with new data and be in a
position to explore and understand them.</p></li>
</ul>
</div>
<div id="references-and-credits" class="section level2 unnumbered">
<h2>References and credits<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('references-and-credits')" onmouseout="reset_tooltip('references-and-credits-tooltip')"><span class="tooltiptext" id="references-and-credits-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>References are provided throughout the course. Several stand out
however, as they cover large parts of the material or provide
complementary resources.</p>
<p>The material for the first chapters, covering the <em>Introduction to
data science with R</em>, was originally based on the <a href="https://datacarpentry.org/lessons/#ecology-workshop"><strong>Data Carpentry</strong>
Ecology
curiculum</a>
<span class="citation">(<label for="tufte-mn-6" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-6" class="margin-toggle">von Hardenberg et al. 2019<span class="marginnote">von Hardenberg, Achaz, Adam Obeng, Aleksandra Pawlik, Alex Pletzer, Alexey Shiklomanov, Anne Fouilloux, April Wright, et al. 2019. <span>“<span>Data Carpentry</span>: <span>R</span> for Data Analysis and Visualization of Ecological Data.”</span> Edited by Francois Michonneau and Auriel Fournier. <a href="https://doi.org/10.5281/zenodo.569338">https://doi.org/10.5281/zenodo.569338</a>.</span>)</span>. The main dataset by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5544260/">Blackmore <em>et al.</em>
(2017)</a>,
introduced in 2021, was prepared by the <a href="https://github.com/Bioconductor/bioconductor-teaching/blob/master/README.md">Bioconductor education
group</a>
as part of their lesson development.</p>
<p>General references for this course are <em>R for Data Science</em>
<span class="citation">(<label for="tufte-mn-7" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-7" class="margin-toggle">Grolemund and Wickham 2017<span class="marginnote">Grolemund, Garrett, and Hadley Wickham. 2017. <em>R for Data Science</em>. O’Reilly Media. <a href="https://r4ds.had.co.nz/">https://r4ds.had.co.nz/</a>.</span>)</span> and <em>Bioinformatics Data Skills</em> <span class="citation">(<label for="tufte-mn-8" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-8" class="margin-toggle">Buffalo 2015<span class="marginnote">Buffalo, Vince. 2015. <em>Bioinformatics Data Skills</em>. O’Reilly Media, Inc.</span>)</span>.</p>
<p>The <a href="https://www.rstudio.com/resources/cheatsheets/">RStudio Cheat
Sheets</a> are also a
handy resource and readers will be pointed to specific sheets in the
respective chapters.</p>
<p>This course is being taught by Prof Laurent Gatto with invaluable
assistance from Mr Jean Fain (since 2019), Ms Valetine Robaux (since
2020), Ms Julie Devis (since 2022), Dr Axelle Loriot (2018 - 2021) and
Mr Kevin Missault (2018 - 2020) at the Faculty of Pharmacy and
Biomedical Sciences (FASB) at the UCLouvain, Belgium.</p>
</div>
<div id="pre-requisites" class="section level2 unnumbered">
<h2>Pre-requisites<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('pre-requisites')" onmouseout="reset_tooltip('pre-requisites-tooltip')"><span class="tooltiptext" id="pre-requisites-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>There are no programming or technical pre-requisities for this course,
other than basic computer usage, such as general knowledge about files
(binary and text files) and folders and as well as downloading
files. Familiarity with a spreadsheet editor is helpful for the first
chapter.</p>
<p>Software requirements are documented in the <em>Setup</em> section below.</p>
</div>
<div id="about-this-course-material" class="section level2 unnumbered">
<h2>About this course material<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('about-this-course-material')" onmouseout="reset_tooltip('about-this-course-material-tooltip')"><span class="tooltiptext" id="about-this-course-material-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>This material is written in R markdown <span class="citation">(<label for="tufte-mn-9" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-9" class="margin-toggle">Allaire et al. 2024<span class="marginnote">Allaire, JJ, Yihui Xie, Christophe Dervieux, Jonathan McPherson, Javier Luraschi, Kevin Ushey, Aron Atkins, et al. 2024. <em>Rmarkdown: Dynamic Documents for r</em>. <a href="https://github.com/rstudio/rmarkdown">https://github.com/rstudio/rmarkdown</a>.</span>)</span> and compiled as a
book using <code>knitr</code> <span class="citation">(<label for="tufte-mn-10" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-10" class="margin-toggle">Xie 2025b<span class="marginnote">Xie, Yihui. 2025b. <em>Knitr: A General-Purpose Package for Dynamic Report Generation in r</em>. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.</span>)</span> <code>bookdown</code> <span class="citation">(<label for="tufte-mn-11" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-11" class="margin-toggle">Xie 2025a<span class="marginnote">Xie, Yihui. 2025a. <em>Bookdown: Authoring Books and Technical Documents with r Markdown</em>. <a href="https://github.com/rstudio/bookdown">https://github.com/rstudio/bookdown</a>.</span>)</span>. The source
code is publicly available in a Github repository
<a href="https://github.com/uclouvain-cbio/WSBIM1207">https://github.com/uclouvain-cbio/WSBIM1207</a>
and the compiled material can be read at <a href="http://bit.ly/WSBIM1207" class="uri">http://bit.ly/WSBIM1207</a>.</p>
<p>Contributions to this material are welcome. The best way to contribute
or contact the maintainers is by means of pull requests and
<a href="https://github.com/uclouvain-cbio/WSBIM1207/issues">issues</a>. Please
familiarise yourself with the <a href="https://github.com/UCLouvain-CBIO/WSBIM1207/blob/master/CONDUCT.md">code of
conduct</a>. By
participating in this project you agree to abide by its terms.</p>
</div>
<div id="citation" class="section level2 unnumbered">
<h2>Citation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('citation')" onmouseout="reset_tooltip('citation-tooltip')"><span class="tooltiptext" id="citation-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>If you use this course, please cite it as</p>
<blockquote>
<p>Laurent Gatto, Kevin Missault, Manon Martin &amp; Axelle Loriot. (2021,
September 24). UCLouvain-CBIO/WSBIM1207: Introduction to
bioinformatics (Version
v2.0.0). Zenodo. DOI: 10.5281/zenodo.5532658</p>
</blockquote>
<p><a href="https://zenodo.org/badge/latestdoi/147494586"><img src="https://zenodo.org/badge/147494586.svg" alt="DOI"></a></p>
</div>
<div id="license" class="section level2 unnumbered">
<h2>License<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('license')" onmouseout="reset_tooltip('license-tooltip')"><span class="tooltiptext" id="license-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>This material is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/">Creative Commons
Attribution-ShareAlike 4.0
License</a>.</p>
</div>
<div id="setup" class="section level2 unnumbered">
<h2>Setup<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('setup')" onmouseout="reset_tooltip('setup-tooltip')"><span class="tooltiptext" id="setup-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>For chapter <a href="sec-dataorg.html#sec-dataorg">1</a> about <em>Data organisation with
Spreadsheets</em>, a spreadsheet programme is necessary.</p>
<p>We will be using the <a href="https://www.r-project.org/">R environment for statistical
computing</a> as main data science language.
We will also use the
<a href="https://www.rstudio.com/products/RStudio/">RStudio</a> interface to
interact with R and write scripts and reports. Both R and RStudio are
easy to install and works on all major operating systems.</p>
<p>Once R and RStudio are installed, a set of packages will need to be
installed. See section <a href="session-information.html#sec-setup2">13.1</a> for details.</p>

</div>
</div>
<p style="text-align: center;">
<a href="sec-dataorg.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2025-06-16
 using 
R version 4.5.0 (2025-04-11)
</p>
</div>
</div>



</body>
</html>
