<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 9 Bioinformatics | Introduction to bioinformatics" />
<meta property="og:type" content="book" />

<meta property="og:description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain." />


<meta name="author" content="Laurent Gatto" />

<meta name="date" content="2025-06-16" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="/usr/share/javascript/mathjax/MathJax.js" type="text/javascript"></script>

<meta name="description" content="Course material for the Introduction to bioinformatics (WSBIM1207) course at UCLouvain.">

<title>Chapter 9 Bioinformatics | Introduction to bioinformatics</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script>
function copy_link(id) {
  var dummy = document.createElement('input'),
  text = window.location.href.split(/[?#]/)[0] + '#' + id;
  document.body.appendChild(dummy);
  dummy.value = text;
  dummy.select();
  document.execCommand('copy');
  document.body.removeChild(dummy);
  
  var tooltip = document.getElementById(id + '-tooltip');
  tooltip.innerHTML = 'Copied!';
}

function reset_tooltip(id) {
  var tooltip = document.getElementById(id);
  tooltip.innerHTML = 'Copy link';
}
</script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }

code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Introduction to bioinformatics<p><p class="author">Laurent Gatto</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html" id="toc-preamble">Preamble</a>
<a href="sec-dataorg.html" id="toc-sec-dataorg"><span class="toc-section-number">1</span> Data organisation with Spreadsheets</a>
<a href="sec-rrstudio.html" id="toc-sec-rrstudio"><span class="toc-section-number">2</span> R and RStudio</a>
<a href="sec-startr.html" id="toc-sec-startr"><span class="toc-section-number">3</span> Introduction to R</a>
<a href="sec-startdata.html" id="toc-sec-startdata"><span class="toc-section-number">4</span> Starting with data</a>
<a href="sec-dplyr.html" id="toc-sec-dplyr"><span class="toc-section-number">5</span> Manipulating and analyzing data with dplyr</a>
<a href="sec-vis.html" id="toc-sec-vis"><span class="toc-section-number">6</span> Data visualization</a>
<a href="sec-join.html" id="toc-sec-join"><span class="toc-section-number">7</span> Joining tables</a>
<a href="sec-rr.html" id="toc-sec-rr"><span class="toc-section-number">8</span> Reproducible research</a>
<a id="active-page" href="sec-bioinfo.html" id="toc-sec-bioinfo"><span class="toc-section-number">9</span> Bioinformatics</a><ul class="toc-sections">
<li class="toc"><a href="#omics-data"> Omics data</a></li>
<li class="toc"><a href="#metadata-and-experimental-design"> Metadata and experimental design</a></li>
<li class="toc"><a href="#sec-bioconductor"> The Bioconductor project</a></li>
<li class="toc"><a href="#omics-data-containers"> Omics data containers</a></li>
<li class="toc"><a href="#bioconductor-data-infrastructure"> Bioconductor data infrastructure</a></li>
<li class="toc"><a href="#navigating-the-bioconductor-project"> Navigating the Bioconductor project</a></li>
<li class="toc"><a href="#exercises"> Exercises</a></li>
</ul>
<a href="sec-prog.html" id="toc-sec-prog"><span class="toc-section-number">10</span> Additional programming concepts</a>
<a href="sec-ccl.html" id="toc-sec-ccl"><span class="toc-section-number">11</span> Conclusions</a>
<a href="sec-anx.html" id="toc-sec-anx"><span class="toc-section-number">12</span> Annex</a>
<a href="session-information.html" id="toc-session-information"><span class="toc-section-number">13</span> Session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="sec-bioinfo" class="section level1" number="9">
<h1>
<span class="header-section-number">Chapter 9</span> Bioinformatics</h1>
<p>As already alluded to earlier, <a href="https://en.wikipedia.org/wiki/Bioinformatics">Wikipedia
defines</a> bioinformatics
as</p>
<blockquote>
<p>Bioinformatics is an interdisciplinary field that develops methods
and software tools for understanding biological data.</p>
</blockquote>
<p>Bioinformatics is as varied as biology itself, and ranges from data
analysis, to software development, computational or statistical
methodological development, more theoretical work, as well as any
combination of these.</p>
<div id="omics-data" class="section level2" number="9.1">
<h2>
<span class="header-section-number">9.1</span> Omics data<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('omics-data')" onmouseout="reset_tooltip('omics-data-tooltip')"><span class="tooltiptext" id="omics-data-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>So far, we have explored broad data science techniques in R. A
widespread and successful area of bioinformatics, and one that you, as
a biology or biomedical science student are likely to be confronted
with, is the analysis and interpretation of omics data.</p>
<p>
<span class="marginnote shownote">
<!--
<div class="figure">--><span style="display:block;" id="fig:infoflow"></span>
<img src="figs/Centraldogma_nodetails.png" alt="Information flow in biological systems (Source [Wikipedia](https://en.wikipedia.org/wiki/Central_dogma_of_molecular_biology))." width="368"><!--
<p class="caption marginnote">-->Figure 9.1: Information flow in biological systems (Source <a href="https://en.wikipedia.org/wiki/Central_dogma_of_molecular_biology">Wikipedia</a>).<!--</p>-->
<!--</div>--></span>
</p>
<p>It is useful to define these omics data along the flow of information
in biology (Figure <a href="sec-bioinfo.html#fig:infoflow">9.1</a>), and define the different
application domains. The technologies that focus on DNA, and the
genome in particular (either whole or parts thereof) are termed
<strong>genomics</strong>, and are currently based on sequencing, in particular
high throughput sequencing (HTS). The domain focusing on the study of
any DNA (or associated proteins) modification (such as for example
methylation) is termed <strong>epigenetics</strong>. The study of RNA, and more
specifically the quantitation of RNA levels in biological samples is
termed <strong>transcriptomics</strong>, as it assays the transcription of DNA into
RNA molecules. Without further specification, transcriptomics refers
to the quantitation of message RNA, although one could also focus on
non-coding RNAs such as micro RNAs. HTS is currently the technology of
choice for any transcriptomics study, while a decade ago, prior to the
development of RNA sequencing (called <strong>RNA-Seq</strong>), microarrays were
widely used. <strong>Proteomics</strong> focuses on the identification and
quantitation of proteins, and can also expand into the study of
protein interactions, post-translational modifications or sub-cellular
localisation of proteins. Further downstream of proteins, small
molecules or lipids can also be assayed under the umbrella terms of
<strong>metabolomics</strong> and <strong>lipidomics</strong>. The technology of choice for
protein, lipids or smaller metabolites is mass spectrometry.</p>
<p>In the next couple of sections, some important concepts related to
omics data and their analysis are repeated and emphasised.</p>
<div id="high-throughput" class="section level3 unnumbered">
<h3>High throughput<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('high-throughput')" onmouseout="reset_tooltip('high-throughput-tooltip')"><span class="tooltiptext" id="high-throughput-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>By it very nature, omics data is high throughput. The goal is to
measure all, or as many as possible molecules of an omics-domain as
possible: sequence the whole genome or all exomes; identify all
epigenetic histone modifications (defining the compactness of DNA and
hence it’s accessibility by the transcription machinery); identify and
quantify as much as possible from the complete proteomics; etc. As a
result, omics data is both large in size and complex in nature, and
requires dedicated software and analysis methods to be processed,
analysed to infer biologically relevant patterns.</p>
</div>
<div id="raw-and-processed-data" class="section level3 unnumbered">
<h3>Raw and processed data<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('raw-and-processed-data')" onmouseout="reset_tooltip('raw-and-processed-data-tooltip')"><span class="tooltiptext" id="raw-and-processed-data-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>The omics data that are produced by the instruments are called raw
data, and their size (generally large), the types of file, and
structure will depend on the technology that is used. Raw data need to
be processed using dedicated software before obtaining data that can
be mapped to the biology that is measured. Below we illustrate two
such examples using Sanger sequencing and mass spectrometry.</p>
<p>In Sanger sequencing (Figure <a href="sec-bioinfo.html#fig:sangerseq">9.2</a>), DNA is labelled
using fluorophores, and different nucleotides are marked with
different colours. Upon acquisition, light signal is acquired and
recording of the different colours can be used to reconstruct the DNA
sequence.</p>
<div class="figure">
<span style="display:block;" id="fig:sangerseq"></span>
<p class="caption marginnote shownote">
Figure 9.2: Processing Sanger sequencing data to a string. (Source <a href="https://bitesizebio.com/27985/sanger-sequencing-genome-won/">BiteSizeBio</a>).
</p>
<img src="figs/sanger-sequencing.jpg" alt="Processing Sanger sequencing data to a string. (Source [BiteSizeBio](https://bitesizebio.com/27985/sanger-sequencing-genome-won/))." width="70%">
</div>
<p>In mass spectrometry, charged molecules are separated based on their
mass-to-charge (M/Z) ratio and their intensities recorded to produce a
spectrum. In proteomics, the molecules that are assayed are protein
fragments called peptides. Upon fragmentation of peptides, the
different between the M/Z peaks of the peptide fragment ions can be
used to reconstruct the peptide sequence (Figure <a href="sec-bioinfo.html#fig:pepseq">9.3</a>).</p>
<div class="figure">
<span style="display:block;" id="fig:pepseq"></span>
<p class="caption marginnote shownote">
Figure 9.3: De novo peptide sequencing using mass spectrometry. (Source <a href="https://www.creative-proteomics.com/services/de-novo-peptides-proteins-sequencing-service.htm">Creative Proteomics</a>).
</p>
<img src="figs/de-novo-pep-sequencing.jpg" alt="De novo peptide sequencing using mass spectrometry. (Source [Creative Proteomics](https://www.creative-proteomics.com/services/de-novo-peptides-proteins-sequencing-service.htm))." width="100%">
</div>
<p>The size and computational cost of processing raw data often require
more serious hardware, including disk space, computer clusters with
100s or more of compute nodes and/or access to high amounts of memory
(RAM).</p>
<p>Processed data themselves often need to be further transformed to
account for a variety of noise that is inherent to sample
collection, preparation and measurement acquisition. Data processing
and transformation will be explored in detail in subsequent course
such as <em>Omics data analysis</em>
(<a href="https://github.com/UCLouvain-CBIO/WSBIM2122">WSBIM2122</a>).</p>
</div>
</div>
<div id="metadata-and-experimental-design" class="section level2" number="9.2">
<h2>
<span class="header-section-number">9.2</span> Metadata and experimental design<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('metadata-and-experimental-design')" onmouseout="reset_tooltip('metadata-and-experimental-design-tooltip')"><span class="tooltiptext" id="metadata-and-experimental-design-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>The acquired data, even once processed, is still of very little use
when it comes to understanding biology. Before samples are collected
and data are generated, it is essential to carefully design a question
of interest (research hypothesis) and the experiment that will allow
to answer it. For example, if we want to understand the effect of a
particular drug on cancer cells, and more specifically understand the
effect on the transcription of all the expressed genes, on would need
to measure gene expression (using for example RNA-Seq) in cancer cells
in presence and absence of that drug. The table below describes a
simple experimental design where 3 conditions (control, drug at a
concentrations of 1 and 5) have been simultaneously processed and
measured by the same operator in 4 replicate.</p>
<table>
<thead><tr class="header">
<th align="left">sample</th>
<th align="left">operator</th>
<th align="left">date</th>
<th align="left">group</th>
<th align="left">concentration</th>
<th align="right">replicate</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">S1</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">CTRL</td>
<td align="left">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">S2</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">CTRL</td>
<td align="left">0</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">S3</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">CTRL</td>
<td align="left">0</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">S4</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">CTRL</td>
<td align="left">0</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">S5</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">DRUG</td>
<td align="left">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">S6</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">DRUG</td>
<td align="left">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">S7</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">DRUG</td>
<td align="left">1</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">S8</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">DRUG</td>
<td align="left">1</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">S9</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">DRUG</td>
<td align="left">5</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">S10</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">DRUG</td>
<td align="left">5</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">S11</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">DRUG</td>
<td align="left">5</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">S12</td>
<td align="left">Kevin</td>
<td align="left">2019-03-02</td>
<td align="left">DRUG</td>
<td align="left">5</td>
<td align="right">4</td>
</tr>
</tbody>
</table>
<p>We have seen a much more complex experimental design, involving many
more samples with the <code>clinical1</code> data.</p>
<pre><code>## # A tibble: 516 × 15
##    patientID    tumor_tissue_site gender age_at_diagnosis vital_status
##    &lt;chr&gt;        &lt;chr&gt;             &lt;chr&gt;             &lt;int&gt;        &lt;int&gt;
##  1 TCGA-05-4249 lung              male              24532            0
##  2 TCGA-05-4382 lung              male              24868            0
##  3 TCGA-05-4384 lung              male              24411            0
##  4 TCGA-05-4389 lung              male              25660            0
##  5 TCGA-05-4390 lung              female            21430            0
##  6 TCGA-05-4395 lung              male              27971            1
##  7 TCGA-05-4396 lung              male              28094            1
##  8 TCGA-05-4398 lung              female            17471            0
##  9 TCGA-05-4402 lung              female            20819            1
## 10 TCGA-05-4403 lung              male              27881            0
## # ℹ 506 more rows
## # ℹ 10 more variables: days_to_death &lt;int&gt;, days_to_last_followup &lt;int&gt;,
## #   pathologic_stage &lt;chr&gt;, pathology_T_stage &lt;chr&gt;, pathology_N_stage &lt;chr&gt;,
## #   pathology_M_stage &lt;chr&gt;, smoking_history &lt;chr&gt;,
## #   number_pack_years_smoked &lt;dbl&gt;, year_of_tobacco_smoking_onset &lt;int&gt;,
## #   stopped_smoking_year &lt;int&gt;</code></pre>
<p>When performing experiments, measurements should also be repeated
several times (typically at least three), to quantify the overall
variability (technical and biological) in the measured variables and,
eventually, identify changes that relate to the conditions of interest
(for example differences in genes expression in the presence or
absence of the drug).</p>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-227"></span>
<p class="caption marginnote shownote">
Figure 9.4: Distribution of the expression of the genes A1CF, BRCA1 and TP53 under the control (no drug) and drug at concentrations 1 and 5.
</p>
<img src="WSBIM1207_files/figure-html/unnamed-chunk-227-1.png" alt="Distribution of the expression of the genes A1CF, BRCA1 and TP53 under the control (no drug) and drug at concentrations 1 and 5." width="672">
</div>
</div>
<div id="sec-bioconductor" class="section level2" number="9.3">
<h2>
<span class="header-section-number">9.3</span> The Bioconductor project<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-bioconductor')" onmouseout="reset_tooltip('sec-bioconductor-tooltip')"><span class="tooltiptext" id="sec-bioconductor-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>The <a href="http://www.bioconductor.org">Bioconductor</a> was initiated by
Robert Gentleman (<span class="citation">R. C. Gentleman et al. (<label for="tufte-mn-26" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-26" class="margin-toggle">2004<span class="marginnote">Gentleman, Robert C., Vincent J. Carey, Douglas M. Bates, Ben Bolstad, Marcel Dettling, Sandrine Dudoit, Byron Ellis, et al. 2004. <span>“Bioconductor: Open Software Development for Computational Biology and Bioinformatics.”</span> <em>Genome Biol</em> 5 (10): –80. <a href="https://doi.org/10.1186/gb-2004-5-10-r80">https://doi.org/10.1186/gb-2004-5-10-r80</a>.</span>)</span>;<span class="citation">Huber et al. (<label for="tufte-mn-27" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-27" class="margin-toggle">2015<span class="marginnote">Huber, W, V J Carey, R Gentleman, S Anders, M Carlson, B S Carvalho, H C Bravo, et al. 2015. <span>“Orchestrating High-Throughput Genomic Analysis with <span>Bioconductor</span>.”</span> <em>Nat Methods</em> 12 (2): 115–21. <a href="https://doi.org/10.1038/nmeth.3252">https://doi.org/10.1038/nmeth.3252</a>.</span>)</span>), one of the two
creators of the R language, and centrally offers dedicated R packages
for bioinformatics.</p>
<blockquote>
<p>Bioconductor provides tools for the analysis and comprehension of
high-throughput genomic data. Bioconductor uses the R statistical
programming language, and is open source and open development. It
has two releases each year, and an active user community.</p>
</blockquote>
<div class="figure">
<span style="display:block;" id="fig:biocwww"></span>
<p class="caption marginnote shownote">
Figure 9.5: The Bioconductor web page.
</p>
<img src="figs/bioc-screenshot.png" alt="The Bioconductor web page." width="100%">
</div>
<p>This <a href="https://www.youtube.com/watch?v=nzY7bPQOXUs">video</a> provides a
great overview of the project at large.</p>
<p>Bioconductor packages are managed installed using a dedicated package,
namely <code>BiocManager</code>, that can be installed from CRAN with</p>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="sec-bioinfo.html#cb583-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span></code></pre></div>
<p>Individuals package such as <code>SummarizedExperiment</code> (see below for
details), <code>DESeq2</code> (for transcriptomics), <code>Spectra</code> (for mass
spectrometry), <code>xcms</code> (metabolomics), … can then be installed with
<code>BiocManager::install</code>.</p>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="sec-bioinfo.html#cb584-1" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"SummarizedExperiment"</span>)</span>
<span id="cb584-2"><a href="sec-bioinfo.html#cb584-2" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"DESeq2"</span>)</span>
<span id="cb584-3"><a href="sec-bioinfo.html#cb584-3" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"Spectra"</span>)</span>
<span id="cb584-4"><a href="sec-bioinfo.html#cb584-4" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"xcms"</span>)</span></code></pre></div>
<p>Note that we can also use that same function to install packages from GitHub:</p>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="sec-bioinfo.html#cb585-1" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"UCLouvain-CBIO/rWSBIM1207"</span>)</span></code></pre></div>
</div>
<div id="omics-data-containers" class="section level2" number="9.4">
<h2>
<span class="header-section-number">9.4</span> Omics data containers<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('omics-data-containers')" onmouseout="reset_tooltip('omics-data-containers-tooltip')"><span class="tooltiptext" id="omics-data-containers-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>Data in bioinformatics is often more complex than the basic data types
we have seen so far. In such situations, developers define specialised
data containers (termed classes that) that match the properties of the
data they need to handle.</p>
<p>An example of general data architecture, that is used across many
omics domains in Bioconductor is represented below:</p>
<div class="figure">
<span style="display:block;" id="fig:msnset"></span>
<p class="caption marginnote shownote">
Figure 9.6: A data structure to store quantitative data, features (rows) annotation, and samples (column) annotations..
</p>
<img src="figs/msnset.png" alt="A data structure to store quantitative data, features (rows) annotation, and samples (column) annotations.." width="80%">
</div>
<ul>
<li><p>An assay data slot containing the quantitative omics data
(expression data), stored as a <code>matrix</code>. Features (genes,
transcripts, proteins, …) are defined along the rows and samples
along the columns.</p></li>
<li><p>A sample metadata slot containing sample co-variates, stored as a
table (<code>data.frame</code> or <code>DataFrame</code>). This dataframe is stored with
rows representing samples and sample covariate along the columns,
and its rows match the expression data columns exactly.</p></li>
<li><p>A feature metadata slot containing feature co-variates, stored as
table annotated (<code>data.frame</code> or <code>DataFrame</code>). This dataframe’s rows
match the expression data rows exactly.</p></li>
</ul>
<p>The coordinated nature of the high throughput data guarantees that the
dimensions of the different slots will always match (i.e the columns
in the expression data and then rows in the sample metadata, as well
as the rows in the expression data and feature metadata) during data
manipulation. The metadata slots can grow additional co-variates
(columns) without affecting the other structures.</p>
<p>To illustrate such an omics data container, we’ll use a variable of
class <code>SummarizedExperiment</code>. Below, we load the <code>GSE96870_intro</code>
dataset from the <code>rWSBIM1207</code> package (version &gt;= 0.1.15).</p>
<div class="sourceCode" id="cb586"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb586-1"><a href="sec-bioinfo.html#cb586-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"SummarizedExperiment"</span>)</span>
<span id="cb586-2"><a href="sec-bioinfo.html#cb586-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"rWSBIM1207"</span>)</span>
<span id="cb586-3"><a href="sec-bioinfo.html#cb586-3" tabindex="-1"></a><span class="fu">data</span>(GSE96870_intro)</span>
<span id="cb586-4"><a href="sec-bioinfo.html#cb586-4" tabindex="-1"></a><span class="fu">class</span>(GSE96870_intro)</span></code></pre></div>
<pre><code>## [1] "RangedSummarizedExperiment"
## attr(,"package")
## [1] "SummarizedExperiment"</code></pre>
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="sec-bioinfo.html#cb588-1" tabindex="-1"></a>GSE96870_intro</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 1474 22 
## metadata(0):
## assays(1): counts
## rownames(1474): Asl Apod ... Pbx1 Rgs4
## rowData names(11): gene ENTREZID ... phenotype_description
##   hsapiens_homolog_associated_gene_name
## colnames(22): GSM2545336 GSM2545337 ... GSM2545363 GSM2545380
## colData names(10): title geo_accession ... tissue mouse</code></pre>
<p>This data contains the same RNA-sequencing data as we have seen in
chapter <a href="sec-startdata.html#sec-startdata">4</a>. It is however formatted as a special
type of data, a <code>SummarizedExperiment</code>, that is specialised for
quantitative omics data (as opposed to the more general <code>data.frame</code>
data structure). We will be learning and using <code>SummarizedExperiment</code>
objects a lot in the
<a href="https://github.com/UCLouvain-CBIO/WSBIM1322">WSBIM1322</a> and
<a href="https://github.com/UCLouvain-CBIO/WSBIM2122">WSBIM2122</a> courses.</p>
<p>The object contains data for 1474 features
(peptides in this case) and 22 samples.</p>
<div class="sourceCode" id="cb590"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb590-1"><a href="sec-bioinfo.html#cb590-1" tabindex="-1"></a><span class="fu">dim</span>(GSE96870_intro)</span></code></pre></div>
<pre><code>## [1] 1474   22</code></pre>
<div class="sourceCode" id="cb592"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb592-1"><a href="sec-bioinfo.html#cb592-1" tabindex="-1"></a><span class="fu">nrow</span>(GSE96870_intro)</span></code></pre></div>
<pre><code>## [1] 1474</code></pre>
<div class="sourceCode" id="cb594"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb594-1"><a href="sec-bioinfo.html#cb594-1" tabindex="-1"></a><span class="fu">ncol</span>(GSE96870_intro)</span></code></pre></div>
<pre><code>## [1] 22</code></pre>
<p>The samples (columns) and rows (protein features) are named:</p>
<div class="sourceCode" id="cb596"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb596-1"><a href="sec-bioinfo.html#cb596-1" tabindex="-1"></a><span class="fu">colnames</span>(GSE96870_intro)</span></code></pre></div>
<pre><code>##  [1] "GSM2545336" "GSM2545337" "GSM2545338" "GSM2545339" "GSM2545340"
##  [6] "GSM2545341" "GSM2545342" "GSM2545343" "GSM2545344" "GSM2545345"
## [11] "GSM2545346" "GSM2545347" "GSM2545348" "GSM2545349" "GSM2545350"
## [16] "GSM2545351" "GSM2545352" "GSM2545353" "GSM2545354" "GSM2545362"
## [21] "GSM2545363" "GSM2545380"</code></pre>
<div class="sourceCode" id="cb598"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb598-1"><a href="sec-bioinfo.html#cb598-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rownames</span>(GSE96870_intro))</span></code></pre></div>
<pre><code>## [1] "Asl"     "Apod"    "Cyp2d22" "Klk6"    "Fcrls"   "Slc2a4"</code></pre>
<div class="sourceCode" id="cb600"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb600-1"><a href="sec-bioinfo.html#cb600-1" tabindex="-1"></a><span class="fu">tail</span>(<span class="fu">rownames</span>(GSE96870_intro))</span></code></pre></div>
<pre><code>## [1] "Mgst3"  "Lrrc52" "Rxrg"   "Lmx1a"  "Pbx1"   "Rgs4"</code></pre>
<p>Using this data structure, we can access the expression matrix with
the <code>assay</code> function, the feature metadata with the <code>rowData</code> function,
and the sample metadata with the <code>colData</code> function:</p>
<div class="sourceCode" id="cb602"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb602-1"><a href="sec-bioinfo.html#cb602-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">assay</span>(GSE96870_intro))</span></code></pre></div>
<pre><code>##         GSM2545336 GSM2545337 GSM2545338 GSM2545339 GSM2545340 GSM2545341
## Asl           1170        361        400        586        626        988
## Apod         36194      10347       9173      10620      13021      29594
## Cyp2d22       4060       1616       1603       1901       2171       3349
## Klk6           287        629        641        578        448        195
##         GSM2545342 GSM2545343 GSM2545344 GSM2545345 GSM2545346 GSM2545347
## Asl            836        535        586        597        938       1035
## Apod         24959      13668      13230      15868      27769      34301
## Cyp2d22       3122       2008       2254       2277       2985       3452
## Klk6           186       1101        537        567        327        233
##         GSM2545348 GSM2545349 GSM2545350 GSM2545351 GSM2545352 GSM2545353
## Asl            494        481        666        937        803        541
## Apod         11258      11812      15816      29242      20415      13682
## Cyp2d22       1883       2014       2417       3678       2920       2216
## Klk6           742        881        828        250        798        710
##         GSM2545354 GSM2545362 GSM2545363 GSM2545380
## Asl            473        748        576       1192
## Apod         11088      15916      11166      38148
## Cyp2d22       1821       2842       2011       4019
## Klk6           894        501        598        259
##  [ reached 'max' / getOption("max.print") -- omitted 2 rows ]</code></pre>
<div class="sourceCode" id="cb604"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb604-1"><a href="sec-bioinfo.html#cb604-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rowData</span>(GSE96870_intro))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 11 columns
##                gene    ENTREZID                product       gbkey
##         &lt;character&gt; &lt;character&gt;            &lt;character&gt; &lt;character&gt;
## Asl             Asl      109900 argininosuccinate ly..        mRNA
## Apod           Apod       11815 apolipoprotein D, tr..        mRNA
## Cyp2d22     Cyp2d22       56448 cytochrome P450, fam..        mRNA
## Klk6           Klk6       19144 kallikrein related-p..        mRNA
## Fcrls         Fcrls       80891 Fc receptor-like S, ..        mRNA
## Slc2a4       Slc2a4       20528 solute carrier famil..        mRNA
##         external_gene_name    ensembl_gene_id external_synonym chromosome_name
##                &lt;character&gt;        &lt;character&gt;      &lt;character&gt;     &lt;character&gt;
## Asl                    Asl ENSMUSG00000025533    2510006M18Rik               5
## Apod                  Apod ENSMUSG00000022548               NA              16
## Cyp2d22            Cyp2d22 ENSMUSG00000061740             2D22              15
## Klk6                  Klk6 ENSMUSG00000050063             Bssp               7
## Fcrls                Fcrls ENSMUSG00000015852    2810439C17Rik               3
## Slc2a4              Slc2a4 ENSMUSG00000018566           Glut-4              11
##           gene_biotype  phenotype_description
##            &lt;character&gt;            &lt;character&gt;
## Asl     protein_coding abnormal circulating..
## Apod    protein_coding abnormal lipid homeo..
## Cyp2d22 protein_coding abnormal skin morpho..
## Klk6    protein_coding abnormal cytokine le..
## Fcrls   protein_coding decreased CD8-positi..
## Slc2a4  protein_coding abnormal circulating..
##         hsapiens_homolog_associated_gene_name
##                                   &lt;character&gt;
## Asl                                       ASL
## Apod                                     APOD
## Cyp2d22                                CYP2D6
## Klk6                                     KLK6
## Fcrls                                   FCRL4
## Slc2a4                                 SLC2A4</code></pre>
<div class="sourceCode" id="cb606"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb606-1"><a href="sec-bioinfo.html#cb606-1" tabindex="-1"></a><span class="fu">colData</span>(GSE96870_intro)</span></code></pre></div>
<pre><code>## DataFrame with 22 rows and 10 columns
##                      title geo_accession     organism         age      sex
##                &lt;character&gt;   &lt;character&gt;  &lt;character&gt; &lt;character&gt; &lt;factor&gt;
## GSM2545336 CNS_RNA-seq_10C    GSM2545336 Mus musculus     8 weeks   Female
## GSM2545337 CNS_RNA-seq_11C    GSM2545337 Mus musculus     8 weeks   Female
## GSM2545338 CNS_RNA-seq_12C    GSM2545338 Mus musculus     8 weeks   Female
## GSM2545339 CNS_RNA-seq_13C    GSM2545339 Mus musculus     8 weeks   Female
## GSM2545340 CNS_RNA-seq_14C    GSM2545340 Mus musculus     8 weeks   Male  
## ...                    ...           ...          ...         ...      ...
## GSM2545353  CNS_RNA-seq_3C    GSM2545353 Mus musculus     8 weeks   Female
## GSM2545354  CNS_RNA-seq_4C    GSM2545354 Mus musculus     8 weeks   Male  
## GSM2545362  CNS_RNA-seq_5C    GSM2545362 Mus musculus     8 weeks   Female
##              infection      strain     time     tissue    mouse
##               &lt;factor&gt; &lt;character&gt; &lt;factor&gt;   &lt;factor&gt; &lt;factor&gt;
## GSM2545336 InfluenzaA      C57BL/6     Day8 Cerebellum       14
## GSM2545337 NonInfected     C57BL/6     Day0 Cerebellum       9 
## GSM2545338 NonInfected     C57BL/6     Day0 Cerebellum       10
## GSM2545339 InfluenzaA      C57BL/6     Day4 Cerebellum       15
## GSM2545340 InfluenzaA      C57BL/6     Day4 Cerebellum       18
## ...                ...         ...      ...        ...      ...
## GSM2545353 NonInfected     C57BL/6     Day0 Cerebellum       4 
## GSM2545354 NonInfected     C57BL/6     Day0 Cerebellum       2 
## GSM2545362 InfluenzaA      C57BL/6     Day4 Cerebellum       20
##  [ reached 'max' / getOption("max.print") -- omitted 2 rows ]</code></pre>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Verify that the expression data dimensions match with number of rows
and columns in the feature and sample data.</p>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="msmb-solution">
<p class="solution-begin">
► Solution<span id="sol-start-48" class="fa fa-plus-square solution-icon clickable" onclick="toggle_visibility('sol-body-48', 'sol-start-48')"></span>
</p>
<div id="sol-body-48" class="solution-body" style="display: none;">
<div class="sourceCode" id="cb608"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb608-1"><a href="sec-bioinfo.html#cb608-1" tabindex="-1"></a><span class="fu">nrow</span>(<span class="fu">assay</span>(GSE96870_intro)) <span class="sc">==</span> <span class="fu">nrow</span>(<span class="fu">rowData</span>(GSE96870_intro))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb610"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb610-1"><a href="sec-bioinfo.html#cb610-1" tabindex="-1"></a><span class="fu">ncol</span>(<span class="fu">assay</span>(GSE96870_intro)) <span class="sc">==</span> <span class="fu">nrow</span>(<span class="fu">colData</span>(GSE96870_intro))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p class="solution-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<p>We can use the <code>[</code> operator to subset the whole object: all parts
thereof will be subset correctly.</p>
<div class="sourceCode" id="cb612"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb612-1"><a href="sec-bioinfo.html#cb612-1" tabindex="-1"></a>small_se <span class="ot">&lt;-</span> GSE96870_intro[<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">5</span>), <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">4</span>)]</span>
<span id="cb612-2"><a href="sec-bioinfo.html#cb612-2" tabindex="-1"></a><span class="fu">dim</span>(small_se)</span></code></pre></div>
<pre><code>## [1] 3 2</code></pre>
<div class="sourceCode" id="cb614"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb614-1"><a href="sec-bioinfo.html#cb614-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">assay</span>(small_se))</span></code></pre></div>
<pre><code>##         GSM2545337 GSM2545339
## Asl            361        586
## Cyp2d22       1616       1901
## Fcrls          233        237</code></pre>
<p>We can also add information with:</p>
<div class="sourceCode" id="cb616"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb616-1"><a href="sec-bioinfo.html#cb616-1" tabindex="-1"></a><span class="fu">colData</span>(small_se)<span class="sc">$</span>new_var  <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"new_val1"</span>, <span class="st">"new_val2"</span>)</span>
<span id="cb616-2"><a href="sec-bioinfo.html#cb616-2" tabindex="-1"></a><span class="fu">colData</span>(small_se)</span></code></pre></div>
<pre><code>## DataFrame with 2 rows and 11 columns
##                      title geo_accession     organism         age      sex
##                &lt;character&gt;   &lt;character&gt;  &lt;character&gt; &lt;character&gt; &lt;factor&gt;
## GSM2545337 CNS_RNA-seq_11C    GSM2545337 Mus musculus     8 weeks   Female
## GSM2545339 CNS_RNA-seq_13C    GSM2545339 Mus musculus     8 weeks   Female
##              infection      strain     time     tissue    mouse     new_var
##               &lt;factor&gt; &lt;character&gt; &lt;factor&gt;   &lt;factor&gt; &lt;factor&gt; &lt;character&gt;
## GSM2545337 NonInfected     C57BL/6     Day0 Cerebellum       9     new_val1
## GSM2545339 InfluenzaA      C57BL/6     Day4 Cerebellum       15    new_val2</code></pre>
<div id="sec-save" class="section level3" number="9.4.1">
<h3>
<span class="header-section-number">9.4.1</span> Saving objects<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-save')" onmouseout="reset_tooltip('sec-save-tooltip')"><span class="tooltiptext" id="sec-save-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Exporting data to a spreadsheet using <code>write.csv()</code> or <code>write_csv()</code>
has several limitations, such as possible inconsistencies with <code>,</code> and
<code>.</code> for decimal separators and lack of variable type
definitions. Furthermore, exporting data to a spreadsheet is only
relevant for rectangular data such as data.frames and matrices and
isn’t applicable to more complex variables such as
<code>SummarizedExperiment</code> objects, that are themselves composed of
mulitple parts.</p>
<p>A more general way to save data, that is specific to R and is
guaranteed to work on any operating system, is to use the <code>save()</code>
function. Saving objects will generate a binary representation of the
object on disk, a <em>R Data</em> file (<code>rda</code> extension) that guarantees to
produce the same object once loaded back into R using the <code>load()</code>
function.</p>
<div class="sourceCode" id="cb618"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb618-1"><a href="sec-bioinfo.html#cb618-1" tabindex="-1"></a><span class="fu">save</span>(GSE96870_intro, <span class="at">file =</span> <span class="st">"data_output/se.rda"</span>)</span>
<span id="cb618-2"><a href="sec-bioinfo.html#cb618-2" tabindex="-1"></a><span class="fu">rm</span>(GSE96870_intro)</span>
<span id="cb618-3"><a href="sec-bioinfo.html#cb618-3" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"data_output/se.rda"</span>)</span>
<span id="cb618-4"><a href="sec-bioinfo.html#cb618-4" tabindex="-1"></a>GSE96870_intro</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 1474 22 
## metadata(0):
## assays(1): counts
## rownames(1474): Asl Apod ... Pbx1 Rgs4
## rowData names(11): gene ENTREZID ... phenotype_description
##   hsapiens_homolog_associated_gene_name
## colnames(22): GSM2545336 GSM2545337 ... GSM2545363 GSM2545380
## colData names(10): title geo_accession ... tissue mouse</code></pre>
<p>Note how the <code>load()</code> function directly loads the object in the file
directly in the global environment using its original name.</p>
<p>The <code>saveRDS()</code> and <code>readRDS()</code> functions save R objects to binary
files (using the <code>rds</code> extension here) and read these back into
R. From a user’s perspective, the main difference is that, <code>load()</code>
loads an object in the global environment while <code>readRDS()</code> reads the
data from disk and returns it. It is thus necessary to store the
output of <code>readRDS()</code> in a variable:</p>
<div class="sourceCode" id="cb620"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb620-1"><a href="sec-bioinfo.html#cb620-1" tabindex="-1"></a><span class="fu">saveRDS</span>(GSE96870_intro, <span class="at">file =</span> <span class="st">"data_output/se.rds"</span>)</span>
<span id="cb620-2"><a href="sec-bioinfo.html#cb620-2" tabindex="-1"></a><span class="fu">rm</span>(GSE96870_intro)</span>
<span id="cb620-3"><a href="sec-bioinfo.html#cb620-3" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data_output/se.rds"</span>)</span>
<span id="cb620-4"><a href="sec-bioinfo.html#cb620-4" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 1474 22 
## metadata(0):
## assays(1): counts
## rownames(1474): Asl Apod ... Pbx1 Rgs4
## rowData names(11): gene ENTREZID ... phenotype_description
##   hsapiens_homolog_associated_gene_name
## colnames(22): GSM2545336 GSM2545337 ... GSM2545363 GSM2545380
## colData names(10): title geo_accession ... tissue mouse</code></pre>
<p>When it comes to saving data from R that will be loaded again in R,
saving and loading is the preferred approach. If tabular data need to
be shared with somebody who is not using R, then exporting to a
text-based spreadsheet is a good alternative.</p>
</div>
</div>
<div id="bioconductor-data-infrastructure" class="section level2" number="9.5">
<h2>
<span class="header-section-number">9.5</span> Bioconductor data infrastructure<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('bioconductor-data-infrastructure')" onmouseout="reset_tooltip('bioconductor-data-infrastructure-tooltip')"><span class="tooltiptext" id="bioconductor-data-infrastructure-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>An essential aspect that is central to Bioconductor and its success is
the availability of core data infrastructure that is used across
packages. Package developers are advised to make use of existing
infrastructure to provide coherence, interoperability and stability to
the project as a whole.</p>
<p>Here are some core classes, taken from the <a href="https://bioconductor.org/developers/how-to/commonMethodsAndClasses/">Common Bioconductor
Methods and
Classes</a>
page:</p>
<div id="importing" class="section level4 unnumbered">
<h4>Importing</h4>
<ul>
<li>GTF, GFF, BED, BigWig, etc., - <em><a href="https://bioconductor.org/packages/3.21/rtracklayer">rtracklayer</a></em><code>::import()</code>
</li>
<li>VCF – <em><a href="https://bioconductor.org/packages/3.21/VariantAnnotation">VariantAnnotation</a></em><code>::readVcf()</code>
</li>
<li>SAM / BAM – <em><a href="https://bioconductor.org/packages/3.21/Rsamtools">Rsamtools</a></em><code>::scanBam()</code>, <em><a href="https://bioconductor.org/packages/3.21/GenomicAlignments">GenomicAlignments</a></em><code>:readGAlignment*()</code>
</li>
<li>FASTA – <em><a href="https://bioconductor.org/packages/3.21/Biostrings">Biostrings</a></em><code>::readDNAStringSet()</code>
</li>
<li>FASTQ – <em><a href="https://bioconductor.org/packages/3.21/ShortRead">ShortRead</a></em><code>::readFastq()</code>
</li>
<li>Mass spectrometry data (XML-based and peaklist formats) – <em><a href="https://bioconductor.org/packages/3.21/Spectra">Spectra</a></em><code>::Spectra()</code>
</li>
</ul>
</div>
<div id="common-classes" class="section level4 unnumbered">
<h4>Common Classes</h4>
<ul>
<li>Rectangular feature x sample data – <em><a href="https://bioconductor.org/packages/3.21/SummarizedExperiment">SummarizedExperiment</a></em><code>::SummarizedExperiment()</code> (RNAseq count matrix, microarray, proteomics, …)</li>
<li>Genomic coordinates – <em><a href="https://bioconductor.org/packages/3.21/GenomicRanges">GenomicRanges</a></em><code>::GRanges()</code> (1-based, closed interval)</li>
<li>DNA / RNA / AA sequences – <em><a href="https://bioconductor.org/packages/3.21/Biostrings">Biostrings</a></em><code>::*StringSet()</code>
</li>
<li>Gene sets – <em><a href="https://bioconductor.org/packages/3.21/GSEABase">GSEABase</a></em><code>::GeneSet()</code> <em><a href="https://bioconductor.org/packages/3.21/GSEABase">GSEABase</a></em><code>::GeneSetCollection()</code>
</li>
<li>Multi-omics data – <em><a href="https://bioconductor.org/packages/3.21/MultiAssayExperiment">MultiAssayExperiment</a></em><code>::MultiAssayExperiment()</code>
</li>
<li>Single cell data – <em><a href="https://bioconductor.org/packages/3.21/SingleCellExperiment">SingleCellExperiment</a></em><code>::SingleCellExperiment()</code>
</li>
<li>Mass spectrometry data – <em><a href="https://bioconductor.org/packages/3.21/Spectra">Spectra</a></em><code>::Spectra()</code>
</li>
</ul>
</div>
</div>
<div id="navigating-the-bioconductor-project" class="section level2" number="9.6">
<h2>
<span class="header-section-number">9.6</span> Navigating the Bioconductor project<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('navigating-the-bioconductor-project')" onmouseout="reset_tooltip('navigating-the-bioconductor-project-tooltip')"><span class="tooltiptext" id="navigating-the-bioconductor-project-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>Bioconductor has become a large project proposing many packages across
many domains of high throughput biology. It continues to grow, at an
increasing rate, and it can be difficult to get started.</p>
<div id="biocviews" class="section level3" number="9.6.1">
<h3>
<span class="header-section-number">9.6.1</span> <em>biocViews</em>
<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('biocviews')" onmouseout="reset_tooltip('biocviews-tooltip')"><span class="tooltiptext" id="biocviews-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>One way to find packages of interest is to navigate the <em>biocViews</em>
hierarchy. Every package is tagged with a set of <em>biocViews</em>
labels. The highest level defines 3 types of packages:</p>
<ul>
<li>Software: packages providing a specific functionality.</li>
<li>AnnotationData: packages providing annotations, such as various
ontologies, species annotations, microarray annotations, …</li>
<li>ExperimentData: packages distributing experiments.</li>
</ul>
<p>The <em>biocViews</em> page is available here</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/BiocViews.html#___Software" class="uri">https://bioconductor.org/packages/release/BiocViews.html#___Software</a></li>
</ul>
<p>It is most easily accessed by clicking on the <em>software packages</em> link
on the homepage, under <em>About Bioconductor</em> (see screenshot above).</p>
<p>See also this
<a href="https://bioconductor.org/developers/how-to/biocViews/">page</a> for
additional information.</p>
</div>
<div id="workflows" class="section level3" number="9.6.2">
<h3>
<span class="header-section-number">9.6.2</span> Workflows<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('workflows')" onmouseout="reset_tooltip('workflows-tooltip')"><span class="tooltiptext" id="workflows-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>On the other hand, people generally don’t approach the Bioconductor
project to learn the whole project, but are interested by a specific
analysis from a Bioconductor package, that they have read in a paper
of interest. In my opinion, it is more effective to restrict ones
attention to a problem or analysis of interest to first immerse
oneself into Bioconductor, then broaden up ones experience to other
topics and packages.</p>
<p>To to that, the project offers workflows that provide a general
introduction to topics such as sequence analysis, annotation
resources, RNA-Seq data analyis, Mass spectrometry and proteomics,
CyTOF analysis, ….</p>
<ul>
<li><a href="https://bioconductor.org/help/workflows/" class="uri">https://bioconductor.org/help/workflows/</a></li>
</ul>
<p>A similar set of resources are published
in <a href="https://f1000research.com/">F1000Research</a> under the Bioconductor
gateway</p>
<ul>
<li><a href="https://f1000research.com/gateways/bioconductor" class="uri">https://f1000research.com/gateways/bioconductor</a></li>
</ul>
<p>These peer-reviewed papers describe more complete pipelines involving
one or several packages.</p>
</div>
<div id="learning-about-specific-packages" class="section level3" number="9.6.3">
<h3>
<span class="header-section-number">9.6.3</span> Learning about specific packages<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('learning-about-specific-packages')" onmouseout="reset_tooltip('learning-about-specific-packages-tooltip')"><span class="tooltiptext" id="learning-about-specific-packages-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Each Bioconductor package has it’s own <em>landing pages</em> that provides
all necessary information about a package, including a short summary,
its current version, the authors, how the cite the package,
installation instructions, and links to all package vignettes.</p>
<p>Any Bioconductor package page can be constructed by appending the
package’s name to <code>https://bioconductor.org/packages/</code> to produce an
URL like</p>
<ul>
<li><a href="https://bioconductor.org/packages/packagename" class="uri">https://bioconductor.org/packages/packagename</a></li>
</ul>
<p>This works for any type of package (software, annotation or data). For
example, the pages for packages <em><a href="https://bioconductor.org/packages/3.21/DESeq2">DESeq2</a></em> or <em><a href="https://bioconductor.org/packages/3.21/QFeatures">QFeatures</a></em>
would be</p>
<ul>
<li><a href="https://bioconductor.org/packages/DESeq2" class="uri">https://bioconductor.org/packages/DESeq2</a></li>
</ul>
<p>and</p>
<ul>
<li><a href="https://bioconductor.org/packages/QFeatures" class="uri">https://bioconductor.org/packages/QFeatures</a></li>
</ul>
<p>These short URLs are then resolved to their longer form to redirect to
the longer package URL leading the user to the current release version
of the packge.</p>
</div>
<div id="package-vignettes" class="section level3" number="9.6.4">
<h3>
<span class="header-section-number">9.6.4</span> Package vignettes<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('package-vignettes')" onmouseout="reset_tooltip('package-vignettes-tooltip')"><span class="tooltiptext" id="package-vignettes-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>An important quality of every Bioconductor package is the availability
of a dedicated <em>vignette</em>. Vignettes are documentations (generally
provided as pdf or html files) that provide a generic overview of the
package, without necessarily going in detail for every function of the
package.</p>
<p>Below, we show how to list all vignettes available in the <code>MSnbase</code>
package and how to open one in particular.</p>
<div class="sourceCode" id="cb622"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb622-1"><a href="sec-bioinfo.html#cb622-1" tabindex="-1"></a><span class="fu">vignette</span>(<span class="at">package =</span> <span class="st">"QFeatures"</span>)</span>
<span id="cb622-2"><a href="sec-bioinfo.html#cb622-2" tabindex="-1"></a><span class="fu">vignette</span>(<span class="st">"QFeatures"</span>)</span></code></pre></div>
<p>Vignettes are special in that respect as they are produced as part of
the package building process. The code in a vignette is executed and
its output, whether in the form of simple text, tables and figures,
are inserted in the vignette before the final file (in pdf or html) is
produced. Hence, all the code and outputs are guaranteed to be correct
and reproduced.</p>
<p>Given a vignette, it is thus possible to re-generate all the
results. To make reproducing a long vignette as easy as possible
without copying and pasting all code chunks one by one, it is possible to
extract the code into an R script running the <code>Stangle</code> (from the
<code>utils</code> package -
see <a href="https://bioconductor.org/help/package-vignettes/">here</a> for
details) or <code>knitr::purl</code> functions on the vignette source document.</p>
</div>
<div id="getting-help" class="section level3" number="9.6.5">
<h3>
<span class="header-section-number">9.6.5</span> Getting help<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('getting-help')" onmouseout="reset_tooltip('getting-help-tooltip')"><span class="tooltiptext" id="getting-help-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>The best way to get help with regards to a Bioconductor package is to
post the question on the <em>Bioconductor support forum</em> at
<a href="https://support.bioconductor.org/" class="uri">https://support.bioconductor.org/</a>. Package developers generally follow
the support site for questions related to their packages. See this
page for <a href="https://bioconductor.org/help/support/">more details</a>.</p>
<p>To maximise the chances of obtaining an answer promptly, it is important
to provide details for others to understand the question and, if
possible, reproduce the observed errors. The Bioconductor project has
a dedicated
<a href="https://bioconductor.org/help/support/posting-guide/">posting guide</a>. Here’s
another useful guide on
<a href="http://adv-r.had.co.nz/Reproducibility.html">how to write a reproducible question</a>.</p>
<p>Packages come with a lot of documentation built in, that users are
advised to read to familiarise themselves with the package and how to
use it. In addition to the package vignettes as describe above, every
function of class in a package is documented in great detail in their
respective <em>main</em> page, that can be accessed with <code>?function</code>.</p>
<p>There is also a
dedicated
<a href="https://bioconductor.org/help/support/posting-guide/"><em>developer mailing list</em></a> that
is dedicated for questions and discussions related to package
development.</p>
</div>
<div id="versions" class="section level3" number="9.6.6">
<h3>
<span class="header-section-number">9.6.6</span> Versions<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('versions')" onmouseout="reset_tooltip('versions-tooltip')"><span class="tooltiptext" id="versions-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>It is also useful to know that at any given time, there are two
Bioconductor versions - there is always a release (stable) and a
development (devel) versions. For example, in October 2017, the
release version is 3.6 and the development is 3.7.</p>
<p>The individual packages have a similar scheme. Every package is
available for the release and the development versions of
Bioconductor. These two versions of the package also have different
version numbers, where the last digit is even for the former and odd
for the later. Currently, the <code>MSnbase</code> has versions <code>2.8.2</code> and
<code>2.9.3</code>, respectively.</p>
<p>Finally, every Bioconductor version is tied to an R version. To
access the current Bioconductor release, one needs to use the latest R
version. Hence, it is important to have an up-to-date R installation
to keep up with the latest developments in Bioconductor. More details
<a href="https://bioconductor.org/developers/how-to/version-numbering/">here</a>.</p>
</div>
</div>
<div id="exercises" class="section level2" number="9.7">
<h2>
<span class="header-section-number">9.7</span> Exercises<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('exercises')" onmouseout="reset_tooltip('exercises-tooltip')"><span class="tooltiptext" id="exercises-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<ol style="list-style-type: decimal">
<li><p>Install a Bioconductor package of your choice, discover the
vignette(s) it offers, open one, and extract the R code out of it.</p></li>
<li><p>Find a package that allows reading raw mass spectrometry data and
identify the specific function. Either use the biocViews tree, look
for a possible workflow, or look in the common methods and classes
page on the Bioconductor page.</p></li>
</ol>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<ol start="3" style="list-style-type: decimal">
<li><p>Load the <code>cptac_se</code> data from the <code>rWSBIM1322</code> package. Verify it
is of class <code>SummarizedExperiment</code>.</p></li>
<li><p>Extract the quantitative information for the peptides <code>AIGVLPQLIIDR</code>,
<code>NLDAAPTLR</code> and <code>YGLNHVVSLIENKK</code> for samples <code>6A_7</code> and
<code>6B_8</code>. Subsetting works as we have seen for <code>data.frames</code> in
chapter 3.</p></li>
<li><p>Look and interpret the experimental design stored in the sample
metadata of this experiment. To help you out, you can also read its
documentation.</p></li>
<li><p>What is the average expression of <code>LSAAQAELAYAETGAHDK</code> in the
groups <code>6A</code> and <code>6B</code>?</p></li>
<li><p>Calculate the average expression of all peptides belonging to
protein <code>P02753ups|RETBP_HUMAN_UPS</code> for each sample. You can
indentify which peptides to use by looking for that protein in the
object’s rowData slot.</p></li>
</ol>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<ol style="list-style-type: decimal">
<li>To be able to access the data for this exercise, make sure you have
<code>rWSBIM1207</code> version 0.1.5 or later. If needed, install a more
recent version with</li>
</ol>
<div class="sourceCode" id="cb623"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb623-1"><a href="sec-bioinfo.html#cb623-1" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"UCLouvain-CBIO/rWSBIM1207"</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li><p>Import the data from two tab-separated files into R. The full paths
to the two files can be accessed with <code>kem.tsv()</code>. Read <code>?kem</code> for
details on the content of the two files. In brief, the
<code>kem_counts.tsv</code> file contains RNA-Seq expression counts for 13
genes and 18 samples and <code>kem_annot.tsv</code> contains annotation about
each sample. Read the data into two <code>tibbles</code> names <code>kem</code> and
<code>annot</code> respectively and familiarise yourself with the content of
the two new tables.</p></li>
<li><p>Convert the counts data into a long table format and annotate
each sample using the experimental design.</p></li>
<li><p>Identify the three transcript identifiers that have the highest
expression count over all samples.</p></li>
<li><p>Visualise the distribution of the expression for the three
transcripts selected above in cell types A and B under both
treatments.</p></li>
<li><p>For all genes, calculate the mean intensities in each experimental
group (as defined by the <code>cell_type</code> and <code>treatment</code> variables).</p></li>
<li><p>Focusing only on the three most expressed transcripts and cell type
A, calculate the fold-change induced by the treatment. The
fold-change is the ratio between the average expressions in two
conditions.</p></li>
</ol>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>
<div class="question">
<p class="question-begin">
► Question
</p>
<div class="question-body">
<p>Download the following <a href="./data/se.rds">file</a> and load its data. The
data is of class <code>SummarizedExperiment</code> and contains 9
genes for 27 samples.</p>
<ul>
<li><p>Explore the object’s <code>colData</code> and interprete it as the data’s
experimental design.</p></li>
<li><p>Generate a figure similar to the one below for your data own data.</p></li>
</ul>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-247"></span>
<p class="caption marginnote shownote">
Figure 9.7: Distribution of the gene expression in each group.
</p>
<img src="figs/gene_exp_se.png" alt="Distribution of the gene expression in each group." width="1518">
</div>
<ul>
<li><p>Calculate the average expression for each gene in each group.</p></li>
<li><p>Calculate the log2 fold-change (i.e. the difference of expression
between two groups) for groups CTRL0 and DRUG5. Which gene shows the
highest fold-change (i.e. where the expression increases most in
DRUG5)? Can you recognise this on the figure?</p></li>
<li><p>Use the <code>var()</code> function to calculate the variance for each
gene. Which gene has the largest variance. Can you recognise this on
the figure?</p></li>
<li><p>Calculate the <a href="https://en.wikipedia.org/wiki/Coefficient_of_variation">coefficient of
variation</a>
of each gene. The coefficient of variation is defined as the ratio
between a gene’s standard deviation (that you can compute using the
<code>sd()</code> function) and it’s mean. Which gene has the largest
variance. Can you recognise this on the figure?</p></li>
<li><p>Visualise the distribution of the gene expression in each
sample. You can use either a boxplot, a violin plot or plot the
density of these distributions with <code>geom_density()</code>. Colour the
sample data based on the groups.</p></li>
</ul>
<p class="question-end">
<span class="fa fa-square-o solution-icon"></span>
</p>
</div>
</div>

</div>
</div>
<p style="text-align: center;">
<a href="sec-rr.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-prog.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2025-06-16
 using 
R version 4.5.0 (2025-04-11)
</p>
</div>
</div>



</body>
</html>
